---
title: "Mapping global shifts in Saccharomyces cerevisiae gene expression across asynchronous time trajectories with diffusion maps"
always_allow_html: yes
author: 
- Taylor Reiter
- Rachel Montpetit
- Ron Runnebaum
- C. Titus Brown
- Ben Montpetit
date: "`r format(Sys.time(), '%d %B, %Y')`"
graphics: yes
figsintext: yes
geometry: "left=1in,right=1in,top=1in,bottom=1in"
header-includes:
- \usepackage{verbatim}
- \newcommand{\comm}[1]{}
- \usepackage{float} \floatplacement{figure}{H}
- \newcommand{\beginsupplement}{\setcounter{table}{0}  \renewcommand{\thetable}{S\arabic{table}}
  \setcounter{figure}{0} \renewcommand{\thefigure}{S\arabic{figure}}}
- \usepackage[font={small,it}, labelfont={bf}]{caption}
- \usepackage{setspace}\singlespacing
keep_tex: yes
output:
  bookdown::pdf_document2:
    fig_caption: yes
    toc: false
---
  
```{r setup, include=FALSE, eval = T}
knitr::opts_chunk$set(echo = F, eval = T, message = F, warning = F, cache = T, 
                      include = T, fig.keep = 'high', fig.path = "Rmd_fig_out/",
                      dpi=400, fig.width = 6.5)
```

```{r libs, eval = T}
library(dplyr)
library(purrr)
library(tidyr)
library(tibble)
library(readr)
library(purrr)
library(stringr)
library(ggplot2)
library(ggthemes)
library(ggpubr)
library(ggrepel)
library(ggplotify)
library(ggcorrplot)
library(limma)
library(edgeR)
library(clusterProfiler)
library(org.Sc.sgd.db)
library(kableExtra)
library(gridExtra)
library(UpSetR)
library(cowplot)
library(knitr)
library(RColorBrewer)
set.seed(1)
```

```{r diffusion_mapping_functions}
## as implemented here: https://doi.org/10.6084/m9.figshare.12864011.v4
## standardize columns func
norm_mat <- function(mat){
  ms <- apply(mat, 2, function(col) (col - mean(col)) / sd(col))
  ms
}

## compute euclid dist and standrdize to similarities
get_euc <- function(mat, n.threads = 1, alt = 'euclidean'){
  if(n.threads > 1){
    eu <- parDist(mat, method = alt, threads = n.threads) %>% as.matrix()
    ieu <- 1 / eu
    diag(ieu) <- 0
  }
  
  if(n.threads == 1){
    eu <- dist(mat, method = alt) %>% as.matrix()
    ieu <- 1 / eu
    diag(ieu) <- 0
  }

  ieu
}

## function to threshold the normalized distance mat
threshold <- function(mat, top_k = 10){
  thr <- mat
  tnr <- nrow(thr)
  ## set similarities outside of the top k to 0
  for(i in 1:tnr){
    ## rank entries in each row in reverse order (so largest value == rank 1), 
    ## and set entries that are outside of 1:k to 0
    thr[i, !rank(-thr[i, ], ties.method = 'random') %in% 1:top_k] <- 0
  }
  
  for(i in 1:tnr){
    for(j in 1:tnr){
      if(thr[i, j] == 0 & thr[j, i] != 0){
        thr[i, j] <- thr[j, i]
      }
    }
  }
  thr
}

## function to calculate norm laplacian
get_laplac <- function(mat){
  L <- -mat
  S <- rowSums(mat)
  nL <- L / S
  diag(nL) <- 1
  nL
}
```

```{r processing_functions}
orf_to_common <- function(keys){
  common <- AnnotationDbi::select(org.Sc.sgd.db,
                                  keys = keys,
                                  columns=c("COMMON"),
                                  keytype="ORF")
  return(common)
}
orf_to_description <- function(keys){
  description <- AnnotationDbi::select(org.Sc.sgd.db::org.Sc.sgd.db,
                                  keys = keys,
                                  columns=c("DESCRIPTION"),
                                  keytype="ORF")
  return(description)
}
lm_eqn <- function(df, y, x){
    formula = as.formula(sprintf('%s ~ %s', y, x))
    m <- lm(formula, data=df);
    # formating the values into a summary string to print out
    # ~ give some space, but equal size and comma need to be quoted
    eq <- substitute(italic(target) == a + b %.% italic(input)*","~~italic(r)^2~"="~r2*","~~p~"="~italic(pvalue), 
         list(target = y,
              input = x,
              a = format(as.vector(coef(m)[1]), digits = 2), 
              b = format(as.vector(coef(m)[2]), digits = 2), 
             r2 = format(summary(m)$r.squared, digits = 3),
             # getting the pvalue is painful
             pvalue = format(summary(m)$coefficients[2,'Pr(>|t|)'], digits=1)
            )
          )
    as.character(as.expression(eq));                 
}

dc_contrast <- function(dc_fit, coeff, adj.p.filt = 0.01, filter_logFC = T){
  # Test diffusion component coefficient
  dc_con <- contrasts.fit(dc_fit, coefficients = coeff)   
  dc_b <- eBayes(dc_con)
  dc_top <- topTable(dc_b, sort.by = "logFC", number = Inf)
  dc_top <- dc_top[dc_top$adj.P.Val < adj.p.filt, ]
  
  # Translate ORF to Common
  dc_top$ORF <- rownames(dc_top)
  commons <- orf_to_common(dc_top$ORF)
  # subset to first occurrence of ORF
  commons <- commons[match(unique(commons$ORF), commons$ORF), ]
  commons <- commons[ , -2]
  commons$COMMON <- ifelse(is.na(commons$COMMON), commons$ORF, commons$COMMON)
  commons <- unique(commons[ , ])
  dc_top <- left_join(dc_top, commons, by = c("ORF"))
  
  # Add SGD Description 
  descriptions <- orf_to_description(dc_top$ORF)
  # subset to first occurrence of ORF
  descriptions <- descriptions[match(unique(descriptions$ORF), descriptions$ORF), ]
  descriptions <- descriptions[ , -2] # remove sgd id
  # descriptions$DESCRIPTION <- ifelse(is.na(descriptions$COMMON), descriptions$ORF, descriptions$DESCRIPTION)
  # descriptions <- unique(descriptions[ , ])
  dc_top <- left_join(dc_top, descriptions, by = c("ORF"))
  
  # normalize logfc based on min and max of DC
  normalizer <- max(dc_fit$design[ , coeff]) - min(dc_fit$design[ , coeff])
  dc_top <- dc_top %>%
    mutate(logFC_adj = logFC*normalizer) %>%
    arrange(logFC_adj) %>%
    dplyr::select(logFC, logFC_adj, AveExpr, t, P.Value, adj.P.Val, B, ORF, COMMON, DESCRIPTION)
  if(filter_logFC == F) {
    return(dc_top)
  } else {
    # Join Common to diffex results
    dc_top <- dc_top %>%
      filter(abs(logFC_adj) > 2) 
    return(dc_top)
  }
}

dc_viz <- function(dc_con){
  # separate to induced and repressed
  up <- dc_con[dc_con$logFC > 0, ]
  down <- dc_con[dc_con$logFC < 0, ] 
  
  # KEGG enrichment
  ekegg_up <- enrichKEGG(up$ORF, "sce", pAdjustMethod = "bonferroni") 
  plt_rows <- ekegg_up@result %>%
    filter(p.adjust < .05) %>%
    nrow()
  plt1 <- dotplot(ekegg_up, showCategory = plt_rows, font.size = 5) + theme_minimal(base_size = 5)
  
  ekegg_down <- enrichKEGG(down$ORF, "sce", pAdjustMethod = "bonferroni")
  plt_rows <- ekegg_down@result %>%
    filter(p.adjust < .05) %>%
    nrow()
  plt2 <- dotplot(ekegg_down, showCategory = plt_rows, font.size = 5) + theme_minimal(base_size = 5)
  
  # GO enrichment
  ego_up <- enrichGO(up$COMMON, OrgDb = org.Sc.sgd.db, keyType = "GENENAME", 
                     ont = "ALL", pAdjustMethod = "bonferroni")
  plt_rows <- ego_up@result %>%
    filter(p.adjust < .05) %>%
    nrow()
  plt3 <- dotplot(ego_up, showCategory = plt_rows, font.size = 5) + theme_minimal(base_size = 5)
  
  ego_down <- enrichGO(down$COMMON, OrgDb = org.Sc.sgd.db, keyType = "GENENAME", 
                       ont = "ALL", pAdjustMethod = "bonferroni")
  plt_rows <- ego_down@result %>%
    filter(p.adjust < .05) %>%
    nrow()
  plt4 <- dotplot(ego_down, showCategory = plt_rows, font.size = 5) + theme_minimal(base_size = 5)
  
  # combine plots
  ggarrange(plt1, plt2, plt3, plt4, ncol = 2, nrow = 2, common.legend = T, 
            legend = "bottom", labels = c("A", "B", "C", "D"),
            heights = c(1, 2))
}

# this plot only makes sense when there are > ~10 genes per component that are
# differentially expressed
dc_top_enrich_plot <- function(dc, dc_num, dc_con, color_by = c("brix", "ava", "hr"), show_genes = F){
  # dc <- dc19
  # dc_num <- "DC3"
  # dc_con <- dc3_19
  # separate to induced and repressed -- note that filtering for 
  # log2fc is done in dc_contrast() when normalize_logFC = T. 
  # default is a normalizes logFC > abs(2) 
  # therefore, up down separates the positive and negative logFC values
  
  up <- dc_con[dc_con$logFC > 0, ]
  down <- dc_con[dc_con$logFC < 0, ] 
               
  # KEGG enrichment
  if(nrow(up) != 0){
    ekegg_up <- enrichKEGG(up$ORF, "sce", pAdjustMethod = "bonferroni") 
    ekegg_up_top <- ekegg_up@result %>%
      filter(p.adjust < 0.05) %>%
      dplyr::select(Description, geneID) %>%
      mutate(enrichment = "KEGG") 
  } else {
    ekegg_up_top <- data.frame("Description" = character(),
                               "geneID" = character(),
                               "enrichment" = character())
  }
  
  if(nrow(down) != 0){
    if(!identical(down$ORF, down$COMMON)) { # Filter out datasets where everything is unknown
      ekegg_down <- enrichKEGG(down$ORF, "sce", pAdjustMethod = "bonferroni")
      ekegg_down_top <- ekegg_down@result %>%
        filter(p.adjust < 0.05) %>%
        dplyr::select(Description, geneID) %>%
        mutate(enrichment = "KEGG")
    } else {
      ekegg_down_top <- data.frame("Description" = character(),
                                   "geneID" = character(),
                                   "enrichment" = character())
    }
  } else {
    ekegg_down_top <- data.frame("Description" = character(),
                                 "geneID" = character(),
                                 "enrichment" = character())
  }
  
  # GO enrichment
  if(nrow(up) != 0){
    ego_bp_up <- enrichGO(up$COMMON, OrgDb = org.Sc.sgd.db, keyType = "GENENAME", 
                          ont = "BP", pAdjustMethod = "bonferroni")
    ego_bp_up_top <- ego_bp_up@result %>%
      filter(p.adjust < 0.05) %>%
      dplyr::select(Description, geneID) %>%
      mutate(enrichment = "GO BP")
    
    ego_mf_up <- enrichGO(up$COMMON, OrgDb = org.Sc.sgd.db, keyType = "GENENAME", 
                          ont = "MF", pAdjustMethod = "bonferroni")
    ego_mf_up_top <- ego_mf_up@result%>%
      filter(p.adjust < 0.05) %>%
      dplyr::select(Description, geneID) %>%
      mutate(enrichment = "GO MF")
    
    ego_cc_up <- enrichGO(up$COMMON, OrgDb = org.Sc.sgd.db, keyType = "GENENAME", 
                          ont = "CC", pAdjustMethod = "bonferroni")
    ego_cc_up_top <- ego_cc_up@result %>%
      filter(p.adjust < 0.05) %>%
      dplyr::select(Description, geneID) %>%
      mutate(enrichment = "GO CC")
  } else {
    ego_cc_up_top <- ego_mf_up_top <- ego_bp_up_top <- data.frame("Description" = character(),
                                                                  "geneID" = character(),
                                                                  "enrichment" = character())
  }
  
  if(nrow(down) != 0){
    if(!identical(down$ORF, down$COMMON)) { # Filter out datasets where everything is unknown
      ego_bp_down <- enrichGO(down$COMMON, OrgDb = org.Sc.sgd.db, keyType = "GENENAME", 
                              ont = "BP", pAdjustMethod = "bonferroni")
      ego_bp_down_top <- ego_bp_down@result %>%
        filter(p.adjust < 0.05) %>%
        dplyr::select(Description, geneID) %>%
        mutate(enrichment = "GO BP")
      
      ego_mf_down <- enrichGO(down$COMMON, OrgDb = org.Sc.sgd.db, keyType = "GENENAME", 
                              ont = "MF", pAdjustMethod = "bonferroni")
      ego_mf_down_top <- ego_mf_down@result %>%
        filter(p.adjust < 0.05) %>%
        dplyr::select(Description, geneID) %>%
        mutate(enrichment = "GO MF")
      
      ego_cc_down <- enrichGO(down$COMMON, OrgDb = org.Sc.sgd.db, keyType = "GENENAME", 
                              ont = "CC", pAdjustMethod = "bonferroni")
      if(class(ego_cc_down) == "NULL") {
        message("No enriched GO CC Down, moving on!")
        # make empty df so obj is rbind compliant
        ego_cc_down_top <- data.frame("Description" = character(), 
                                      geneID = character(), 
                                      enrichment = character())
      } else {
        ego_cc_down_top <- ego_cc_down@result %>%
          filter(p.adjust < 0.05) %>%
          dplyr::select(Description, geneID) %>%
          mutate(enrichment = "GO CC")
      } 
    } else {
      ego_cc_down_top <- ego_mf_down_top <- ego_bp_down_top <- data.frame("Description" = character(),
                                                                          "geneID" = character(),
                                                                          "enrichment" = character())
    }
  }  else {
    ego_cc_down_top <- ego_mf_down_top <- ego_bp_down_top <- data.frame("Description" = character(),
                                                                        "geneID" = character(),
                                                                        "enrichment" = character())
  }
  
  # COMBINE ENRICHMENT TABLES
  if(show_genes == T){
    down_tbl <- rbind(ego_bp_down_top, ego_cc_down_top, ego_mf_down_top, ekegg_down_top) 
    up_tbl <- rbind(ego_bp_up_top, ego_cc_up_top, ego_mf_up_top, ekegg_up_top)
  } else if(show_genes == F) {
    down_tbl <- rbind(ego_bp_down_top, ego_cc_down_top, ego_mf_down_top, ekegg_down_top) %>%
      dplyr::select(-geneID, -enrichment, Pathway = Description)
    up_tbl <- rbind(ego_bp_up_top, ego_cc_up_top, ego_mf_up_top, ekegg_up_top) %>%
      dplyr::select(-geneID, -enrichment, Pathway = Description)
  }


  # PRUNE PATHWAY NAMES TO FIRST X WORDS
  up_tbl$Pathway <- gsub(" \\(.*\\)", "", up_tbl$Pathway)
  up_tbl$Pathway <-gsub(" SSU.* from tricistronic rRNA transcript", " rRNA", up_tbl$Pathway)
  down_tbl$Pathway <- gsub(" \\(.*\\)", "", down_tbl$Pathway)
  down_tbl$Pathway <-gsub(" SSU.* from tricistronic rRNA transcript", " rRNA", down_tbl$Pathway)
  
  # PLOT ANNOTATION TABLES
  if(nrow(up_tbl) == 0){
    # make an empty table if there are no results
    up_tbl <- ggtexttable(data.frame("No_sig_genes" = ""), rows = NULL, theme = ttheme(base_size = 5, base_style = "blank", padding = unit(c(.5, .5), "mm")))
  } else {
    up_tbl <- ggtexttable(up_tbl,rows = NULL, theme = ttheme(base_size = 5, base_style = "blank", padding = unit(c(.5, .5), "mm")))
  }
  
  if(nrow(down_tbl) == 0){
    # make an empty table if there are no results
    down_tbl <- ggtexttable(data.frame("No_sig_genes" = ""), rows = NULL, theme = ttheme(base_size = 5, base_style = "blank", padding = unit(c(.5, .5), "mm")))
  } else {
    down_tbl <- ggtexttable(down_tbl,rows = NULL, theme = ttheme(base_size = 5, base_style = "blank", padding = unit(c(.5, .5), "mm")))
  }
  
  # PLOT DC
  # prep DF for plotting. Rename DC* to DC so that the same innvocation can be used
  # for ggplot across all plots
  dc <- dc %>%
    dplyr::select(DC = all_of(dc_num), brix, ava_id, ava, hr, sample)
  
  if(color_by == "brix"){
    plt_dc <- ggplot(dc, aes(x = DC, y = 0, color = brix)) +
      geom_point(position = position_jitter(seed = 42)) +
      scale_color_viridis_c() +      
      labs(x = dc_num, color = "Brix") +
      theme_minimal() +
      theme(plot.title = element_text(hjust = 0.5),
            axis.text.y = element_blank(),
            axis.title.y = element_blank(),
            legend.position = "bottom") 
  } else if(color_by == "ava") {
    plt_dc <- ggplot(dc, aes(x = DC, y = 0, color = ava)) +
      geom_point(position = position_jitter(seed = 42)) +
      theme_minimal() +
      labs(x = dc_num, color = "AVA") +
      theme(plot.title = element_text(hjust = 0.5),
            axis.text.y = element_blank(),
            axis.title.y = element_blank(),
            legend.position = "bottom") +
      scale_color_manual(values = c("OR"=or,  "AV" = av, "SNC" = snc, "RRV" = rrv, "CRN" = crn,
                                    "AS" = as, "SMV" = smv, "SRH" = srh)) 
  } else if(color_by == "hr"){
    plt_dc <- ggplot(dc, aes(x = DC, y = 0, color = factor(hr, levels = c("0", "2", "6", "16", "64", "112")))) +
      geom_point() +
      scale_color_brewer(palette = "Paired", name = "hour") + 
      geom_point(position = position_jitter(seed = 42)) +
      labs(x = dc_num) +
      theme_minimal() +
      theme(plot.title = element_text(hjust = 0.5),
            axis.text.y = element_blank(),
            axis.title.y = element_blank(),
            legend.position = "bottom")
    }

  # Plot chart and table into one object
  return(ggarrange(down_tbl, plt_dc, up_tbl, ncol = 3))
}

dc_top_diffex_plot <- function(dc, dc_num, dc_con, color_by = c("brix", "ava", "hr")){
  # dc <- dc19
  # dc_num <- "DC7"
  # dc_con <- dc7_19
  # color_by = "ava"
  # separate to induced and repressed -- note that filtering for 
  # log2fc is done in dc_contrast() when normalize_logFC = T. 
  # default is a normalizes logFC > abs(2) 
  # therefore, up down separates the positive and negative logFC values
  
  up_most <- dc_con %>%
    filter(logFC_adj > 0) %>%
    dplyr::select(Gene = COMMON, logFC_adj) %>%
    mutate(logFC = round(logFC_adj, digits = 1)) %>%
    dplyr::select(-logFC_adj) %>%
    arrange(desc(logFC)) %>%
    head(n = 10)
  
  down_most <- dc_con %>%
    filter(logFC_adj < 0) %>%
    dplyr::select(Gene = COMMON, logFC_adj) %>%
    mutate(logFC = round(logFC_adj, digits = 1)) %>% 
    dplyr::select(-logFC_adj) %>%
    head(n = 10)
  
  # COMBINE ENRICHMENT TABLES
  # make an annotation table
  
  if(nrow(up_most) == 0){
    # make an empty table if there are no results
    up_tbl <- ggtexttable(data.frame("No_sig_genes" = ""), rows = NULL, theme = ttheme(base_size = 6, base_style = "classic", padding = unit(c(1, 1), "mm")))
  } else{
    up_tbl <- ggtexttable(up_most, rows = NULL, theme = ttheme(base_size = 6, base_style = "classic", padding = unit(c(1, 1), "mm")))
  }
  if(nrow(down_most) == 0){
    # make an empty table if there are no results
    down_tbl <- ggtexttable(data.frame("No_sig_genes" = ""), rows = NULL, theme = ttheme(base_size = 6, base_style = "classic", padding = unit(c(1, 1), "mm")))
  } else {
    down_tbl <- ggtexttable(down_most,rows = NULL, theme = ttheme(base_size = 6, base_style = "classic", padding = unit(c(1, 1), "mm")))
  }
  # PLOT DC
  # prep DF for plotting. Rename DC* to DC so that the same innvocation can be used
  # for ggplot across all plots
  dc <- dc %>%
    dplyr::select(DC = all_of(dc_num), brix, ava_id, ava, hr, sample)
  
  if(color_by == "brix"){
    plt_dc <- ggplot(dc, aes(x = DC, y = 0, color = brix)) +
      geom_point(position = position_jitter(seed = 42)) +
      scale_color_viridis_c() +
      labs(x = dc_num, color = "Brix") +
      theme_minimal() +
      theme(plot.title = element_text(hjust = 0.5),
            axis.text.y = element_blank(),
            axis.title.y = element_blank(),
            legend.position = "bottom") 
  } else if(color_by == "ava") {
    plt_dc <- ggplot(dc, aes(x = DC, y = 0, color = ava)) +
      geom_point(position = position_jitter(seed = 42)) +
      theme_minimal() +
      labs(x = dc_num, color = "AVA") +
      theme(plot.title = element_text(hjust = 0.5),
            axis.text.y = element_blank(),
            axis.title.y = element_blank(),
            legend.position = "bottom") +
      scale_color_manual(values = c("OR"=or,  "AV" = av, "SNC" = snc, "RRV" = rrv, "CRN" = crn,
                                    "AS" = as, "SMV" = smv, "SRH" = srh)) 
  } else if(color_by == "hr"){
    plt_dc <- ggplot(dc, aes(x = DC, y = 0, color = factor(hr, levels = c("0", "2", "6", "16", "64", "112")))) +
      geom_point() +
      scale_color_brewer(palette = "Paired", name = "hour") + 
      geom_point(position = position_jitter(seed = 42)) +
      labs(x = dc_num) +
      theme_minimal() +
      theme(plot.title = element_text(hjust = 0.5),
            axis.text.y = element_blank(),
            axis.title.y = element_blank(),
            legend.position = "bottom")
  } else {
      message("Please try a legitimate color values, either brix, ava, or hr")
    }

  # Plot chart and table into one object
  return(ggarrange(down_tbl, plt_dc, up_tbl, ncol = 3))
}

dc_top_table <- function(dc_con){
  up_most <- dc_con %>%
    filter(logFC > 0) %>%
    dplyr::select(ORF, Gene = COMMON, logFC_adj) %>%
    mutate(logFC = round(logFC_adj, digits = 1)) %>%
    dplyr::select(-logFC_adj) %>%
    head(n = 15)
  
  down_most <- dc_con %>%
    filter(logFC < 0) %>%
    dplyr::select(ORF, Gene = COMMON, logFC_adj) %>%
    mutate(logFC = round(logFC_adj, digits = 1)) %>%
    dplyr::select(-logFC_adj) %>%
    head(n = 15)
  
  most <- rbind(up_most, down_most)
  
  # add blank rows to enable printing when fewer than 30 diffex genes
  if(30-nrow(most) == 0){
    return(most)
  } else {
    blank_rows_to_add <- 30-nrow(most)
    blank_rows <- data.frame(ORF = rep("", blank_rows_to_add),
                             Gene = rep("", blank_rows_to_add),
                             logFC = rep("", blank_rows_to_add))
    most <- rbind(most, blank_rows)
    return(most)
  }
}
```

```{r colors_avas}
#  ordered south to north
srh = "#FFAABB" # pink
smv = "#EE8866" # orange
as  = "#EEDD88" # light yellow
snc = "#99DDFF" # light cyan 
crn = "#AAAA00" # olive
rrv = "#BBCC33" # pear 
av  = "#77AADD" # light blue
or  = "#DDDDDD" # pale grey
```

# Results and Discussion

## Diffusion mapping captures the global shift of gene expression during primary fermentation

```{r metadata}
info <- read_csv("samples_ava.csv")
info$ava_id <- factor(info$ava_id, levels = c('OR1', 'OR2', 'AV1', 'AV2', 
                                               'SNC1', 'SNC2', 'RRV1', 'RRV2', 'RRV3',
                                              'CRN1', 'AS1', 'AS2', 'SMV1', 
                                              'SMV2','SRH1'))

info$ava <- factor(info$ava, levels = c('OR', 'AV', 'SNC', 'RRV', 'CRN', 
                                        'AS', 'SMV','SRH'))
```

```{r diffusion_mapping_17, include = F}
## preprocess counts
counts17 <- read_tsv("v2017/outputs/counts/raw_counts.tsv") %>%
  filter(grepl(pattern = "^Y", x = gene)) %>%
  as.data.frame() %>%
  column_to_rownames("gene")
colnames(counts17) <- gsub("_readcounts.txt", "", colnames(counts17))
m17 <- sweep(counts17, 2, colSums(counts17), `/`)
m17 <- t(m17)

## build diffusion map
## as implemented here: https://doi.org/10.6084/m9.figshare.12864011.v4

nm <- m17 %>% norm_mat()          # normalize
aff <- nm %>%
  get_euc(., n.threads = 1) %>% 
  threshold(., top_k = 10)      # make affinity matrix
Lij <- aff %>% get_laplac()     # compute laplacian
eig <- eigen(Lij)               # smallest keig vectors
evl <- eig$values %>%           # get eigenvalues
  Re() %>%
  round(., digits = 10)
evc <- eig$vectors %>%          # get eigenvectors
  Re() %>%
  round(., digits = 10)

# create objects containing diffusion components
for(d in 1:length(evl)){
  assign(paste('dim', d, sep = '_'),
         Re(evc[, rank(evl) == (d + 1)])
  )
}

k_eig <- 10                     # specify num of variables you want to keep
dat <- do.call(mapply, c(FUN = cbind, mget(paste0("dim_", 1:(k_eig))))) %>%
  t() %>%
  as.data.frame()               # merge in array

colnames(dat) <- paste(paste('DC', 1:(k_eig), sep = ''))
rownames(dat) <- rownames(Lij)  # add labels

dc17 <- dat %>%                  # join with metadata
  rownames_to_column("sample") %>%
  left_join(info, by = c("sample" = "id"))
```

```{r plot_eig_vals17, eval = F}
tmp <- data.frame(val = eig$values) %>%
  arrange(val) %>%
  mutate(num = 1:length(eig$values))

ggplot(tmp, aes(y = val, x = num)) + 
  geom_point() +
  theme_minimal() +
  ggtitle("k=10 2017")
```

```{r diffusion_mapping_19, include = F}
counts19 <- read_tsv("v2019/outputs/counts/raw_counts.tsv") %>%
  dplyr::filter(grepl(pattern = "^Y", x = gene)) %>%
  dplyr::select(-`2019_inoculum_SRH1_readcounts.txt`,
                -`2019_inoculum_AS1_readcounts.txt`, 
                -`2019_inoculum_SMV2_readcounts.txt`,
                -`2019_inoculum_AS2_readcounts.txt`,
                -`2019_inoculum_RRV1_readcounts.txt`,
                -`2019_inoculum_SMV1_readcounts.txt`) %>%
  as.data.frame() %>%
  column_to_rownames("gene")
colnames(counts19) <- gsub("_readcounts.txt", "", colnames(counts19))

m19 <- sweep(counts19, 2, colSums(counts19), `/`)
m19 <- t(m19)                                             # transpose counts

## build diffusion map
## as implemented here: https://doi.org/10.6084/m9.figshare.12864011.v4

nm <- m19 %>% norm_mat()        # normalize
aff <- nm %>%
  get_euc(., n.threads = 1) %>% 
  threshold(., top_k = 10)      # make affinity matrix
Lij <- aff %>% get_laplac()     # compute laplacian
eig <- eigen(Lij)               # smallest keig vectors
evl <- eig$values %>%           # get eigenvalues
  Re() %>%
  round(., digits = 10)
evc <- eig$vectors %>%          # get eigenvectors
  Re() %>%
  round(., digits = 10)

# create objects containing diffusion components
for(d in 1:length(evl)){
  assign(paste('dim', d, sep = '_'),
         Re(evc[, rank(evl) == (d + 1)])
  )
}

k_eig <- 10                     # specify num of variables you want to keep
dat <- do.call(mapply, c(FUN = cbind, mget(paste0("dim_", 1:(k_eig))))) %>%
  t() %>%
  as.data.frame()               # merge in array

colnames(dat) <- paste(paste('DC', 1:(k_eig), sep = ''))
rownames(dat) <- rownames(Lij)  # add labels

dc19 <- dat %>%                 # join with metadata
  rownames_to_column("sample") %>%
  left_join(info, by = c("sample" = "id"))
```

```{r plot_eig_vals19, eval = F}
tmp <- data.frame(val = eig$values) %>%
  arrange(val) %>%
  mutate(num = 1:length(eig$values))

ggplot(tmp, aes(y = val, x = num)) + 
  geom_point() +
  theme_minimal() +
  ggtitle("k=10 2019")
```

```{r diffex17}
d0_17 <- DGEList(counts17)                         # initialize the DGE object
d17 <- calcNormFactors(d0_17)
dc_mm17 <- model.matrix(~dc17$DC1 + ~dc17$DC2 + ~dc17$DC3 + dc17$DC4 + dc17$DC5 + 
                          dc17$DC6 + dc17$DC7 + dc17$DC8 + dc17$DC9 + dc17$DC10)
dc_y17 <- voom(d17, dc_mm17, plot = F)
dc_fit17 <- lmFit(dc_y17, dc_mm17)
dc1_17 <- dc_contrast(dc_fit = dc_fit17, coeff = 2) # Test "DC1" coefficient
dc2_17 <- dc_contrast(dc_fit = dc_fit17, coeff = 3) # Test "DC2" coefficient
dc3_17 <- dc_contrast(dc_fit = dc_fit17, coeff = 4) # Test "DC3" coefficient
dc4_17 <- dc_contrast(dc_fit = dc_fit17, coeff = 5) # Test "DC4" coefficient
dc5_17 <- dc_contrast(dc_fit = dc_fit17, coeff = 6) # Test "DC5" coefficient
dc6_17 <- dc_contrast(dc_fit = dc_fit17, coeff = 7) # Test "DC6" coefficient
dc7_17 <- dc_contrast(dc_fit = dc_fit17, coeff = 8) # Test "DC7" coefficient
dc8_17 <- dc_contrast(dc_fit = dc_fit17, coeff = 9) # Test "DC8" coefficient
```

```{r diffex19}
d0_19 <- DGEList(counts19)              # initialize the DGE object
d19 <-calcNormFactors(d0_19)            # calculate normalization factors
dc_mm19 <- model.matrix(~dc19$DC1 + ~dc19$DC2 + ~dc19$DC3 + dc19$DC4 + dc19$DC5 + 
                          dc19$DC6 + dc19$DC7 + ~dc19$DC8 + dc19$DC9 + dc19$DC10)
dc_y19 <- voom(d19, dc_mm19, plot = F)
dc_fit19 <- lmFit(dc_y19, dc_mm19)
dc1_19 <- dc_contrast(dc_fit = dc_fit19, coeff = 2) # Test "DC1" coefficient
dc2_19 <- dc_contrast(dc_fit = dc_fit19, coeff = 3) # Test "DC2" coefficient
dc3_19 <- dc_contrast(dc_fit = dc_fit19, coeff = 4) # Test "DC3" coefficient
dc4_19 <- dc_contrast(dc_fit = dc_fit19, coeff = 5) # Test "DC4" coefficient
dc5_19 <- dc_contrast(dc_fit = dc_fit19, coeff = 6) # Test "DC5" coefficient
dc6_19 <- dc_contrast(dc_fit = dc_fit19, coeff = 7) # Test "DC6" coefficient
dc7_19 <- dc_contrast(dc_fit = dc_fit19, coeff = 8) # Test "DC7" coefficient
dc8_19 <- dc_contrast(dc_fit = dc_fit19, coeff = 9) # Test "DC8" coefficient
```

```{r supplemental_table_wine, eval = F}
library(xlsx)
write.xlsx(dc1_17, file="supplemental_tables/supplementary_table1.xlsx", sheetName="DC1_2017", row.names=FALSE)
write.xlsx(dc2_17, file="supplemental_tables/supplementary_table1.xlsx", sheetName="DC2_2017", append = T, row.names=FALSE)
write.xlsx(dc3_17, file="supplemental_tables/supplementary_table1.xlsx", sheetName="DC3_2017", append = T, row.names=FALSE)
write.xlsx(dc4_17, file="supplemental_tables/supplementary_table1.xlsx", sheetName="DC4_2017", append = T, row.names=FALSE)
write.xlsx(dc5_17, file="supplemental_tables/supplementary_table1.xlsx", sheetName="DC5_2017", append = T, row.names=FALSE)
write.xlsx(dc6_17, file="supplemental_tables/supplementary_table1.xlsx", sheetName="DC6_2017", append = T, row.names=FALSE)
write.xlsx(dc7_17, file="supplemental_tables/supplementary_table1.xlsx", sheetName="DC7_2017", append = T, row.names=FALSE)
write.xlsx(dc8_17, file="supplemental_tables/supplementary_table1.xlsx", sheetName="DC8_2017", append = T, row.names=FALSE)
write.xlsx(dc1_19, file="supplemental_tables/supplementary_table1.xlsx", sheetName="DC1_2019", append = T, row.names=FALSE)
write.xlsx(dc2_19, file="supplemental_tables/supplementary_table1.xlsx", sheetName="DC2_2019", append = T, row.names=FALSE)
write.xlsx(dc3_19, file="supplemental_tables/supplementary_table1.xlsx", sheetName="DC3_2019", append = T, row.names=FALSE)
write.xlsx(dc4_19, file="supplemental_tables/supplementary_table1.xlsx", sheetName="DC4_2019", append = T, row.names=FALSE)
write.xlsx(dc5_19, file="supplemental_tables/supplementary_table1.xlsx", sheetName="DC5_2019", append = T, row.names=FALSE)
write.xlsx(dc6_19, file="supplemental_tables/supplementary_table1.xlsx", sheetName="DC6_2019", append = T, row.names=FALSE)
write.xlsx(dc7_19, file="supplemental_tables/supplementary_table1.xlsx", sheetName="DC7_2019", append = T, row.names=FALSE)
write.xlsx(dc8_19, file="supplemental_tables/supplementary_table1.xlsx", sheetName="DC8_2019", append = T, row.names=FALSE)
```

```{r brixplot, fig.width = 3.25, fig.height = 5.5}
brix17 <- read_csv("../2020-pn-tmp/site_metadata/2017_fermentation_log.csv")
brix19 <- read_csv("../2020-pn-tmp/site_metadata/2019_fermentation_log.csv") 
brix <- rbind(brix17, brix19) %>%
  filter(!is.na(hours_post_innoculation))%>%
  dplyr::select(-time) %>%
  pivot_longer(cols = A:D, names_to = "tank", values_to = "brix") %>%
  mutate(ava = gsub("[123]", "", ava_id))
brix_plot <- ggplot(brix, aes(x = hours_post_innoculation, y = brix, color = ava_id, shape = tank)) +
  geom_line(alpha = .7) +
  facet_wrap(~year, ncol = 1, nrow = 2) +
  labs(x = "time (hours)", y = "Brix", color = "site", shape = "tank") +
  theme_minimal() +
  theme(legend.position = "bottom",
        strip.text.x = element_text(size = 14)) +
  scale_color_manual(values = c(SRH1 = srh, SMV2 = smv, SMV1 = smv, AS1 = as, 
                                AS2 = as, SNC1 = snc, SNC2 = snc, CRN1 = crn, 
                                RRV1 = rrv, RRV2 = rrv, RRV3 = rrv, AV1 = av,
                                AV2 = av, OR1 = or, OR2 = or),
                     breaks = c('OR1', 'OR2', 'AV1', 'AV2',  'SNC1','SNC2', 'RRV1', 'RRV2', 'RRV3',
                                'CRN1', 'AS1', 'AS2', 'SMV1', 'SMV2','SRH1')) +
  scale_shape(guide = 'none')
brix_plot
```

```{r diffex_brix_17}
brix_counts17  <- counts17                    # non-wine samples already filtered
d0_17 <- DGEList(brix_counts17)               # initialize the DGE object
d17 <- calcNormFactors(d0_17)
brix_mm17 <- model.matrix(~dc17$brix)
brix_y17 <- voom(d17, brix_mm17, plot = F)
brix_fit17 <- lmFit(brix_y17, brix_mm17)
brix_17 <- dc_contrast(dc_fit = brix_fit17, coeff = 2) # Test brix coefficient
```

```{r diffex_brix_19, eval = T}
brix_counts19  <- counts19[ , 1:150]               # remove inoculum samples
d0_19 <- DGEList(brix_counts19)                    # initialize the DGE object
d19 <- calcNormFactors(d0_19)
brix19 <- info %>%
  dplyr::filter(year == 2019) %>%
  dplyr::filter(!is.na(brix))
brix_mm19 <- model.matrix(~brix19$brix)
brix_y19 <- voom(d19, brix_mm19, plot = F)
brix_fit19 <- lmFit(brix_y19, brix_mm19)
brix_19 <- dc_contrast(dc_fit = brix_fit19, coeff = 2) # Test brix coefficient
```

```{r brixcorr}
# make sure that everything has the same number of genes by not filtering by expression
# consider doing this from the outset (see above code) instead of just for this
# figure

brix_19_all <- dc_contrast(dc_fit = brix_fit19, coeff = 2, 
                           adj.p.filt = 1, filter_logFC = F) %>%
  arrange(ORF) %>%
  mutate(logFC = logFC*-1)
brix_17_all <- dc_contrast(dc_fit = brix_fit17, coeff = 2,
                           adj.p.filt = 1, filter_logFC = F) %>%
  arrange(ORF) %>%
  mutate(logFC = logFC*-1)
dc1_17_all <- dc_contrast(dc_fit = dc_fit17, coeff = 2, 
                          adj.p.filt = 1, filter_logFC = F) %>%
  arrange(ORF)
dc1_19_all <- dc_contrast(dc_fit = dc_fit19, coeff = 2,
                          adj.p.filt = 1, filter_logFC = F) %>%
  arrange(ORF)

# all.equal(brix_19_all$ORF, dc1_19_all$ORF, brix_17_all$ORF, dc1_17_all$ORF)

all_l2fc <- data.frame(brix19 = brix_19_all$logFC,
                       brix17 = brix_17_all$logFC,
                       DC1_17 = dc1_17_all$logFC,
                       DC1_19 = dc1_19_all$logFC) 
corr_l2fc <- cor(all_l2fc)
colnames(corr_l2fc) <- c("Brix 2019", "Brix 2017", "DC1 2017", "DC1 2019")
rownames(corr_l2fc) <- c("Brix 2019", "Brix 2017", "DC1 2017", "DC1 2019")


corr_plt <- ggcorrplot(corr_l2fc, type = "upper", show.diag = T, lab = T, 
                       hc.order = T, digits = 3, lab_size = 3, tl.cex = 9, 
                       lab_col = "white", show.legend = T,  legend.title =  "",
                       colors = brewer.pal(n = 3, name = "RdYlBu")) + 
  scale_fill_gradient2(low = brewer.pal(n = 3, name = "RdYlBu")[1], 
                       high = brewer.pal(n = 3, name = "RdYlBu")[3], 
                       mid = brewer.pal(n = 3, name = "RdYlBu")[2], 
                       midpoint = 0, limit = c(0, 1), space = "Lab", 
                       name = "") +
  theme(legend.position = "bottom")
```

```{r dc1}
p17 <- ggplot(dc17, aes(x = DC1, y = 0, color = brix, label = ava_id)) +
  geom_point(position = position_jitter(seed = 42)) +
  scale_color_viridis_c() +
  theme_minimal() +
  theme(plot.title = element_text(hjust = 0.5),
        axis.text.y = element_blank(),
        axis.title.y = element_blank()) +
  labs(title = "2017", color = "Brix") 

p19 <- ggplot(dc19, aes(x = DC1, y = 0, color = brix, label = ava_id)) +
  geom_point(position = position_jitter(seed = 42)) +
  scale_color_viridis_c() +
  theme_minimal() +
  theme(plot.title = element_text(hjust = 0.5),
        axis.text.y = element_blank(),
        axis.title.y = element_blank()) +
  labs(title = "2019", color = "Brix") 

# ggarrange(ggarrange(p17, p19, ncol = 1, nrow = 2, common.legend = T, legend = "bottom", labels = c(" ", " ")), corr_plt, labels = c("", " "), ncol = 1, nrow =2)
ggarrange(ggarrange(p17, p19, ncol = 1, nrow = 2, common.legend = T, legend = "bottom", labels = c(" ", " ")), corr_plt, labels = c("", " "), ncol = 2, nrow = 1)
```

```{r numdiffex}
num_diffex <- data.frame(diffusion_component = c("DC1", "DC2", "DC3", "DC4", 
                                                 "DC5", "DC6", "DC7", "DC8"),
                         induced = c(length(filter(dc1_17, logFC > 0)$ORF),
                                     length(filter(dc2_17, logFC > 0)$ORF),
                                     length(filter(dc3_17, logFC > 0)$ORF),
                                     length(filter(dc4_17, logFC > 0)$ORF),
                                     length(filter(dc5_17, logFC > 0)$ORF),
                                     length(filter(dc6_17, logFC > 0)$ORF),
                                     length(filter(dc7_17, logFC > 0)$ORF),
                                     length(filter(dc8_17, logFC > 0)$ORF)),
                         repressed = c(length(filter(dc1_17, logFC < 0)$ORF),
                                       length(filter(dc2_17, logFC < 0)$ORF),
                                       length(filter(dc3_17, logFC < 0)$ORF),
                                       length(filter(dc4_17, logFC < 0)$ORF),
                                       length(filter(dc5_17, logFC < 0)$ORF),
                                       length(filter(dc6_17, logFC < 0)$ORF),
                                       length(filter(dc7_17, logFC < 0)$ORF),
                                       length(filter(dc8_17, logFC < 0)$ORF)),
                         induced = c(length(filter(dc1_19, logFC > 0)$ORF),
                                     length(filter(dc2_19, logFC > 0)$ORF),
                                     length(filter(dc3_19, logFC > 0)$ORF),
                                     length(filter(dc4_19, logFC > 0)$ORF),
                                     length(filter(dc5_19, logFC > 0)$ORF),
                                     length(filter(dc6_19, logFC > 0)$ORF),
                                     length(filter(dc7_19, logFC > 0)$ORF),
                                     length(filter(dc8_19, logFC > 0)$ORF)),
                         repressed = c(length(filter(dc1_19, logFC < 0)$ORF),
                                       length(filter(dc2_19, logFC < 0)$ORF),
                                       length(filter(dc3_19, logFC < 0)$ORF),
                                       length(filter(dc4_19, logFC < 0)$ORF),
                                       length(filter(dc5_19, logFC < 0)$ORF),
                                       length(filter(dc6_19, logFC < 0)$ORF),
                                       length(filter(dc7_19, logFC < 0)$ORF),
                                       length(filter(dc8_19, logFC < 0)$ORF))
)

kable(num_diffex, "latex", booktabs = T, col.names = c("diffusion component", "induced", "repressed", "induced", "repressed"),
      caption = 'Number of significantly differentially expressed genes for the top diffusion components for the 2017 and 2019 vintages') %>%
  kable_styling(font_size = 9) %>%
  add_header_above(c(" " = 1, "2017" = 2, "2019" = 2), align = "l", escape = F)
```

```{r all_dc19_hour}
dc2_19p <- ggplot(dc19, aes(x = DC2, y = 0, color =  factor(hr, levels = c("0", "2", "6", "16", "64", "112")), label = ava_id)) +
  geom_point(alpha = .8, position = position_jitter(seed = 42)) +
  scale_color_brewer(palette = "Paired", name = "hour")  +
  xlim(c(-0.2, .2)) +
  theme_minimal() +
  theme(plot.title = element_text(hjust = 0.5),
        axis.text.y = element_blank(),
        axis.title.y = element_blank()) 

dc3_19p <- ggplot(dc19, aes(x = DC3, y = 0, color = factor(hr, levels = c("0", "2", "6", "16", "64", "112")), label = ava_id)) +
  geom_point(alpha = .8, position = position_jitter(seed = 42)) +
  scale_color_brewer(palette = "Paired", name = "hour")  +
  xlim(c(-0.2, .2)) +
  theme_minimal() +
  theme(plot.title = element_text(hjust = 0.5),
        axis.text.y = element_blank(),
        axis.title.y = element_blank())

dc4_19p <- ggplot(dc19, aes(x = DC4, y = 0, color = factor(hr, levels = c("0", "2", "6", "16", "64", "112")), label = ava_id)) +
  geom_point(alpha = .8, position = position_jitter(seed = 42)) +
  scale_color_brewer(palette = "Paired", name = "hour")  +
  xlim(c(-0.2, .2)) +
  theme_minimal() +
  theme(plot.title = element_text(hjust = 0.5),
        axis.text.y = element_blank(),
        axis.title.y = element_blank()) 

dc5_19p <-ggplot(dc19, aes(x = DC5, y = 0, color = factor(hr, levels = c("0", "2", "6", "16", "64", "112")), label = ava_id)) +
  geom_point(alpha = .8, position = position_jitter(seed = 42)) +
  xlim(c(-0.51, .51)) +
  scale_color_brewer(palette = "Paired", name = "hour")  +
  theme_minimal() +
  theme(plot.title = element_text(hjust = 0.5),
        axis.text.y = element_blank(),
        axis.title.y = element_blank()) 

dc6_19p <-ggplot(dc19, aes(x = DC6, y = 0, color = factor(hr, levels = c("0", "2", "6", "16", "64", "112")), label = ava_id)) +
  geom_point(alpha = .8, position = position_jitter(seed = 42)) +
  xlim(c(-0.51, .51)) +
  scale_color_brewer(palette = "Paired", name = "hour")  +
  theme_minimal() +
  theme(plot.title = element_text(hjust = 0.5),
        axis.text.y = element_blank(),
        axis.title.y = element_blank())

dc7_19p <-ggplot(dc19, aes(x = DC7, y = 0, color = factor(hr, levels = c("0", "2", "6", "16", "64", "112")), label = ava_id)) +
  geom_point(alpha = .8, position = position_jitter(seed = 42)) +
  xlim(c(-0.51, .51)) +
  scale_color_brewer(palette = "Paired", name = "hour")  +
  theme_minimal() +
  theme(plot.title = element_text(hjust = 0.5),
        axis.text.y = element_blank(),
        axis.title.y = element_blank()) 

dc8_19p <-ggplot(dc19, aes(x = DC8, y = 0, color = factor(hr, levels = c("0", "2", "6", "16", "64", "112")), label = ava_id)) +
  geom_point(alpha = .8, position = position_jitter(seed = 42)) +
  xlim(c(-0.51, .51)) +
  scale_color_brewer(palette = "Paired", name = "hour")  +
  theme_minimal() +
  theme(plot.title = element_text(hjust = 0.5),
        axis.text.y = element_blank(),
        axis.title.y = element_blank()) 


```

```{r all_dc19_ava}
dc2_19p_ava <- ggplot(dc19, aes(x = DC2, y = 0, color = ava, label = ava_id)) +
  geom_point(alpha = .8, position = position_jitter(seed = 42)) +
  scale_color_manual(values = c("OR"=or,  "AV" = av, "SNC" = snc, "RRV" = rrv, "CRN" = crn,
                                "AS" = as, "SMV" = smv, "SRH" = srh)) +
  xlim(c(-0.2, .2)) +
  theme_minimal() +
  theme(plot.title = element_text(hjust = 0.5),
        axis.text.y = element_blank(),
        axis.title.y = element_blank()) 

dc3_19p_ava <- ggplot(dc19, aes(x = DC3, y = 0, color = ava, label = ava_id)) +
  geom_point(alpha = .8, position = position_jitter(seed = 42)) +
  scale_color_manual(values = c("OR"=or,  "AV" = av, "SNC" = snc, "RRV" = rrv, "CRN" = crn,
                                "AS" = as, "SMV" = smv, "SRH" = srh)) +
  xlim(c(-0.2, .2)) +
  theme_minimal() +
  theme(plot.title = element_text(hjust = 0.5),
        axis.text.y = element_blank(),
        axis.title.y = element_blank())

dc4_19p_ava <- ggplot(dc19, aes(x = DC4, y = 0, color = ava, label = ava_id)) +
  geom_point(alpha = .8, position = position_jitter(seed = 42)) +
  scale_color_manual(values = c("OR"=or,  "AV" = av, "SNC" = snc, "RRV" = rrv, "CRN" = crn,
                                "AS" = as, "SMV" = smv, "SRH" = srh)) +
  xlim(c(-0.2, .2)) +
  theme_minimal() +
  theme(plot.title = element_text(hjust = 0.5),
        axis.text.y = element_blank(),
        axis.title.y = element_blank()) 

dc5_19p_ava <-ggplot(dc19, aes(x = DC5, y = 0, color = ava, label = ava_id)) +
  geom_point(alpha = .8, position = position_jitter(seed = 42)) +
  xlim(c(-0.51, .51)) +
  scale_color_manual(values = c("OR"=or,  "AV" = av, "SNC" = snc, "RRV" = rrv, "CRN" = crn,
                                "AS" = as, "SMV" = smv, "SRH" = srh)) +
  theme_minimal() +
  theme(plot.title = element_text(hjust = 0.5),
        axis.text.y = element_blank(),
        axis.title.y = element_blank()) 

dc6_19p_ava <-ggplot(dc19, aes(x = DC6, y = 0, color = ava, label = ava_id)) +
  geom_point(alpha = .8, position = position_jitter(seed = 42)) +
  xlim(c(-0.51, .51)) +
  scale_color_manual(values = c("OR"=or,  "AV" = av, "SNC" = snc, "RRV" = rrv, "CRN" = crn,
                                "AS" = as, "SMV" = smv, "SRH" = srh)) +
  theme_minimal() +
  theme(plot.title = element_text(hjust = 0.5),
        axis.text.y = element_blank(),
        axis.title.y = element_blank())

dc7_19p_ava <-ggplot(dc19, aes(x = DC7, y = 0, color = ava, label = ava_id)) +
  geom_point(alpha = .8, position = position_jitter(seed = 42)) +
  xlim(c(-0.51, .51)) +
  scale_color_manual(values = c("OR"=or,  "AV" = av, "SNC" = snc, "RRV" = rrv, "CRN" = crn,
                                "AS" = as, "SMV" = smv, "SRH" = srh)) +
  theme_minimal() +
  theme(plot.title = element_text(hjust = 0.5),
        axis.text.y = element_blank(),
        axis.title.y = element_blank()) 

dc8_19p_ava <-ggplot(dc19, aes(x = DC8, y = 0, color = ava, label = ava_id)) +
  geom_point(alpha = .8, position = position_jitter(seed = 42)) +
  xlim(c(-0.51, .51)) +
  scale_color_manual(values = c("OR"=or,  "AV" = av, "SNC" = snc, "RRV" = rrv, "CRN" = crn,
                                "AS" = as, "SMV" = smv, "SRH" = srh)) +
  theme_minimal() +
  theme(plot.title = element_text(hjust = 0.5),
        axis.text.y = element_blank(),
        axis.title.y = element_blank()) 


```

```{r DC2DC42019, fig.height= 3.7}

# combine relevant ava plts
dc2_dc4_19_ava <- ggarrange(dc2_19p_ava, dc3_19p_ava, dc4_19p_ava, 
                            ncol = 1, legend = "none",
                            common.legend = T, labels = c("A", "", ""))

# combine relevant hr plts
dc2_dc4_19_hr <- ggarrange(dc2_19p, dc3_19p, dc4_19p, ncol = 1, legend = "none",
                           common.legend = T, labels = c("B", "", ""))

ggarrange(dc2_dc4_19_ava, dc2_dc4_19_hr, legend = "none")
```

```{r DC5DC82019, fig.height= 4}

# combine relevant ava plts
dc5_dc8_19_ava <- ggarrange(dc5_19p_ava, dc6_19p_ava, dc7_19p_ava, dc8_19p_ava, 
                            ncol = 1, legend = "none", common.legend = T, 
                            labels = c("A", "", "", ""))

# combine relevant hr plts
dc5_dc8_19_hr <- ggarrange(dc5_19p, dc6_19p, dc7_19p, dc8_19p, ncol = 1, 
                           legend = "none", common.legend = T, 
                           labels = c("B", "", "", ""))

ggarrange(dc5_dc8_19_ava, dc5_dc8_19_hr, legend = "none")
```

```{r all_dc17_brix}
dc2_17p <- ggplot(dc17, aes(x = DC2, y = 0, color = brix, label = ava_id)) +
  geom_point(alpha = .8, position = position_jitter(seed = 42)) +
  scale_color_viridis_c() +
  labs(color = "Brix") +
  xlim(c(-0.3, .3)) +
  theme_minimal() +
  theme(plot.title = element_text(hjust = 0.5),
        axis.text.y = element_blank(),
        axis.title.y = element_blank())

dc3_17p <- ggplot(dc17, aes(x = DC3, y = 0, color = brix, label = ava_id)) +
  geom_point(alpha = .8, position = position_jitter(seed = 42)) +
  scale_color_viridis_c() +
  labs(color = "Brix") +
  xlim(c(-0.3, .3)) +
  theme_minimal() +
  theme(plot.title = element_text(hjust = 0.5),
        axis.text.y = element_blank(),
        axis.title.y = element_blank()) 

dc4_17p <- ggplot(dc17, aes(x = DC4, y = 0, color = brix, label = ava_id)) +
  geom_point(alpha = .8, position = position_jitter(seed = 42)) +
  scale_color_viridis_c() +
  xlim(c(-0.3, .3)) +
  labs(color = "Brix") +
  theme_minimal() +
  theme(plot.title = element_text(hjust = 0.5),
        axis.text.y = element_blank(),
        axis.title.y = element_blank()) 

all_dc_17 <- ggarrange(dc2_17p, dc3_17p, dc4_17p,
                       ncol = 1, common.legend = T, labels = c("B", "", ""))
```

```{r all_dc17_ava}
dc2_17p <- ggplot(dc17, aes(x = DC2, y = 0, color = ava, label = ava_id)) +
  geom_point(alpha = .8, position = position_jitter(seed = 42)) +
  scale_color_manual(values = c("OR"=or,  "AV" = av, "SNC" = snc, "RRV" = rrv, "CRN" = crn,
                                "AS" = as, "SMV" = smv, "SRH" = srh)) +
  xlim(c(-0.3, .3)) +
  labs(color = "AVA") +
  theme_minimal() +
  theme(plot.title = element_text(hjust = 0.5),
        axis.text.y = element_blank(),
        axis.title.y = element_blank()) 

dc3_17p <- ggplot(dc17, aes(x = DC3, y = 0, color = ava, label = ava_id)) +
  geom_point(alpha = .8, position = position_jitter(seed = 42)) +
  scale_color_manual(values = c("OR"=or,  "AV" = av, "SNC" = snc, "RRV" = rrv, "CRN" = crn,
                                "AS" = as, "SMV" = smv, "SRH" = srh)) +
  xlim(c(-0.3, .3)) +
  labs(color = "AVA") +
  theme_minimal() +
  theme(plot.title = element_text(hjust = 0.5),
        axis.text.y = element_blank(),
        axis.title.y = element_blank()) 

dc4_17p <- ggplot(dc17, aes(x = DC4, y = 0, color = ava, label = ava_id)) +
  geom_point(alpha = .8, position = position_jitter(seed = 42)) +
  scale_color_manual(values = c("OR"=or,  "AV" = av, "SNC" = snc, "RRV" = rrv, "CRN" = crn,
                                "AS" = as, "SMV" = smv, "SRH" = srh)) +
  xlim(c(-0.3, .3)) +
  labs(color = "AVA") +
  theme_minimal() +
  theme(plot.title = element_text(hjust = 0.5),
        axis.text.y = element_blank(),
        axis.title.y = element_blank()) 

all_dc_17_ava <- ggarrange(dc2_17p, dc3_17p, dc4_17p,
                           ncol = 1, common.legend = T, labels = c("A", "", ""))
```

```{r dm2017, fig.height= 3, fig.width = 6}
ggarrange(all_dc_17_ava, all_dc_17, legend = "none")
```

## Diffusion mapping identified global shifts in gene expression during hypoxia


```{r TEST_what_is_deleted, eval = F}
# figure out what dsc1 is 
tmp2 <- counts %>%
  rownames_to_column("gene") %>%
  filter(gene == "YOL073C") %>%
  select(contains("NH_78"))
  
counts_filt <- counts[rowSums(counts) > 0, ] # remove rows that sum to zero
dim(counts_filt)
tmp <- counts %>% 
  select(contains("NH_76"))
tmp$total_counts <- rowSums(tmp)
tmp2 <- tmp %>%
  rownames_to_column("gene") %>%
  select(gene, total_counts) %>%
  filter(total_counts == 0)
View(tmp2)

orfs <- orf_to_common(tmp2$gene)
```

```{r hypoxia_functions}
dc_top_enrich_plot_hyp <- function(dc, dc_num, dc_con, show_genes = F){
  # dc <- dc19
  # dc_num <- "DC3"
  # dc_con <- dc3_19
  # separate to induced and repressed -- note that filtering for 
  # log2fc is done in dc_contrast() when normalize_logFC = T. 
  # default is a normalizes logFC > abs(2) 
  # therefore, up down separates the positive and negative logFC values
  
  up <- dc_con[dc_con$logFC > 0, ]
  down <- dc_con[dc_con$logFC < 0, ] 
               
  # KEGG enrichment
  if(nrow(up) != 0){
    ekegg_up <- enrichKEGG(up$ORF, "sce", pAdjustMethod = "bonferroni") 
    ekegg_up_top <- ekegg_up@result %>%
      filter(p.adjust < 0.05) %>%
      dplyr::select(Description, geneID) %>%
      mutate(enrichment = "KEGG") 
  } else {
    ekegg_up_top <- data.frame("Description" = character(),
                               "geneID" = character(),
                               "enrichment" = character())
  }
  
  if(nrow(down) != 0){
    if(!identical(down$ORF, down$COMMON)) { # Filter out datasets where everything is unknown
      ekegg_down <- enrichKEGG(down$ORF, "sce", pAdjustMethod = "bonferroni")
      ekegg_down_top <- ekegg_down@result %>%
        filter(p.adjust < 0.05) %>%
        dplyr::select(Description, geneID) %>%
        mutate(enrichment = "KEGG")
    } else {
      ekegg_down_top <- data.frame("Description" = character(),
                                   "geneID" = character(),
                                   "enrichment" = character())
    }
  } else {
    ekegg_down_top <- data.frame("Description" = character(),
                                 "geneID" = character(),
                                 "enrichment" = character())
  }
  
  # GO enrichment
  if(nrow(up) != 0){
    ego_bp_up <- enrichGO(up$COMMON, OrgDb = org.Sc.sgd.db, keyType = "GENENAME", 
                          ont = "BP", pAdjustMethod = "bonferroni")
    ego_bp_up_top <- ego_bp_up@result %>%
      filter(p.adjust < 0.05) %>%
      dplyr::select(Description, geneID) %>%
      mutate(enrichment = "GO BP")
    
    ego_mf_up <- enrichGO(up$COMMON, OrgDb = org.Sc.sgd.db, keyType = "GENENAME", 
                          ont = "MF", pAdjustMethod = "bonferroni")
    ego_mf_up_top <- ego_mf_up@result%>%
      filter(p.adjust < 0.05) %>%
      dplyr::select(Description, geneID) %>%
      mutate(enrichment = "GO MF")
    
    ego_cc_up <- enrichGO(up$COMMON, OrgDb = org.Sc.sgd.db, keyType = "GENENAME", 
                          ont = "CC", pAdjustMethod = "bonferroni")
    ego_cc_up_top <- ego_cc_up@result %>%
      filter(p.adjust < 0.05) %>%
      dplyr::select(Description, geneID) %>%
      mutate(enrichment = "GO CC")
  } else {
    ego_cc_up_top <- ego_mf_up_top <- ego_bp_up_top <- data.frame("Description" = character(),
                                                                  "geneID" = character(),
                                                                  "enrichment" = character())
  }
  
  if(nrow(down) != 0){
    if(!identical(down$ORF, down$COMMON)) { # Filter out datasets where everything is unknown
      ego_bp_down <- enrichGO(down$COMMON, OrgDb = org.Sc.sgd.db, keyType = "GENENAME", 
                              ont = "BP", pAdjustMethod = "bonferroni")
      ego_bp_down_top <- ego_bp_down@result %>%
        filter(p.adjust < 0.05) %>%
        dplyr::select(Description, geneID) %>%
        mutate(enrichment = "GO BP")
      
      ego_mf_down <- enrichGO(down$COMMON, OrgDb = org.Sc.sgd.db, keyType = "GENENAME", 
                              ont = "MF", pAdjustMethod = "bonferroni")
      ego_mf_down_top <- ego_mf_down@result %>%
        filter(p.adjust < 0.05) %>%
        dplyr::select(Description, geneID) %>%
        mutate(enrichment = "GO MF")
      
      ego_cc_down <- enrichGO(down$COMMON, OrgDb = org.Sc.sgd.db, keyType = "GENENAME", 
                              ont = "CC", pAdjustMethod = "bonferroni")
      if(class(ego_cc_down) == "NULL") {
        message("No enriched GO CC Down, moving on!")
        # make empty df so obj is rbind compliant
        ego_cc_down_top <- data.frame("Description" = character(), 
                                      geneID = character(), 
                                      enrichment = character())
      } else {
        ego_cc_down_top <- ego_cc_down@result %>%
          filter(p.adjust < 0.05) %>%
          dplyr::select(Description, geneID) %>%
          mutate(enrichment = "GO CC")
      } 
    } else {
      ego_cc_down_top <- ego_mf_down_top <- ego_bp_down_top <- data.frame("Description" = character(),
                                                                          "geneID" = character(),
                                                                          "enrichment" = character())
    }
  }  else {
    ego_cc_down_top <- ego_mf_down_top <- ego_bp_down_top <- data.frame("Description" = character(),
                                                                        "geneID" = character(),
                                                                        "enrichment" = character())
  }
  
  # COMBINE ENRICHMENT TABLES
  if(show_genes == T){
    down_tbl <- rbind(ego_bp_down_top, ego_cc_down_top, ego_mf_down_top, ekegg_down_top) 
    up_tbl <- rbind(ego_bp_up_top, ego_cc_up_top, ego_mf_up_top, ekegg_up_top)
  } else if(show_genes == F) {
    down_tbl <- rbind(ego_bp_down_top, ego_cc_down_top, ego_mf_down_top, ekegg_down_top) %>%
      dplyr::select(-geneID, -enrichment, Pathway = Description)
    up_tbl <- rbind(ego_bp_up_top, ego_cc_up_top, ego_mf_up_top, ekegg_up_top) %>%
      dplyr::select(-geneID, -enrichment, Pathway = Description)
  }


  # PRUNE PATHWAY NAMES TO FIRST X WORDS
  up_tbl$Pathway <- gsub(" \\(.*\\)", "", up_tbl$Pathway)
  up_tbl$Pathway <-gsub(" SSU.* from tricistronic rRNA transcript", " rRNA", up_tbl$Pathway)
  down_tbl$Pathway <- gsub(" \\(.*\\)", "", down_tbl$Pathway)
  down_tbl$Pathway <-gsub(" SSU.* from tricistronic rRNA transcript", " rRNA", down_tbl$Pathway)
  
  # PLOT ANNOTATION TABLES
  if(nrow(up_tbl) == 0){
    # make an empty table if there are no results
    up_tbl <- ggtexttable(data.frame("No_sig_genes" = ""), rows = NULL, theme = ttheme(base_size = 5, base_style = "blank", padding = unit(c(.5, .5), "mm")))
  } else {
    up_tbl <- ggtexttable(up_tbl,rows = NULL, theme = ttheme(base_size = 5, base_style = "blank", padding = unit(c(.5, .5), "mm")))
  }
  
  if(nrow(down_tbl) == 0){
    # make an empty table if there are no results
    down_tbl <- ggtexttable(data.frame("No_sig_genes" = ""), rows = NULL, theme = ttheme(base_size = 5, base_style = "blank", padding = unit(c(.5, .5), "mm")))
  } else {
    down_tbl <- ggtexttable(down_tbl,rows = NULL, theme = ttheme(base_size = 5, base_style = "blank", padding = unit(c(.5, .5), "mm")))
  }
  
  # PLOT DC
  # prep DF for plotting. Rename DC* to DC so that the same innvocation can be used
  # for ggplot across all plots
  dc <- dc %>%
    dplyr::select(DC = all_of(dc_num), sample, time, deletion, category)
  
  plt_dc <- ggplot(dc, aes(x = DC, y = 0, color = time)) +
    geom_point(position = position_jitter(seed = 42)) +
    #scale_color_manual(values = brewer.pal(9, "Blues")[2:9]) +
    scale_color_manual(values = c("#CE2220", "#E67F33", "#D0B541", "#7EB875", "#57A2AC", "#4E78C4", "#824D99", "#B997C7")) +
    labs(x = dc_num) +
    theme_minimal() +
    theme(plot.title = element_text(hjust = 0.5),
          axis.text.y = element_blank(),
          axis.title.y = element_blank(),
          legend.position = "bottom") 


  # Plot chart and table into one object
  return(ggarrange(down_tbl, plt_dc, up_tbl, ncol = 3))
}

dc_top_diffex_plot_hyp <- function(dc, dc_num, dc_con){
  # dc <- dc19
  # dc_num <- "DC7"
  # dc_con <- dc7_19
  # color_by = "hr"
  # separate to induced and repressed -- note that filtering for 
  # log2fc is done in dc_contrast() when normalize_logFC = T. 
  # default is a normalizes logFC > abs(2) 
  # therefore, up down separates the positive and negative logFC values
  
  up_most <- dc_con %>%
    filter(logFC_adj > 0) %>%
    dplyr::select(COMMON, logFC_adj) %>%
    mutate(logFC = round(logFC_adj, digits = 1)) %>%
    dplyr::select(-logFC_adj) %>%
    arrange(desc(logFC)) %>%
    head(n = 10)
  
  down_most <- dc_con %>%
    filter(logFC_adj < 0) %>%
    dplyr::select(COMMON, logFC_adj) %>%
    mutate(logFC = round(logFC_adj, digits = 1)) %>% 
    dplyr::select(-logFC_adj) %>%
    head(n = 10)
  
  # COMBINE ENRICHMENT TABLES
  # make an annotation table
  
  if(nrow(up_most) == 0){
    # make an empty table if there are no results
    up_tbl <- ggtexttable(data.frame("Pathway" = ""), rows = NULL, theme = ttheme(base_size = 6, base_style = "classic", padding = unit(c(1, 1), "mm")))
  } else{
    up_tbl <- ggtexttable(up_most, rows = NULL, theme = ttheme(base_size = 6, base_style = "classic", padding = unit(c(1, 1), "mm")))
  }
  if(nrow(down_most) == 0){
    # make an empty table if there are no results
    down_tbl <- ggtexttable(data.frame("Pathway" = ""), rows = NULL, theme = ttheme(base_size = 6, base_style = "classic", padding = unit(c(1, 1), "mm")))
  } else {
    down_tbl <- ggtexttable(down_most,rows = NULL, theme = ttheme(base_size = 6, base_style = "classic", padding = unit(c(1, 1), "mm")))
  }
  # PLOT DC
  # prep DF for plotting. Rename DC* to DC so that the same innvocation can be used
  # for ggplot across all plots
  dc <- dc %>%
    dplyr::select(DC = all_of(dc_num), time, deletion, category, sample)
  
    plt_dc <- ggplot(dc, aes(x = DC, y = 0, color = time)) +
      geom_point(position = position_jitter(seed = 42)) +
      #scale_color_manual(values = brewer.pal(9, "Blues")[2:9]) +
      scale_color_manual(values = c("#CE2220", "#E67F33", "#D0B541", "#7EB875", "#57A2AC", "#4E78C4", "#824D99", "#B997C7")) +
      labs(x = dc_num) +
      theme_minimal() +
      theme(plot.title = element_text(hjust = 0.5),
            axis.text.y = element_blank(),
            axis.title.y = element_blank(),
            legend.position = "bottom") 

  # Plot chart and table into one object
  return(ggarrange(down_tbl, plt_dc, up_tbl, ncol = 3))
}
```

```{r preprocess_hypoxia}
hypoxia_info <- read_csv("hypoxia/hypoxia_info.csv")
files <- list.files("hypoxia/GSE115171_RAW", pattern = ".txt.gz$", recursive = T, full.names = T)
files <- c(files, list.files("hypoxia/GSE85595_RAW", pattern = "tab.gz", recursive = T, full.names = T))
counts <- files %>%
  purrr::set_names() %>% 
  map_dfr(function(x) read_tsv(x, col_names = c("gene", "count")), .id = "source") %>%
  mutate(source = gsub("\\.tabular\\.txt\\.gz", "", source)) %>%
  mutate(source = gsub("\\.tab.gz", "", source)) %>%
  mutate(source = gsub(".*\\/", "", source)) %>%
  filter(grepl(pattern = "^Y", x = gene)) %>%
  pivot_wider(id_cols = gene, names_from = source, values_from = count) %>%
  as.data.frame() %>%
  column_to_rownames("gene")

m <- counts
m <- sweep(m, 2, colSums(m), `/`)
m <- m[grepl("^Y", rownames(m)), ]                    # subset to mRNA
m <- t(m)                                             # transpose counts
```

```{r diffusion_mapping_hypoxia}
## Build diffusion map
nm <- m %>% norm_mat()        # normalize
aff <- nm %>%
  get_euc(., n.threads = 1) %>% 
  threshold(., top_k = 20)      # make affinity matrix
Lij <- aff %>% get_laplac()     # compute laplacian
eig <- eigen(Lij)               # smallest keig vectors
evl <- eig$values %>%           # get eigenvalues
  Re() %>%
  round(., digits = 10)
evc <- eig$vectors %>%          # get eigenvectors
  Re() %>%
  round(., digits = 10)

# create objects containing diffusion components
for(d in 1:length(evl)){
  assign(paste('dim', d, sep = '_'),
         Re(evc[, rank(evl) == (d + 1)])
  )
}

k_eig <- 20                     # specify num of variables you want to keep
dat <- do.call(mapply, c(FUN = cbind, mget(paste0("dim_", 1:(k_eig))))) %>%
  t() %>%
  as.data.frame()               # merge in array

colnames(dat) <- paste(paste('DC', 1:(k_eig), sep = ''))
rownames(dat) <- rownames(Lij)  # add labels

dc_hyp <- dat %>%               # join with metadata
  rownames_to_column("sample") %>%
  mutate(sample = gsub("GSM......._", "", sample)) %>%
  left_join(hypoxia_info, by = "sample")
```

```{r, eval = F}
tmp <- data.frame(val = eig$values) %>%
  arrange(val) %>%
  mutate(num = 1:length(eig$values))

ggplotly(ggplot(tmp, aes(y = val, x = num)) + 
  geom_point() +
  theme_minimal() +
  ggtitle("k=20 hypoxia"))
```

```{r DC1hypoxiaWT, fig.height = 3, fig.width= 3.5}
# "#E8ECFB", "#521A13"
dc_hyp$time <- as.factor(dc_hyp$time)
wt_dc1 <- ggplot(data=subset(dc_hyp, deletion == "HAP1"), aes(x = DC1, y = 0, color = time)) +
  geom_jitter(size = 3) +
  #scale_color_manual(values = brewer.pal(9, "Blues")[2:9]) +
  #scale_color_viridis_d(option = "cividis") +
  #scale_color_brewer(palette = "PiYG") +
  scale_color_manual(values = c("#CE2220", "#E67F33", "#D0B541", "#7EB875", "#57A2AC", "#4E78C4", "#824D99", "#B997C7")) +
  theme_minimal() +
  theme(plot.title.position = "plot", 
        axis.text.y = element_blank(),
        axis.title.y = element_blank()) 

wt_dc6 <- ggplot(data=subset(dc_hyp, deletion == "HAP1"), aes(x = DC6, y = 0, color = time)) +
  geom_jitter(size = 3) +
  scale_color_manual(values = c("#CE2220", "#E67F33", "#D0B541", "#7EB875", "#57A2AC", "#4E78C4", "#824D99", "#B997C7")) +
  theme_minimal() +
  theme(plot.title.position = "plot", 
        axis.text.y = element_blank(),
        axis.title.y = element_blank()) 
ggarrange(wt_dc1, wt_dc6, legend = "bottom", common.legend = T,
          nrow = 2, labels = c("A", "B"))
```

```{r diffex_hypoxia}
d0 <- DGEList(counts)                          # initialize the DGE object
d <- calcNormFactors(d0)                       # calculate normalization factors
dc_hyp_mm <- model.matrix(~dc_hyp$DC1 + ~dc_hyp$DC2 + ~dc_hyp$DC3 + dc_hyp$DC4 + dc_hyp$DC5 + ~dc_hyp$DC6 + ~dc_hyp$DC7 + dc_hyp$DC8 + dc_hyp$DC9 + dc_hyp$DC10)
dc_hyp_y <- voom(d, dc_hyp_mm, plot = F)
dc_hyp_fit <- lmFit(dc_hyp_y, dc_hyp_mm)

dc1_hyp <- dc_contrast(dc_fit = dc_hyp_fit, coeff = 2)
dc2_hyp <- dc_contrast(dc_fit = dc_hyp_fit, coeff = 3)
dc3_hyp <- dc_contrast(dc_fit = dc_hyp_fit, coeff = 4)
dc4_hyp <- dc_contrast(dc_fit = dc_hyp_fit, coeff = 5)
dc5_hyp <- dc_contrast(dc_fit = dc_hyp_fit, coeff = 6)
dc6_hyp <- dc_contrast(dc_fit = dc_hyp_fit, coeff = 7)
dc7_hyp <- dc_contrast(dc_fit = dc_hyp_fit, coeff = 8)
dc8_hyp <- dc_contrast(dc_fit = dc_hyp_fit, coeff = 9)
dc9_hyp <- dc_contrast(dc_fit = dc_hyp_fit, coeff = 10)
dc10_hyp <- dc_contrast(dc_fit = dc_hyp_fit, coeff = 11)

dc1_hyp_all <- dc_contrast(dc_fit = dc_hyp_fit, coeff = 2, adj.p.filt = 1, filter_logFC = F)
dc2_hyp_all <- dc_contrast(dc_fit = dc_hyp_fit, coeff = 2, adj.p.filt = 1, filter_logFC = F)
```

```{r supplemental_table_hyp, eval = F}
library(xlsx)
write.xlsx(dc1_hyp, file="supplemental_tables/supplementary_table2.xlsx", sheetName="DC1_hypoxia", row.names=FALSE)
write.xlsx(dc6_hyp, file="supplemental_tables/supplementary_table2.xlsx", sheetName="DC6_hypoxia", append = T, row.names=FALSE)
```

```{r dc_hyp1_filt}
# with log2fc > 2
dc1_hyp_up <- dc1_hyp %>%
  filter(logFC_adj > 0)
dc1_hyp_down <- dc1_hyp %>%
  filter(logFC_adj < 0)

# all sig
dc1_hyp_all_up <- dc1_hyp_all %>%
  filter(logFC_adj > 0) %>%
  filter(adj.P.Val < 0.05)
dc1_hyp_all_down <- dc1_hyp_all %>%
  filter(logFC_adj < 0) %>%
  filter(adj.P.Val < 0.05)
```

```{r dc_hyp2_filt}
dc2_hyp_up <- dc2_hyp %>%
  filter(logFC_adj > 0)
dc2_hyp_down <- dc2_hyp %>%
  filter(logFC_adj < 0)

dc2_hyp_all_up <- dc2_hyp_all %>%
  filter(logFC_adj > 0) %>%
  filter(adj.P.Val < 0.05)
dc2_hyp_all_down <- dc2_hyp_all %>%
  filter(logFC_adj < 0) %>%
  filter(adj.P.Val < 0.05)
```

```{r oxygen_regulated_genes}
url <- "https://www.g3journal.org/highwire/filestream/473070/field_highwire_adjunct_files/15/TableS3.xlsx"
download.file(url = url, destfile = "hypoxia/inputs/tabs3_oxygen_regulated_genes.xlsx")
ox_reg <- readxl::read_excel(path = "hypoxia/inputs/tabs3_oxygen_regulated_genes.xlsx", 
                             col_names = c('ORF', 'COMMON', 'description', 'Becerra_2002',
                                           'Hickman_2007', 'Hickman_2011', 'Kwast_2002', 
                                           'Lai_2005', 'Lai_2006', 'terLinde_1999', 
                                           'terLinde_2002', 'hypoxic', 'aerobic', 'total'),
                             sheet = 1, skip = 1)

hypoxic1 <- ox_reg %>%
  group_by(hypoxic) %>%
  tally()

aerobic1 <- ox_reg %>%
  group_by(aerobic) %>%
  tally()
```

```{r test_hypoxia_intersect_microarray}
hypoxic_genes <- ox_reg %>%
  filter(hypoxic >= 6) %>%
  dplyr::select(ORF, COMMON)

hypoxic_genes_dc1_hyp <- hypoxic_genes[hypoxic_genes$ORF %in% dc1_hyp$ORF, ]

# try with log fc 1.5 instead of 2
dc1_hyp_p <- dc_contrast(dc_fit = dc_hyp_fit, coeff = 2, adj.p.filt = 1, filter_logFC = F) %>%
  filter(adj.P.Val < 0.05)
hypoxic_genes_dc1_hyp <- hypoxic_genes[hypoxic_genes$ORF %in% dc1_hyp_p$ORF, ]

```

```{r test_aerobic_intersect_microarray}
aerobic_genes <- ox_reg %>%
  filter(aerobic >= 6) %>%
  dplyr::select(ORF, COMMON)

aerobic_genes_dc1_hyp <- aerobic_genes[aerobic_genes$ORF %in% dc1_hyp$ORF, ]
aerobic_genes_dc2_hyp <- aerobic_genes[aerobic_genes$ORF %in% dc2_hyp$ORF, ]

# table(aerobic_genes_dc1_hyp$ORF %in% aerobic_genes_dc2_hyp$ORF)
# table(aerobic_genes$ORF %in% dc1_hyp$ORF)
# table(aerobic_genes$ORF %in% dc2_hyp$ORF)
```

```{r bendjilali, eval = F}
url <- 'https://www.g3journal.org/highwire/filestream/473070/field_highwire_adjunct_files/13/TableS1.xlsx'
download.file(url = url, destfile = "hypoxia/inputs/tabs1_rnaseq_res.xlsx")
rnaseq_oxreg <- readxl::read_excel(path = "hypoxia/inputs/tabs1_rnaseq_res.xlsx", sheet = 1)%>%
  filter(grepl("^Y", gene))

hypoxic <- rnaseq_oxreg %>%
  filter(oxygen.regulated == "hypoxic") 

aerobic <- rnaseq_oxreg %>%
  filter(oxygen.regulated == "aerobic")

oxreg_int <- data.frame(intersection = c("DC1 induced", "DC1 repressed"),
                        hypoxic= c(table(hypoxic$gene %in% dc1_hyp_all_up$ORF)[2],
                                   table(hypoxic$gene %in% dc1_hyp_all_down$ORF)[2]),
                        aerobic = c(table(aerobic$gene %in% dc1_hyp_all_up$ORF)[2],
                                    table(aerobic$gene %in% dc1_hyp_all_down$ORF)[2]))
kable(oxreg_int, "latex", booktabs = T, col.names = c("Intersection", "Hypoxic (519 predicted)", "Aerobic (291 predicted)"),
      caption = 'Consistent hypoxic metabolic remodelling detected by Bendjilali et al. 2017 and this study. Genes induced along diffusion component 1 are involved in the hypoxic reponse while those that are repressed are involved in aerobic metabolism.') %>%
  kable_styling(font_size = 9)
```

## Diffusion mapping detected metabolic remodeling throughout wine fermentation 

## Lower diffusion components captured varied progression through fermentation 

### Higher diffusion components identified vineyard site specific gene expression patterns

```{r DC2uvarum}
dc19_2hr <- read_csv("samples_ava.csv") %>%
  filter(!is.na(tank)) %>%
  filter(hr == 2) %>%
  dplyr::select(id, hr, year, tank, ava_id) %>%
  left_join(dc19)

## 2019
v19 <- read_tsv("v2019/outputs/counts_all_organisms/raw_counts.tsv") %>%
  filter(!gene %in% c("__no_feature", "__ambiguous", "__too_low_aQual", "__not_aligned", "__alignment_not_unique"))
colnames(v19) <- gsub("outputs\\/htseq_all_organisms\\/", "", colnames(v19))
colnames(v19) <- gsub("_readcounts\\.txt", "", colnames(v19))
v19 <- dplyr::select(v19, c(colnames(v19)[colnames(v19) %in% dc19_2hr$id], gene))
v19 <- v19[rowSums(v19[-which(names(v19) %in% "gene")]) > 0, ] # remove rows that sum to zero
v19 <- column_to_rownames(v19, "gene") # move gene to rownames

# normalize counts
v0_19 <- DGEList(v19)              # initialize the DGE object
v19 <-calcNormFactors(v0_19)       # calculate normalization factors
v_mm19 <- model.matrix(~dc19_2hr$DC1 + ~dc19_2hr$DC2 + ~dc19_2hr$DC3 + dc19_2hr$DC4 +
                         dc19_2hr$DC5 + dc19_2hr$DC6 + dc19_2hr$DC7 + ~dc19_2hr$DC8 +
                         dc19_2hr$DC9 + dc19_2hr$DC10)
v_y19 <- voom(v19, v_mm19, plot = F)
# v_y19$E contains normalized counts


v19 <- v_y19$E %>%
  as.data.frame() %>%
  rownames_to_column("gene") %>%
  mutate(organism = gsub("_.*", "", gene)) %>%
  mutate(organism = gsub("KLTH0.*", "KLTH0", organism)) %>%
  mutate(organism = gsub("PEGC.*", "PEGC", organism)) %>%
  mutate(organism = gsub("MWSF.*", "MWSF", organism)) %>%
  mutate(organism = gsub("^Y.*", "scer", organism)) %>%
  mutate(organism = gsub("sn.*", "scer", organism)) %>%
  mutate(organism = gsub("LPNK.*", "LPNK", organism)) %>%
  mutate(organism = gsub("^t.*", "scer", organism)) %>%
  mutate(organism = gsub("RPR1", "scer", organism)) %>%
  mutate(organism = gsub("SCR1", "scer", organism)) %>%
  mutate(organism = gsub("TLC1", "scer", organism)) %>%
  mutate(organism = gsub("LSR1", "scer", organism)) %>%
  mutate(organism = gsub("PWR1", "scer", organism)) %>%
  mutate(organism = gsub("NME1", "scer", organism)) %>%
  mutate(organism = gsub("ICR1", "scer", organism)) %>%
  mutate(organism = gsub("RUF23", "scer", organism)) %>%
  mutate(organism = gsub("RNA170", "scer", organism)) %>%
  mutate(organism = gsub("ZOD1", "scer", organism)) %>%
  mutate(organism = gsub("IRT1", "scer", organism)) %>%
  mutate(organism = gsub("RUF20", "scer", organism)) %>%
  mutate(organism = gsub("RUF21", "scer", organism)) %>%
  mutate(organism = gsub("RUF22", "scer", organism)) %>%
  mutate(organism = gsub("RUF5-2", "scer", organism)) %>%
  mutate(organism = gsub("SRG1", "scer", organism)) %>%
  mutate(organism = gsub("RME2", "scer", organism)) %>%
  mutate(organism = gsub("RME3", "scer", organism)) %>%
  mutate(organism = gsub("RDN5-1", "scer", organism)) %>%
  mutate(organism = gsub("RDN37-2", "scer", organism)) %>%
  mutate(organism = gsub("HRA1", "scer", organism)) %>%
  mutate(organism = gsub("KLTH0", "Lachancea_thermotolerans", organism)) %>%
  mutate(organism = gsub("PEGC", "Cladosporium_sp_SL_16", organism)) %>%
  mutate(organism = gsub("MWSF", "Starmerella_bacillaris", organism)) %>%
  mutate(organism = gsub("LPNK", "Metschnikowia_sp_AWRI3582", organism)) %>%
  mutate(organism = gsub("scer", "Saccharomyces_cerevisiae", organism)) %>%
  mutate(organism = gsub("BCIN", "Botrytis_cinerea", organism)) %>%
  mutate(organism = gsub("BcDW1", "Botrytis_cinerea", organism)) %>%
  mutate(organism = gsub("VIT", "Vitis_vinifera", organism)) %>%
  mutate(organism = gsub("CU098", "Rhizopus_stolonifer", organism)) %>%
  mutate(organism = gsub("metfru2", "Metschnikowia_fructicola", organism)) %>%
  mutate(organism = gsub("M438DRAFT", "Aureobasidium_pullulans", organism)) %>%
  mutate(organism = gsub("AWRI3578", "Hanseniaspora_opuntiae", organism)) %>%
  mutate(organism = gsub("C5L36", "Pichia_kudriavzevii", organism)) %>%
  mutate(organism = gsub("D499", "Hanseniaspora_uvarum", organism)) %>%
  mutate(organism = gsub("D6D18", "Aureobasidium_pullulans", organism))

dc19_w_v19 <- v19 %>%
  pivot_longer(cols = starts_with("2019"), names_to = "sample", values_to = "counts") %>%
  group_by(sample, organism) %>%
  summarize(total_counts = sum(counts)) %>%
  pivot_wider(id_cols = sample, names_from = organism, values_from = total_counts) %>%
  right_join(dc19, v19, by = "sample")

# linear model:
# summary(lm(DC5 ~ Hanseniaspora_uvarum, data = dc19_w_v19))


plt <- ggplot(dc19_w_v19, aes(x = DC5, y =  Hanseniaspora_uvarum, color = ava)) +
  geom_point(size = 2, alpha = .9) +
  labs(y = expression(italic("Hanseniaspora uvarum"))) +
  theme_minimal()+
  theme(legend.position = "bottom",
        axis.text.x = element_text(angle = 90, vjust = .5),
        legend.title = element_text(size=12, face ="bold"),
        strip.text.x = element_text(size = 12, face = "bold")) +
  scale_color_manual(values = c(SRH = srh, SMV = smv,
                                AS = as, SNC = snc,  CRN = crn,
                                RRV = rrv, AV = av, OR = or),
                     breaks = c('OR', 'AV', 'SNC', 'RRV', 'CRN', 'AS', 'SMV', 'SRH'),
                     name = "AVA")

# orgs to test:
# Aureobasidium_pullulans   Botrytis_cinerea
# Cladosporium_sp_SL_16     Hanseniaspora_opuntiae    Hanseniaspora_uvarum
# Lachancea_thermotolerans  Metschnikowia_fructicola  Metschnikowia_sp_AWRI3582
# Pichia_kudriavzevii       Rhizopus_stolonifer       Saccharomyces_cerevisiae
# ggplot(dc19_w_v19, aes(x = DC5, y = Cladosporium_sp_SL_16)) +
#   geom_point()
# ggplot(dc19_w_v19, aes(x = DC5, y = Hanseniaspora_opuntiae)) +
#   geom_point()
# ggplot(dc19_w_v19, aes(x = DC5, y = Lachancea_thermotolerans)) +
#   geom_point()
# ggplot(dc19_w_v19, aes(x = DC5, y = Saccharomyces_cerevisiae)) +
#   geom_point()
# ggplot(dc19_w_v19, aes(x = DC5, y =  Metschnikowia_fructicola)) +
#   geom_point()
# ggplot(dc19_w_v19, aes(x = DC5, y =  Pichia_kudriavzevii)) +
#   geom_point()
# ggplot(dc19_w_v19, aes(x = DC5, y = Rhizopus_stolonifer)) +
#   geom_point()

lmpullulans <- summary(lm(DC5 ~ Aureobasidium_pullulans, data = dc19_w_v19))
lmbcin <- summary(lm(DC5 ~ Botrytis_cinerea, data = dc19_w_v19))
lmclad <- summary(lm(DC5 ~ Cladosporium_sp_SL_16, data = dc19_w_v19))
lmopuntiae <- summary(lm(DC5 ~ Hanseniaspora_opuntiae, data = dc19_w_v19))
lmuvarum <- summary(lm(DC5 ~ Hanseniaspora_uvarum, data = dc19_w_v19))
lmthermo <- summary(lm(DC5 ~ Lachancea_thermotolerans, data = dc19_w_v19))
lmmetsch <- summary(lm(DC5 ~ Metschnikowia_fructicola, data = dc19_w_v19))
lmpich <- summary(lm(DC5 ~ Pichia_kudriavzevii, data = dc19_w_v19))
lmrhiz <- summary(lm(DC5 ~ Rhizopus_stolonifer, data = dc19_w_v19))

# compile results
lmres <- data.frame("organisms" = c("Aureobasidium pullulans", "Botrytis cinerea", "Cladosporium sp SL 16",
                                    "Hanseniaspora opuntiae", "Hanseniaspora uvarum", "Lachancea thermotolerans",
                                    "Metschnikowia fructicola", "Pichia kudriavzevii", "Rhizopus stolonifer"),
                    "R2" = c(lmpullulans$adj.r.squared, lmbcin$adj.r.squared, lmclad$adj.r.squared,
                             lmopuntiae$adj.r.squared, lmuvarum$adj.r.squared, lmthermo$adj.r.squared,
                             lmmetsch$adj.r.squared, lmpich$adj.r.squared, lmrhiz$adj.r.squared),
                    "p value" = round(c(lmpullulans$coefficients[2, 4], lmbcin$coefficients[2, 4],
                                        lmclad$coefficients[2, 4],  lmopuntiae$coefficients[2, 4],
                                        lmuvarum$coefficients[2, 4], lmthermo$coefficients[2, 4],
                                        lmmetsch$coefficients[2, 4], lmpich$coefficients[2, 4],
                                        lmrhiz$coefficients[2, 4]), digits = 3))

kable(lmres)
```

```{r DC6nitrogen, fig.width = 5}
nitrogen <- info %>% 
  filter(!is.na(tank)) %>%
  filter(year == 2019) %>%
  dplyr::select(ava_id, year, ava, initial_nh3, initial_nopa) %>%
  distinct() %>%
  pivot_longer(cols = starts_with("initial"), names_to = "variable", values_to = "values") %>%
  mutate(variable = gsub("initial_nh3", "initial NH3", variable)) %>%
  mutate(variable = gsub("initial_nopa", "initial NOPA", variable)) 

nitrogen$ava_id <- factor(nitrogen$ava_id, 
                          levels = c('OR1', 'OR2', 'AV1', 'AV2',  
                                     'SNC1', 'SNC2', 'RRV1', 'RRV2', 'RRV3',  
                                     'CRN1', 'AS1', 'AS2', 'SMV1', 
                                     'SMV2','SRH1'))
nitrogen <- ggplot(nitrogen, aes(x = ava_id, y = values, fill = ava)) +
  geom_col() +
  facet_wrap(~variable, ncol = 2) +
  labs(x = "vineyard site", y = "concentration") +
  theme_minimal() +
  theme(legend.position = "bottom",
        axis.text.x = element_text(angle = 90, vjust = .5),
        legend.title = element_text(size=12),
        strip.text.x = element_text(size = 12)) +
  scale_fill_manual(values = c(SRH = srh, SMV = smv,  
                                AS = as, SNC = snc,  CRN = crn, 
                                RRV = rrv, AV = av, OR = or),
                    breaks = c('OR', 'AV', 'SNC',  'RRV', 'CRN', 'AS', 'SMV', 'SRH'), 
                    name = "AVA")


# correlation -----------------
dc19_64 <- dc19 %>%
  filter(hr == 64)

# linear models
# summary(lm(DC6 ~ initial_nopa, dc19_64))
# summary(lm(DC6 ~ initial_nh3, dc19_64))

# correlation results
nh3_corr <- ggplot(dc19_64, aes(x = DC6, y = initial_nh3, color = ava)) + 
  geom_point() + 
  labs(y = "initial NH3") +
  theme_minimal()+
  theme(legend.position = "none") +
  scale_color_manual(values = c(SRH = srh, SMV = smv,  
                               AS = as, SNC = snc,  CRN = crn, 
                               RRV = rrv, AV = av, OR = or),
                    breaks = c('OR', 'AV', 'SNC', 'RRV', 'CRN', 'AS', 'SMV', 'SRH'))

nopa_corr <- ggplot(dc19_64, aes(x = DC6, y = initial_nopa, color = ava)) + 
  geom_point() +
  labs(y = "initial NOPA") +
  theme_minimal() +
  theme(legend.position = "none") +
  scale_color_manual(values = c(SRH = srh, SMV = smv,  
                               AS = as, SNC = snc,  CRN = crn, 
                               RRV = rrv, AV = av, OR = or),
                    breaks = c('OR', 'AV', 'SNC', 'RRV', 'CRN', 'AS', 'SMV', 'SRH'))

ggarrange(nitrogen, ggarrange(nh3_corr, nopa_corr, ncol = 2), nrow = 2, 
          common.legend = T, legend = "bottom")
```


\newpage
# Supplementary material {-}

\beginsupplement


```{r brix_upset17}
brix_list_17 <- list(dc1_17_induced = filter(dc1_17, logFC > 0)$ORF,
                     dc1_17_repressed = filter(dc1_17, logFC < 0)$ORF,
                     brix_17_induced = filter(brix_17, logFC > 0)$ORF,
                     brix_17_repressed = filter(brix_17, logFC < -0)$ORF
)
upset_brix_17 <- UpSetR::upset(fromList(brix_list_17), nsets = 8, nintersects = 200, 
                       order.by = "freq", keep.order = T)
```

```{r brix_upset19}
brix_list_19 <- list(dc1_19_induced = filter(dc1_19, logFC > 0)$ORF,
                     dc1_19_repressed = filter(dc1_19, logFC < 0)$ORF,
                     brix_19_induced = filter(brix_19, logFC > 0)$ORF,
                     brix_19_repressed = filter(brix_19, logFC < 0)$ORF
)
upset_brix_19 <- UpSetR::upset(fromList(brix_list_19), nsets = 8, nintersects = 200, 
                               order.by = "freq", keep.order = T)
```

```{r dc1_upset}
dc1_list <- list(dc1_17_induced = filter(dc1_17, logFC > 0)$ORF,
                 dc1_17_repressed = filter(dc1_17, logFC < 0)$ORF,
                 dc1_19_induced = filter(dc1_19, logFC > 0)$ORF,
                 dc1_19_repressed = filter(dc1_19, logFC < 0)$ORF)
upset_dc1 <- UpSetR::upset(fromList(dc1_list), nsets = 8, nintersects = 200, 
                   order.by = "freq", keep.order = T)
```

```{r brix_upset}
brix_list <- list(brix_17_induced = filter(brix_17, logFC > 0)$ORF,
                  brix_17_repressed = filter(brix_17, logFC < -0)$ORF,
                  brix_19_induced = filter(brix_19, logFC > 0)$ORF,
                  brix_19_repressed = filter(brix_19, logFC < 0)$ORF)
upset_brix <- UpSetR::upset(fromList(brix_list), nsets = 8, nintersects = 200, 
                   order.by = "freq", keep.order = T)
```

```{r upsetgrid, fig.height= 5, fig.cap="Upset plot depicting intersections between differentially expressed genes using diffusion component 1 or brix as continuous variable. **A.** Intersection of differentially expressed genes from the 2017 vintage. **B.** Intersection of genes from the 2019 vintage. **C.** Intersection of genes differentially expressed along diffusion component 1 from the 2017 and 2019 vintages. **D** Intersection of genes differentially expressed relative to change in brix from the 2017 and 2019 vintages."}

plot_grid(as.grob(upset_brix_17), as.grob(upset_brix_19), 
          as.grob(upset_dc1), as.grob(upset_brix),
          ncol = 2, labels = c("A", "B", "C", "D"))
```

<!-- TOP GENES -->

```{r annotatedplts19hr, fig.height = 6, fig.width = 6, eval = T, fig.cap="Top genes induced along diffusion components that separate stages of fermentation in the 2019 vintage. Colored by hour at which samples were taken."}
ggarrange(dc_top_diffex_plot(dc = dc19, dc_num = "DC2", dc_con = dc2_19, color_by = "hr"),
          dc_top_diffex_plot(dc = dc19, dc_num = "DC3", dc_con = dc3_19, color_by = "hr"),
          dc_top_diffex_plot(dc = dc19, dc_num = "DC4", dc_con = dc4_19, color_by = "hr"),
          nrow = 3, labels = c("A", "B", "C"))
```

```{r annotatedplts19ava, fig.height = 8, fig.width=6, eval = T, fig.cap="Top genes induced along diffusion components that separate fermentations within a stage of fermentation in the 2019 vintage. Colored by American Viticulture Area (AVA) of each sample."}
ggarrange(dc_top_diffex_plot(dc = dc19, dc_num = "DC5", dc_con = dc5_19, color_by = "ava"),
          dc_top_diffex_plot(dc = dc19, dc_num = "DC6", dc_con = dc6_19, color_by = "ava"),
          dc_top_diffex_plot(dc = dc19, dc_num = "DC7", dc_con = dc7_19, color_by = "ava"),
          dc_top_diffex_plot(dc = dc19, dc_num = "DC8", dc_con = dc8_19, color_by = "ava"),
          nrow = 4, labels = c("A", "B", "C", "D"))
```

<!-- ENRICHMENT -->


```{r DC2viz2019, fig.height = 6.5, fig.cap= "Gene set enrichment for differentially expressed genes along diffusion component 2 in the 2019 vintage. All enriched categories with p < 0.05 after Bonferroni correction are shown. Pathways on the right side of the figure are induced in samples with a high diffusion component value, while pathways on the left of the figure are induced in samples with a low diffusion component value."}
dc_top_enrich_plot(dc = dc19, dc_num = "DC2", dc_con = dc2_19, color_by = "hr", show_genes = F)
```

```{r DC3viz2019, fig.height = 7.5, fig.cap= "Gene set enrichment for differentially expressed genes along diffusion component 3 in the 2019 vintage. All enriched categories with p < 0.05 after Bonferroni correction are shown. Pathways on the right side of the figure are induced in samples with a high diffusion component value, while pathways on the left of the figure are induced in samples with a low diffusion component value."}
dc_top_enrich_plot(dc = dc19, dc_num = "DC3", dc_con = dc3_19, color_by = "hr", show_genes = F)
```

```{r DC4viz2019, fig.height = 3.2, fig.cap="Gene set enrichment for differentially expressed genes along diffusion component 4 in the 2019 vintage. All enriched categories with p < 0.05 after Bonferroni correction are shown. Pathways on the right side of the figure are induced in samples with a high diffusion component value, while pathways on the left of the figure are induced in samples with a low diffusion component value."}
dc_top_enrich_plot(dc = dc19, dc_num = "DC4", dc_con = dc4_19, color_by = "hr", show_genes = F)
```

```{r DC5viz2019, fig.height = 2, fig.cap="Gene set enrichment for differentially expressed genes along diffusion component 5 in the 2019 vintage. All enriched categories with p < 0.05 after Bonferroni correction are shown. Pathways on the right side of the figure are induced in samples with a high diffusion component value, while pathways on the left of the figure are induced in samples with a low diffusion component value."}
dc_top_enrich_plot(dc = dc19, dc_num = "DC5", dc_con = dc5_19, color_by = "ava", show_genes = F)
```

```{r DC8viz2019, fig.height = 2, fig.cap="Gene set enrichment for differentially expressed genes along diffusion component 5 in the 2019 vintage. All enriched categories with p < 0.05 after Bonferroni correction are shown. Pathways on the right side of the figure are induced in samples with a high diffusion component value, while pathways on the left of the figure are induced in samples with a low diffusion component value."}
dc_top_enrich_plot(dc = dc19, dc_num = "DC8", dc_con = dc8_19, color_by = "ava", show_genes = F)
```

```{r nonsacc, echo = F, warning = F, message = F}
nonsacc <- " organism | R2 | pvalue 
\\textit{Aureobasidium pullulans} | -0.03555 | 0.947
\\textit{Botrytis cinerea} | -0.03261 | 0.774
\\textit{Cladosporium sp SL 16} | -0.03341 | 0.804
\\textit{Hanseniaspora opuntiae} | -0.03524 | 0.911
\\textit{Hanseniaspora uvarum} | 0.490605 | < 0.001
\\textit{Lachancea thermotolerans} | -0.02741 | 0.638
\\textit{Metschnikowia fructicola} | 0.069637 | 0.086
\\textit{Pichia kudriavzevii} | -0.02819 | 0.654
\\textit{Rhizopus stolonifer} | -0.02439 | 0.582"
df <- read.delim(textConnection(nonsacc), header=T, sep="|", strip.white=TRUE, stringsAsFactors=FALSE)
row.names(df)<-NULL
kable(df, "latex", booktabs = T, col.names = c("organism", "R\\textsuperscript{2}", "p value"),
      caption = 'Correlation between total gene expression by non-Saccharomyces organisms and DC5. Only \\textit{Hanseniapsora uvarum} was significantly correlated with DC5.') %>%
  kable_styling(font_size = 9)
```

## Hypoxia

```{r DC1vizhyp, fig.height = 6, fig.cap="Gene set enrichment for differentially expressed genes along diffusion component 1 during onset of hypoxia. All enriched categories with p < 0.05 after Bonferroni correction are shown. Pathways on the right side of the figure are induced in samples with a high diffusion component value, while pathways on the left of the figure are induced in samples with a low diffusion component value."}
dc_top_enrich_plot_hyp(dc = dc_hyp, dc_num = "DC1", dc_con = dc1_hyp)
```