---
title: "Diffusion mapping reveals vineyard-site specific fermentation trajectories"
always_allow_html: yes
author: 
- Taylor Reiter
- Rachel Montpetit
- ...
- C. Titus Brown?
- Ron Runnebaum
- Ben Montpetit
date: "`r format(Sys.time(), '%d %B, %Y')`"
bibliography: ../2020-pn-tmp/bibliography.bib
graphics: yes
figsintext: yes
geometry: "left=1in,right=1in,top=1in,bottom=1in"
header-includes:
- \usepackage{verbatim}
- \newcommand{\comm}[1]{}
- \usepackage{float} \floatplacement{figure}{H}
- \newcommand{\beginsupplement}{\setcounter{table}{0}  \renewcommand{\thetable}{S\arabic{table}}
  \setcounter{figure}{0} \renewcommand{\thefigure}{S\arabic{figure}}}
- \usepackage[font={small,it}, labelfont={bf}]{caption}
keep_tex: yes
output:
  bookdown::pdf_document2:
    fig_caption: yes
    toc: false
---
  
```{r setup, include=FALSE, eval = T}
knitr::opts_chunk$set(echo = F, eval = T, message = F, warning = F, cache = T, 
                      include = T, fig.keep = 'high', fig.path = "Rmd_fig_out/",
                      dpi=400, fig.width = 6.5)
```

```{r libs, eval = T}
library(dplyr)
library(purrr)
library(tidyr)
library(tibble)
library(readr)
library(purrr)
library(stringr)
library(ggplot2)
library(ggthemes)
library(ggpubr)
library(ggrepel)
library(ggplotify)
library(ggcorrplot)
library(limma)
library(edgeR)
library(clusterProfiler)
library(org.Sc.sgd.db)
library(kableExtra)
library(gridExtra)
library(UpSetR)
library(cowplot)
library(knitr)
library(RColorBrewer)
set.seed(1)
```

```{r diffusion_mapping_functions}
## standardize columns func
norm.mat <- function(mat){
  ms <- apply(mat, 2, function(col) (col - mean(col)) / sd(col))
  ms
}

## compute euclid dist and standrdize to similarities
get_euc <- function(mat, n.threads = 1, alt = 'euclidean'){
  if(n.threads > 1){
    eu <- parDist(mat, method = alt, threads = n.threads) %>% as.matrix()
    ieu <- 1 / eu
    diag(ieu) <- 0
  }
  
  if(n.threads == 1){
    eu <- dist(mat, method = alt) %>% as.matrix()
    ieu <- 1 / eu
    diag(ieu) <- 0
  }

  ieu
}

## function to threshold the normalized distance mat
threshold <- function(mat, top_k = 10){
  thr <- mat
  tnr <- nrow(thr)
  ## set similarities outside of the top k to 0
  for(i in 1:tnr){
    ## rank entries in each row in reverse order (so largest value == rank 1), 
    ## and set entries that are outside of 1:k to 0
    thr[i, !rank(-thr[i, ], ties.method = 'random') %in% 1:top_k] <- 0
  }
  
  for(i in 1:tnr){
    for(j in 1:tnr){
      if(thr[i, j] == 0 & thr[j, i] != 0){
        thr[i, j] <- thr[j, i]
      }
    }
  }
  thr
}

## function to calculate norm laplacian
get_laplac <- function(mat){
  L <- -mat
  S <- rowSums(mat)
  nL <- L / S
  diag(nL) <- 1
  nL
}
```

```{r processing_functions}
orf_to_common <- function(keys){
  common <- AnnotationDbi::select(org.Sc.sgd.db,
                                  keys = keys,
                                  columns=c("COMMON"),
                                  keytype="ORF")
  return(common)
}

lm_eqn <- function(df, y, x){
    formula = as.formula(sprintf('%s ~ %s', y, x))
    m <- lm(formula, data=df);
    # formating the values into a summary string to print out
    # ~ give some space, but equal size and comma need to be quoted
    eq <- substitute(italic(target) == a + b %.% italic(input)*","~~italic(r)^2~"="~r2*","~~p~"="~italic(pvalue), 
         list(target = y,
              input = x,
              a = format(as.vector(coef(m)[1]), digits = 2), 
              b = format(as.vector(coef(m)[2]), digits = 2), 
             r2 = format(summary(m)$r.squared, digits = 3),
             # getting the pvalue is painful
             pvalue = format(summary(m)$coefficients[2,'Pr(>|t|)'], digits=1)
            )
          )
    as.character(as.expression(eq));                 
}

dc_contrast <- function(dc_fit, coeff, adj.p.filt = 0.01, filter_logFC = T){
  # Test diffusion component coefficient
  dc_con <- contrasts.fit(dc_fit, coefficients = coeff)   
  dc_b <- eBayes(dc_con)
  dc_top <- topTable(dc_b, sort.by = "logFC", number = Inf)
  dc_top <- dc_top[dc_top$adj.P.Val < adj.p.filt, ]
  
  # Translate ORF to Common
  dc_top$ORF <- rownames(dc_top)
  commons <- orf_to_common(dc_top$ORF)
  # subset to first occurrence of ORF
  commons <- commons[match(unique(commons$ORF), commons$ORF), ]
  commons <- commons[ , -2]
  commons$COMMON <- ifelse(is.na(commons$COMMON), commons$ORF, commons$COMMON)
  commons <- unique(commons[ , ])
  dc_top <- left_join(dc_top, commons, by = c("ORF"))
  # normalize logfc based on min and max of DC
  normalizer <- max(dc_fit$design[ , coeff]) - min(dc_fit$design[ , coeff])
  dc_top <- dc_top %>%
    mutate(logFC_adj = logFC*normalizer)
  if(filter_logFC == F) {
    return(dc_top)
  } else {
    # Join Common to diffex results
    dc_top <- dc_top %>%
      filter(abs(logFC_adj) > 2)
    return(dc_top)
  }
}

dc_viz <- function(dc_con){
  # separate to induced and repressed
  up <- dc_con[dc_con$logFC > 3, ]
  down <- dc_con[dc_con$logFC < -3, ] 
  
  # KEGG enrichment
  ekegg_up <- enrichKEGG(up$ORF, "sce", pAdjustMethod = "bonferroni") 
  plt_rows <- ekegg_up@result %>%
    filter(p.adjust < .05) %>%
    nrow()
  plt1 <- dotplot(ekegg_up, showCategory = plt_rows, font.size = 5) + theme_minimal(base_size = 5)
  
  ekegg_down <- enrichKEGG(down$ORF, "sce", pAdjustMethod = "bonferroni")
  plt_rows <- ekegg_down@result %>%
    filter(p.adjust < .05) %>%
    nrow()
  plt2 <- dotplot(ekegg_down, showCategory = plt_rows, font.size = 5) + theme_minimal(base_size = 5)
  
  # GO enrichment
  ego_up <- enrichGO(up$COMMON, OrgDb = org.Sc.sgd.db, keyType = "GENENAME", 
                     ont = "ALL", pAdjustMethod = "bonferroni")
  plt_rows <- ego_up@result %>%
    filter(p.adjust < .05) %>%
    nrow()
  plt3 <- dotplot(ego_up, showCategory = plt_rows, font.size = 5) + theme_minimal(base_size = 5)
  
  ego_down <- enrichGO(down$COMMON, OrgDb = org.Sc.sgd.db, keyType = "GENENAME", 
                       ont = "ALL", pAdjustMethod = "bonferroni")
  plt_rows <- ego_down@result %>%
    filter(p.adjust < .05) %>%
    nrow()
  plt4 <- dotplot(ego_down, showCategory = plt_rows, font.size = 5) + theme_minimal(base_size = 5)
  
  # combine plots
  ggarrange(plt1, plt2, plt3, plt4, ncol = 2, nrow = 2, common.legend = T, 
            legend = "bottom", labels = c("A", "B", "C", "D"),
            heights = c(1, 2))
}

# this plot only makes sense when there are > ~10 genes per component that are
# differentially expressed
dc_top_enrich_plot <- function(dc, dc_num, dc_con, color_by = c("brix", "ava", "hr"), show_genes = F){
  # dc <- dc17
  # dc_num <- "DC1"
  # dc_con <- dc1_17
  # separate to induced and repressed -- note that filtering for 
  # log2fc is done in dc_contrast() when normalize_logFC = T. 
  # default is a normalizes logFC > abs(2) 
  # therefore, up down separates the positive and negative logFC values
  
  up <- dc_con[dc_con$logFC > 0, ]
  down <- dc_con[dc_con$logFC < 0, ] 
               
  # KEGG enrichment
  ekegg_up <- enrichKEGG(up$ORF, "sce", pAdjustMethod = "bonferroni") 
  ekegg_up_top <- ekegg_up@result %>%
    filter(p.adjust < 0.05) %>%
    dplyr::select(Description, geneID) %>%
    mutate(enrichment = "KEGG") 
 
  ekegg_down <- enrichKEGG(down$ORF, "sce", pAdjustMethod = "bonferroni")
  ekegg_down_top <- ekegg_down@result %>%
    filter(p.adjust < 0.05) %>%
    dplyr::select(Description, geneID) %>%
    mutate(enrichment = "KEGG")
  
  # GO enrichment
  ego_bp_up <- enrichGO(up$COMMON, OrgDb = org.Sc.sgd.db, keyType = "GENENAME", 
                        ont = "BP", pAdjustMethod = "bonferroni")
  ego_bp_up_top <- ego_bp_up@result %>%
    filter(p.adjust < 0.05) %>%
    dplyr::select(Description, geneID) %>%
    mutate(enrichment = "GO BP")
  
  ego_mf_up <- enrichGO(up$COMMON, OrgDb = org.Sc.sgd.db, keyType = "GENENAME", 
                        ont = "MF", pAdjustMethod = "bonferroni")
  ego_mf_up_top <- ego_mf_up@result%>%
    filter(p.adjust < 0.05) %>%
    dplyr::select(Description, geneID) %>%
    mutate(enrichment = "GO MF")
  
  ego_cc_up <- enrichGO(up$COMMON, OrgDb = org.Sc.sgd.db, keyType = "GENENAME", 
                        ont = "CC", pAdjustMethod = "bonferroni")
  ego_cc_up_top <- ego_cc_up@result %>%
    filter(p.adjust < 0.05) %>%
    dplyr::select(Description, geneID) %>%
    mutate(enrichment = "GO CC")
  
  ego_bp_down <- enrichGO(down$COMMON, OrgDb = org.Sc.sgd.db, keyType = "GENENAME", 
                          ont = "BP", pAdjustMethod = "bonferroni")
  ego_bp_down_top <- ego_bp_down@result %>%
    filter(p.adjust < 0.05) %>%
    dplyr::select(Description, geneID) %>%
    mutate(enrichment = "GO BP")
  
  ego_mf_down <- enrichGO(down$COMMON, OrgDb = org.Sc.sgd.db, keyType = "GENENAME", 
                          ont = "MF", pAdjustMethod = "bonferroni")
  ego_mf_down_top <- ego_mf_down@result %>%
    filter(p.adjust < 0.05) %>%
    dplyr::select(Description, geneID) %>%
    mutate(enrichment = "GO MF")
  
  ego_cc_down <- enrichGO(down$COMMON, OrgDb = org.Sc.sgd.db, keyType = "GENENAME", 
                          ont = "CC", pAdjustMethod = "bonferroni")
  if(class(ego_cc_down) == "NULL") {
    message("No enriched GO CC Down, moving on!")
    # make empty df so obj is rbind compliant
    ego_cc_down_top <- data.frame("Description" = character(), 
                                  geneID = character(), 
                                  enrichment = character())
  } else {
    ego_cc_down_top <- ego_cc_down@result %>%
      filter(p.adjust < 0.05) %>%
      dplyr::select(Description, geneID) %>%
      mutate(enrichment = "GO CC")
  }
  
  # COMBINE ENRICHMENT TABLES
  if(show_genes == T){
    down_tbl <- rbind(ego_bp_down_top, ego_cc_down_top, ego_mf_down_top, ekegg_down_top) 
    up_tbl <- rbind(ego_bp_up_top, ego_cc_up_top, ego_mf_up_top, ekegg_up_top)
  } else if(show_genes == F) {
    down_tbl <- rbind(ego_bp_down_top, ego_cc_down_top, ego_mf_down_top, ekegg_down_top) %>%
      dplyr::select(-geneID)
    up_tbl <- rbind(ego_bp_up_top, ego_cc_up_top, ego_mf_up_top, ekegg_up_top) %>%
      dplyr::select(-geneID)
  }


  # PRUNE PATHWAY NAMES TO FIRST X WORDS
  up_tbl$Description <- gsub(" \\(.*\\)", "", up_tbl$Description)
  up_tbl$Description <-gsub(" SSU.* from tricistronic rRNA transcript", " rRNA", up_tbl$Description)
  down_tbl$Description <- gsub(" \\(.*\\)", "", down_tbl$Description)
  down_tbl$Description <-gsub(" SSU.* from tricistronic rRNA transcript", " rRNA", down_tbl$Description)
  
  # PLOT ANNOTATION TABLES
  if(nrow(up_tbl) == 0){
    # make an empty table if there are no results
    up_tbl <- ggtexttable(data.frame("No_sig_genes" = ""), rows = NULL, theme = ttheme(base_size = 5, base_style = "blank", padding = unit(c(.5, .5), "mm")))
  } else {
    up_tbl <- ggtexttable(up_tbl,rows = NULL, theme = ttheme(base_size = 5, base_style = "blank", padding = unit(c(.5, .5), "mm")))
  }
  
  if(nrow(down_tbl) == 0){
    # make an empty table if there are no results
    down_tbl <- ggtexttable(data.frame("No_sig_genes" = ""), rows = NULL, theme = ttheme(base_size = 5, base_style = "blank", padding = unit(c(.5, .5), "mm")))
  } else {
    down_tbl <- ggtexttable(down_tbl,rows = NULL, theme = ttheme(base_size = 5, base_style = "blank", padding = unit(c(.5, .5), "mm")))
  }
  
  # PLOT DC
  # prep DF for plotting. Rename DC* to DC so that the same innvocation can be used
  # for ggplot across all plots
  dc <- dc %>%
    dplyr::select(DC = all_of(dc_num), brix, ava_id, ava, hr, sample)
  
  if(color_by == "brix"){
    plt_dc <- ggplot(dc, aes(x = DC, y = 0, color = brix, label = ava_id)) +
      geom_jitter() +
      scale_color_viridis_c() +      
      labs(x = dc_num) +
      theme_minimal() +
      theme(plot.title = element_text(hjust = 0.5),
            axis.text.y = element_blank(),
            axis.title.y = element_blank(),
            legend.position = "bottom") 
  } else if(color_by == "ava") {
    plt_dc <- ggplot(dc, aes(x = DC, y = 0, color = ava, label = ava_id)) +
      geom_jitter() +
      theme_minimal() +
      labs(x = dc_num) +
      theme(plot.title = element_text(hjust = 0.5),
            axis.text.y = element_blank(),
            axis.title.y = element_blank(),
            legend.position = "bottom") +
      scale_color_manual(values = c("OR"=or,  "AV" = av, "SNC" = snc, "RRV" = rrv, "CRN" = crn,
                                    "AS" = as, "SMV" = smv, "SRH" = srh)) 
  } else if(color_by == "hr"){
    plt_dc <- ggplot(dc, aes(x = DC, y = 0, color = factor(hr, levels = c("0", "2", "6", "16", "64", "112")), 
                             label = ava_id)) +
      geom_point() +
      scale_color_brewer(palette = "Paired", name = "hour") + 
      geom_jitter() +
      labs(x = dc_num) +
      theme_minimal() +
      theme(plot.title = element_text(hjust = 0.5),
            axis.text.y = element_blank(),
            axis.title.y = element_blank(),
            legend.position = "bottom")
    }

  # Plot chart and table into one object
  return(ggarrange(down_tbl, plt_dc, up_tbl, ncol = 3))
}

dc_top_diffex_plot <- function(dc, dc_num, dc_con, color_by = c("brix", "ava", "hr")){
  # dc <- dc19
  # dc_num <- "DC7"
  # dc_con <- dc7_19
  # color_by = "ava"
  # separate to induced and repressed -- note that filtering for 
  # log2fc is done in dc_contrast() when normalize_logFC = T. 
  # default is a normalizes logFC > abs(2) 
  # therefore, up down separates the positive and negative logFC values
  
  up_most <- dc_con %>%
    filter(logFC_adj > 0) %>%
    dplyr::select(COMMON, logFC_adj) %>%
    mutate(logFC = round(logFC_adj, digits = 1)) %>%
    dplyr::select(-logFC_adj) %>%
    head(n = 10)
  
  down_most <- dc_con %>%
    filter(logFC_adj < 0) %>%
    dplyr::select(COMMON, logFC_adj) %>%
    mutate(logFC = round(logFC_adj, digits = 1)) %>% 
    dplyr::select(-logFC_adj) %>%
    head(n = 10)
  
  # COMBINE ENRICHMENT TABLES
  # make an annotation table
  
  if(nrow(up_most) == 0){
    # make an empty table if there are no results
    up_tbl <- ggtexttable(data.frame("No_sig_genes" = ""), rows = NULL, theme = ttheme(base_size = 6, base_style = "classic", padding = unit(c(1, 1), "mm")))
  } else{
    up_tbl <- ggtexttable(up_most, rows = NULL, theme = ttheme(base_size = 6, base_style = "classic", padding = unit(c(1, 1), "mm")))
  }
  if(nrow(down_most) == 0){
    # make an empty table if there are no results
    down_tbl <- ggtexttable(data.frame("No_sig_genes" = ""), rows = NULL, theme = ttheme(base_size = 6, base_style = "classic", padding = unit(c(1, 1), "mm")))
  } else {
    down_tbl <- ggtexttable(down_most,rows = NULL, theme = ttheme(base_size = 6, base_style = "classic", padding = unit(c(1, 1), "mm")))
  }
  # PLOT DC
  # prep DF for plotting. Rename DC* to DC so that the same innvocation can be used
  # for ggplot across all plots
  dc <- dc %>%
    dplyr::select(DC = all_of(dc_num), brix, ava_id, ava, hr, sample)
  
  if(color_by == "brix"){
    plt_dc <- ggplot(dc, aes(x = DC, y = 0, color = brix, label = ava_id)) +
      geom_jitter() +
      scale_color_viridis_c() +
      labs(x = dc_num) +
      theme_minimal() +
      theme(plot.title = element_text(hjust = 0.5),
            axis.text.y = element_blank(),
            axis.title.y = element_blank(),
            legend.position = "bottom") 
  } else if(color_by == "ava") {
    plt_dc <- ggplot(dc, aes(x = DC, y = 0, color = ava, label = ava_id)) +
      geom_jitter() +
      theme_minimal() +
      labs(x = dc_num) +
      theme(plot.title = element_text(hjust = 0.5),
            axis.text.y = element_blank(),
            axis.title.y = element_blank(),
            legend.position = "bottom") +
      scale_color_manual(values = c("OR"=or,  "AV" = av, "SNC" = snc, "RRV" = rrv, "CRN" = crn,
                                    "AS" = as, "SMV" = smv, "SRH" = srh)) 
  } else if(color_by == "hr"){
    plt_dc <- ggplot(dc, aes(x = DC, y = 0, color = factor(hr, levels = c("0", "2", "6", "16", "64", "112")), 
                             label = ava_id)) +
      geom_point() +
      scale_color_brewer(palette = "Paired", name = "hour") + 
      geom_jitter() +
      labs(x = dc_num) +
      theme_minimal() +
      theme(plot.title = element_text(hjust = 0.5),
            axis.text.y = element_blank(),
            axis.title.y = element_blank(),
            legend.position = "bottom")
  } else {
      message("Please try a legitimate color values, either brix, ava, or hr")
    }

  # Plot chart and table into one object
  return(ggarrange(down_tbl, plt_dc, up_tbl, ncol = 3))
}

dc_top_table <- function(dc_con){
  up_most <- dc_con %>%
    filter(logFC > 0) %>%
    dplyr::select(ORF, COMMON, logFC_adj) %>%
    mutate(logFC = round(logFC_adj, digits = 1)) %>%
    dplyr::select(-logFC_adj) %>%
    head(n = 15)
  
  down_most <- dc_con %>%
    filter(logFC < 0) %>%
    dplyr::select(ORF, COMMON, logFC_adj) %>%
    mutate(logFC = round(logFC_adj, digits = 1)) %>%
    dplyr::select(-logFC_adj) %>%
    head(n = 15)
  
  most <- rbind(up_most, down_most)
  
  # add blank rows to enable printing when fewer than 30 diffex genes
  if(30-nrow(most) == 0){
    return(most)
  } else {
    blank_rows_to_add <- 30-nrow(most)
    blank_rows <- data.frame(ORF = rep("", blank_rows_to_add),
                             COMMON = rep("", blank_rows_to_add),
                             logFC = rep("", blank_rows_to_add))
    most <- rbind(most, blank_rows)
    return(most)
  }
}
```

```{r colors_avas}
#  ordered south to north
srh = "#FFAABB" # pink
smv = "#EE8866" # orange
as  = "#EEDD88" # light yellow
snc = "#99DDFF" # light cyan 
crn = "#AAAA00" # olive
rrv = "#BBCC33" # pear 
av  = "#77AADD" # light blue
or  = "#DDDDDD" # pale grey
```

# Introduction

<!-- This could be the introduction or it could be the first section of the results...it's going to depend on the target audience. -->

Gene expression is a dynamic process whereby a cell adjusts to its internal and external environment. 
While some biological processes occur in steady state, many are dynamic and require observation across time to understand the full complement of transcriptomic responses elicited by a stimuli.
In particular, environmental conditions that elicit a global shift in gene expression (e.g. onset of hypoxia) require time series sampling to capture the continuous metabolic remodelling that occurs.
Time series gene expression surveys have been successfully applied to these types of problems (CITE: first hypoxia paper, first wine paper). 
However, in these experiments, a single conditions was assayed to understand the ubiquitous global shifts in gene expression. 
Expanding such inquiries to multiple groups raises problems with data analysis procedures.

First, samples become asynchronous.
Time series profiles of cell populations rely on synchronicity between cells to identify coordinated shifts in gene expression.
However, when expanding scientific inquiry to groups of cell populations to determine differences between groups, synchronicity is difficult to maintain across groups.
Experimental techniques have been developed to resynchronize groups of cells, for example by arresting the cell cylce in the same phase across groups (CITE).
However these experimental approaches are not appropriate across broad experimental designs and may not be completely effective at re-synchronization (CITE). 

An additional challenge is the identification of differences between groups that are separate from or in excess of the dominant and ubiquitous global shift in gene expression.





Further, while coordinated shifts in gene expression may occur across groups, sometimes there are additional differences between groups that can be difficult to identify when they are masked by global shifts. 

These computational problems have been addressed in part in the single cell RNA-sequencing literature. 
When a population of cells undergoes a global shift in gene expression, concepts like pseudotime separate cells out along their trajectory of gene expression changes. 

<!-- These studies -- one of which captured the response of an external perturbation while the other captured the yeast cell cycle -- relied on synchronicity between the cells that were profiled to understand uniform transcriptional response performed by the cell population at hand.  -->
<!-- However, these approaches are not effective when comparing separate cell populations as response rates to stimuli may vary between populations leading to asynchronization in gene expression with respect to time (reveiwed in XXX Bar-Joseph 2012). -->
<!-- Various approaches for data alignment have been developed that principally rely on time-wrapping (CITE) or hidden Markov models (CITE). -->
<!-- DRAW BACK TO THESE METHODS AND WHY THEY WONT WORK. -->
<!-- clust, masigpro, metacycle, timesvector -->
<!-- Given that we aimed to identify the commonalities and differences in gene expression across fermentations, we sought an analysis approach that did not rely on synchronicity between samples. -->

In many RNA-sequencing experiments, capturing gene expression when there are few differences between groups allows researchers to identify the 

In a previous investigation, we used brix, a proxy for sugar concentration during fermentation, to perform continuous differential expression and identified the ubiquitous transcriptome response of *S. cerevisiae* to Pinot noir fermentation (CITE BRIX PAPER).
Given the success of this method, we sought an approach to extract latent variables with which to perform differential expression, and thereby identify and explain the sources of variation not attributable to the major metabolic transformation that occurs as sugar concentration decreases throughout fermentation.
Unlike brix measurements, latent variables are not observed but are inferred from relationships in the sequencing data.
Dimensionality reduction algorithms like principal component analysis are commonly applied to sequencing data to identify sources of variation therein, although many algorithms can be used for this purpose [@way2020compressing].

We extract these latent variables using diffusion mapping, a manifold learning technique that uses information from the *k* most similar samples to construct non-linear composites of the major sources of variation among samples  [@coifman2005geometric; @coifman2006diffusion].
Diffusion mapping is a common visualization technique in single cell RNA-sequencing, where the transcriptome of each cell represents a discrete point during cellular development or differentiation [@destiny2015].
We refer to these latent variables as *diffusion components*, and to scatter plots of diffusion components as *diffusion maps*.
Diffusion mapping extracts latent variables while differential expression provides interpretation for the drivers of separation among samples and fermentations.

Here, we explore the application of diffusion mapping to two time-series RNA-sequencing data sets that profile *S. cerevisiae* populations as they undergo global shifts in gene expression.
In the first, WINE
In the second, HYPOXIA
In this study, we aimed to understand how vineyard site impacts primary fermentation of Pinot noir wine.
We performed 60 inoculated primary fermentations of genetically similar Pinot noir grapes grown in 15 vineyards in California and Oregon (**Figure \@ref(fig:map)**). We profiled each fermentation using time-course RNA sequencing and used *Saccharomyces cerevisiae* gene expression as an indicator of similarities and difference across fermentation. We find...

```{r map, fig.cap = "A. Map of vineyard site American Viticulture Areas. B. Primary fermentation sampling time points for 2017 and 2019 vintages. Times are shown in hours and are relative to inoculation."}
include_graphics("figures/map_and_sampling.pdf") 
```

# Results

```{r metadata}
info <- read_csv("../samples_ava.csv")
info$ava_id <- factor(info$ava_id, levels = c('OR1', 'OR2', 'AV1', 'AV2', 'RRV1', 
                                              'RRV2', 'RRV3', 'SNC1', 'SNC2', 
                                              'CRN1', 'AS1', 'AS2', 'SMV1', 
                                              'SMV2','SRH1'))

info$ava <- factor(info$ava, levels = c('OR', 'AV', 'RRV', 'SNC', 'CRN', 
                                        'AS', 'SMV','SRH'))
```

```{r diffusion_mapping_17, include = F}
## preprocess counts
counts17 <- read_tsv("../v2017/outputs/counts/raw_counts.tsv") %>%
  filter(grepl(pattern = "^Y", x = gene)) %>%
  as.data.frame() %>%
  column_to_rownames("gene")
colnames(counts17) <- gsub("_readcounts.txt", "", colnames(counts17))
m17 <- sweep(counts17, 2, colSums(counts17), `/`)
m17 <- t(m17)

## build diffusion map
nm <- m17 %>% norm.mat()          # normalize
aff <- nm %>%
  get_euc(., n.threads = 1) %>% 
  threshold(., top_k = 10)      # make affinity matrix
Lij <- aff %>% get_laplac()     # compute laplacian
eig <- eigen(Lij)               # smallest keig vectors
evl <- eig$values %>%           # get eigenvalues
  Re() %>%
  round(., digits = 10)
evc <- eig$vectors %>%          # get eigenvectors
  Re() %>%
  round(., digits = 10)

# create objects containing diffusion components
for(d in 1:length(evl)){
  assign(paste('dim', d, sep = '_'),
         Re(evc[, rank(evl) == (d + 1)])
  )
}

k_eig <- 10                     # specify num of variables you want to keep
dat <- do.call(mapply, c(FUN = cbind, mget(paste0("dim_", 1:(k_eig))))) %>%
  t() %>%
  as.data.frame()               # merge in array

colnames(dat) <- paste(paste('DC', 1:(k_eig), sep = ''))
rownames(dat) <- rownames(Lij)  # add labels

dc17 <- dat %>%                  # join with metadata
  rownames_to_column("sample") %>%
  left_join(info, by = c("sample" = "id"))
```

```{r diffusion_mapping_19, include = F}
counts19 <- read_tsv("../v2019/outputs/counts/raw_counts.tsv") %>%
  dplyr::filter(grepl(pattern = "^Y", x = gene)) %>%
  dplyr::select(-`2019_inoculum_SRH1_readcounts.txt`,
                -`2019_inoculum_AS1_readcounts.txt`, 
                -`2019_inoculum_SMV2_readcounts.txt`,
                -`2019_inoculum_AS2_readcounts.txt`,
                -`2019_inoculum_RRV1_readcounts.txt`,
                -`2019_inoculum_SMV1_readcounts.txt`) %>%
  as.data.frame() %>%
  column_to_rownames("gene")
colnames(counts19) <- gsub("_readcounts.txt", "", colnames(counts19))

m19 <- sweep(counts19, 2, colSums(counts19), `/`)
m19 <- t(m19)                                             # transpose counts

## build diffusion map
nm <- m19 %>% norm.mat()        # normalize
aff <- nm %>%
  get_euc(., n.threads = 1) %>% 
  threshold(., top_k = 10)      # make affinity matrix
Lij <- aff %>% get_laplac()     # compute laplacian
eig <- eigen(Lij)               # smallest keig vectors
evl <- eig$values %>%           # get eigenvalues
  Re() %>%
  round(., digits = 10)
evc <- eig$vectors %>%          # get eigenvectors
  Re() %>%
  round(., digits = 10)

# create objects containing diffusion components
for(d in 1:length(evl)){
  assign(paste('dim', d, sep = '_'),
         Re(evc[, rank(evl) == (d + 1)])
  )
}

k_eig <- 10                     # specify num of variables you want to keep
dat <- do.call(mapply, c(FUN = cbind, mget(paste0("dim_", 1:(k_eig))))) %>%
  t() %>%
  as.data.frame()               # merge in array

colnames(dat) <- paste(paste('DC', 1:(k_eig), sep = ''))
rownames(dat) <- rownames(Lij)  # add labels

dc19 <- dat %>%                 # join with metadata
  rownames_to_column("sample") %>%
  left_join(info, by = c("sample" = "id"))
```

```{r diffex17}
d0_17 <- DGEList(counts17)                         # initialize the DGE object
d17 <- calcNormFactors(d0_17)
dc_mm17 <- model.matrix(~dc17$DC1 + ~dc17$DC2 + ~dc17$DC3 + dc17$DC4 + dc17$DC5 + 
                          dc17$DC6 + dc17$DC7 + dc17$DC8 + dc17$DC9 + dc17$DC10)
dc_y17 <- voom(d17, dc_mm17, plot = F)
dc_fit17 <- lmFit(dc_y17, dc_mm17)
dc1_17 <- dc_contrast(dc_fit = dc_fit17, coeff = 2) # Test "DC1" coefficient
dc2_17 <- dc_contrast(dc_fit = dc_fit17, coeff = 3) # Test "DC2" coefficient
dc3_17 <- dc_contrast(dc_fit = dc_fit17, coeff = 4) # Test "DC3" coefficient
dc4_17 <- dc_contrast(dc_fit = dc_fit17, coeff = 5) # Test "DC4" coefficient
dc5_17 <- dc_contrast(dc_fit = dc_fit17, coeff = 6) # Test "DC5" coefficient
dc6_17 <- dc_contrast(dc_fit = dc_fit17, coeff = 7) # Test "DC6" coefficient
dc7_17 <- dc_contrast(dc_fit = dc_fit17, coeff = 8) # Test "DC7" coefficient
dc8_17 <- dc_contrast(dc_fit = dc_fit17, coeff = 9) # Test "DC8" coefficient
dc9_17 <- dc_contrast(dc_fit = dc_fit17, coeff = 10) # Test "DC9" coefficient
dc10_17 <- dc_contrast(dc_fit = dc_fit17, coeff = 11) # Test "DC10" coefficient
```

```{r diffex19}
d0_19 <- DGEList(counts19)              # initialize the DGE object
d19 <-calcNormFactors(d0_19)            # calculate normalization factors
dc_mm19 <- model.matrix(~dc19$DC1 + ~dc19$DC2 + ~dc19$DC3 + dc19$DC4 + dc19$DC5 + 
                          dc19$DC6 + dc19$DC7 + ~dc19$DC8 + dc19$DC9 + dc19$DC10)
dc_y19 <- voom(d19, dc_mm19, plot = F)
dc_fit19 <- lmFit(dc_y19, dc_mm19)
dc1_19 <- dc_contrast(dc_fit = dc_fit19, coeff = 2) # Test "DC1" coefficient
dc2_19 <- dc_contrast(dc_fit = dc_fit19, coeff = 3) # Test "DC2" coefficient
dc3_19 <- dc_contrast(dc_fit = dc_fit19, coeff = 4) # Test "DC3" coefficient
dc4_19 <- dc_contrast(dc_fit = dc_fit19, coeff = 5) # Test "DC4" coefficient
dc5_19 <- dc_contrast(dc_fit = dc_fit19, coeff = 6) # Test "DC5" coefficient
dc6_19 <- dc_contrast(dc_fit = dc_fit19, coeff = 7) # Test "DC6" coefficient
dc7_19 <- dc_contrast(dc_fit = dc_fit19, coeff = 8) # Test "DC7" coefficient
dc8_19 <- dc_contrast(dc_fit = dc_fit19, coeff = 9) # Test "DC8" coefficient
dc9_19 <- dc_contrast(dc_fit = dc_fit19, coeff = 10) # Test "DC9" coefficient
dc10_19 <- dc_contrast(dc_fit = dc_fit19, coeff = 11) # Test "DC10" coefficient
```

## Diffusion mapping captures the dominant biological transition of primary fermentation

During fermentation, *S. cerevisiae* encounters overlapping stresses as acid and osmotic stress in early fermentation give way to nutrient limitation and high ethanol concentration.
As the concentration of sugar drops during fermentation, 
Diffusion component 1 captures the dominant structure between samples. 
In **Figure \@ref(fig:dc1)**, points are plotted along diffusion component 1, but ordered by brix. 
To test whether latent variables generated by diffusion mapping capture biologically-relevant structure of *S. cerevisiae* gene expression during wine fermentation, we performed differential expression using the continuous variable brix and the values of diffusion component 1. 
For both vintages, the majority of differentially expressed genes were shared by both methods (**Figure \@ref(fig:dc1)C, Figure \@ref(fig:upsetgrid)**). 
This indicates that diffusion mapping captures relationships between samples driven by biological differences. 
Further, the largest fraction of differentially expressed genes are shared between vintages using both methods.
This indicates that there is a conserved response by *S. cervisiae* across fermentations as brix declines.

```{r diffex_brix_17}
brix_counts17  <- counts17                    # non-wine samples already filtered
d0_17 <- DGEList(brix_counts17)               # initialize the DGE object
d17 <- calcNormFactors(d0_17)
brix_mm17 <- model.matrix(~dc17$brix)
brix_y17 <- voom(d17, brix_mm17, plot = F)
brix_fit17 <- lmFit(brix_y17, brix_mm17)
brix_17 <- dc_contrast(dc_fit = brix_fit17, coeff = 2) # Test brix coefficient
```

```{r diffex_brix_19, eval = T}
brix_counts19  <- counts19[ , 1:150]               # remove inoculum samples
d0_19 <- DGEList(brix_counts19)                    # initialize the DGE object
d19 <- calcNormFactors(d0_19)
brix19 <- info %>%
  dplyr::filter(year == 2019) %>%
  dplyr::filter(!is.na(brix))
brix_mm19 <- model.matrix(~brix19$brix)
brix_y19 <- voom(d19, brix_mm19, plot = F)
brix_fit19 <- lmFit(brix_y19, brix_mm19)
brix_19 <- dc_contrast(dc_fit = brix_fit19, coeff = 2) # Test brix coefficient
```

```{r brixcorr}
# make sure that everything has the same number of genes by not filtering by expression
# consider doing this from the outset (see above code) instead of just for this
# figure

brix_19_all <- dc_contrast(dc_fit = brix_fit19, coeff = 2, 
                           adj.p.filt = 1, filter_logFC = F) %>%
  arrange(ORF)
brix_17_all <- dc_contrast(dc_fit = brix_fit17, coeff = 2,
                           adj.p.filt = 1, filter_logFC = F) %>%
  arrange(ORF)
dc1_17_all <- dc_contrast(dc_fit = dc_fit17, coeff = 2, 
                          adj.p.filt = 1, filter_logFC = F) %>%
  arrange(ORF)
dc1_19_all <- dc_contrast(dc_fit = dc_fit19, coeff = 2,
                          adj.p.filt = 1, filter_logFC = F) %>%
  arrange(ORF)

# all.equal(brix_19_all$ORF, dc1_19_all$ORF, brix_17_all$ORF, dc1_17_all$ORF)

all_l2fc <- data.frame(brix19 = brix_19_all$logFC,
                       brix17 = brix_17_all$logFC,
                       DC1_17 = dc1_17_all$logFC,
                       DC1_19 = dc1_19_all$logFC) 
corr_l2fc <- cor(all_l2fc)
colnames(corr_l2fc) <- c("Brix 2019", "Brix 2017", "DC1 2017", "DC1 2019")
rownames(corr_l2fc) <- c("Brix 2019", "Brix 2017", "DC1 2017", "DC1 2019")


corr_plt <- ggcorrplot(corr_l2fc, type = "upper", show.diag = T, lab = T, 
                       hc.order = T, digits = 3, lab_size = 3, tl.cex = 9, 
                       lab_col = "white", show.legend = T,  legend.title =  "",
                       colors = brewer.pal(n = 3, name = "RdYlBu")) + 
  scale_fill_gradient2(low = brewer.pal(n = 3, name = "RdYlBu")[1], 
                       high = brewer.pal(n = 3, name = "RdYlBu")[3], 
                       mid = brewer.pal(n = 3, name = "RdYlBu")[2], 
                       midpoint = 0, limit = c(0, 1), space = "Lab", 
                       name = "") +
  theme(legend.position = "bottom")
```

```{r dc1, fig.height = 3.5, fig.cap = "Diffusion component 1 captures the metabolic transition that occurs as brix decreases during fermentation. The same transition occurs in **A.** the 2017 vintage and **B.** the 2019 vintage. Each point represents a sample from one time from one fermentation. Points that are closer along the x axis are more similar. The y axis is jittered to allow all points to be visualized. Points are colored by brix, where brix = 0 indicates end of fermentation. **C.** High concordance between differentially expressed genes in diffusion component 1 and genes that are differentially expressed as brix decreases."}
p17 <- ggplot(dc17, aes(x = DC1, y = 0, color = brix, label = ava_id)) +
  geom_jitter() +
  scale_color_viridis_c() +
  theme_minimal() +
  theme(plot.title = element_text(hjust = 0.5),
        axis.text.y = element_blank(),
        axis.title.y = element_blank()) +
  ggtitle("2017") 

p19 <- ggplot(dc19, aes(x = DC1, y = 0, color = brix, label = ava_id)) +
  geom_jitter() +
  scale_color_viridis_c() +
  theme_minimal() +
  theme(plot.title = element_text(hjust = 0.5),
        axis.text.y = element_blank(),
        axis.title.y = element_blank()) +
  ggtitle("2019") 

ggarrange(ggarrange(p17, p19, ncol = 1, nrow = 2, common.legend = T, legend = "bottom", labels = c("A", "B")), corr_plt, labels = c("", "C"))
```

Decreasing brix during fermentation accounts for the largest variation across samples in both vintages. 
In both vintages, processes involved in ribosome biogenesis and biosynthesis of amino acids are enriched in induced genes when brix is high, while authophagy and endocytosis is enriched in induced genes when brix is low (**Figure \@ref(fig:DC1viz2017)**, **Figure \@ref(fig:DC1viz2019)**).
Of the top genes induced when brix low, we see genes like *HXT6*, which encodes a high-affinity hexose transporter that is required at the end of alchoholic fermentation (CITATION). *HXT6*, has a high affinity for fructose, which is the only sugar left at this stage of fermentation (CITATION). 
(discuss *RSB1*, *CYC7*, *DAL80*).

## Other diffusion components also capture structure

Given that the first diffusion component captures the dominant biological transition that *S. cerevisiae* undergoes during primary fermentation, we can examine subsequent diffusion components to understand non-dominant transitions that occur, particularly those that are different between vineyard sites.

**Figure \@ref(fig:allDCs2017)** and **Figure \@ref(fig:allDCs2019)** depict one dimensional plots of the first ten diffusion components for each vintage, while **Table \@ref(tab:numdiffex)** summarizes the number of differentially expressed genes across each component. 
When colored by brix, the structure of the samples is sometimes apparent. 
(Note samples at the origin of a diffusion component are neutral toward structure for that component.)
For example, in the 2019 vintage, DC2, DC4, DC6, and DC7 capture differences in early fermenation, DC5 captures differences in late fermentation, and DC3 captures difference in the inoculum samples. 
In the 2017 vintage, DC2 captures variation in late-fermentation samples.
While there is generaly a diminishing number of differentially expressed genes in higher components, there is still a significant difference in each of these components (**Table \@ref(tab:numdiffex)**).

```{r all_dc17_brix}
dc1_17p <- ggplot(dc17, aes(x = DC1, y = 0, color = brix, label = ava_id)) +
  geom_jitter(alpha = .8) +
  xlim(c(-0.3, .3)) +
  scale_color_viridis_c() +
  theme_minimal() +
  theme(plot.title = element_text(hjust = 0.5),
        axis.text.y = element_blank(),
        axis.title.y = element_blank()) 

dc2_17p <- ggplot(dc17, aes(x = DC2, y = 0, color = brix, label = ava_id)) +
  geom_jitter(alpha = .8) +
  scale_color_viridis_c() +
  xlim(c(-0.3, .3)) +
  theme_minimal() +
  theme(plot.title = element_text(hjust = 0.5),
        axis.text.y = element_blank(),
        axis.title.y = element_blank()) 

dc3_17p <- ggplot(dc17, aes(x = DC3, y = 0, color = brix, label = ava_id)) +
  geom_jitter(alpha = .8) +
  scale_color_viridis_c() +
  xlim(c(-0.3, .3)) +
  theme_minimal() +
  theme(plot.title = element_text(hjust = 0.5),
        axis.text.y = element_blank(),
        axis.title.y = element_blank())

dc4_17p <- ggplot(dc17, aes(x = DC4, y = 0, color = brix, label = ava_id)) +
  geom_jitter(alpha = .8) +
  scale_color_viridis_c() +
  xlim(c(-0.3, .3)) +
  theme_minimal() +
  theme(plot.title = element_text(hjust = 0.5),
        axis.text.y = element_blank(),
        axis.title.y = element_blank()) 

dc5_17p <-ggplot(dc17, aes(x = DC5, y = 0, color = brix, label = ava_id)) +
  geom_jitter(alpha = .8) +
  xlim(c(-0.3, .3)) +
  scale_color_viridis_c() +
  theme_minimal() +
  theme(plot.title = element_text(hjust = 0.5),
        axis.text.y = element_blank(),
        axis.title.y = element_blank()) 

dc6_17p <-ggplot(dc17, aes(x = DC6, y = 0, color = brix, label = ava_id)) +
  geom_jitter(alpha = .8) +
  xlim(c(-0.3, .3)) +
  scale_color_viridis_c() +
  theme_minimal() +
  theme(plot.title = element_text(hjust = 0.5),
        axis.text.y = element_blank(),
        axis.title.y = element_blank())

dc7_17p <-ggplot(dc17, aes(x = DC7, y = 0, color = brix, label = ava_id)) +
  geom_jitter(alpha = .8) +
  xlim(c(-0.3, .3)) +
  scale_color_viridis_c() +
  theme_minimal() +
  theme(plot.title = element_text(hjust = 0.5),
        axis.text.y = element_blank(),
        axis.title.y = element_blank()) 

dc8_17p <-ggplot(dc17, aes(x = DC8, y = 0, color = brix, label = ava_id)) +
  geom_jitter(alpha = .8) +
  xlim(c(-0.3, .3)) +
  scale_color_viridis_c() +
  theme_minimal() +
  theme(plot.title = element_text(hjust = 0.5),
        axis.text.y = element_blank(),
        axis.title.y = element_blank()) 

dc9_17p <-ggplot(dc17, aes(x = DC9, y = 0, color = brix, label = ava_id)) +
  geom_jitter(alpha = .8) +
  xlim(c(-0.3, .3)) +
  scale_color_viridis_c() +
  theme_minimal() +
  theme(plot.title = element_text(hjust = 0.5),
        axis.text.y = element_blank(),
        axis.title.y = element_blank()) 

dc10_17p <-ggplot(dc17, aes(x = DC10, y = 0, color = brix, label = ava_id)) +
  geom_jitter(alpha = .8) +
  xlim(c(-0.3, .3)) +
  scale_color_viridis_c() +
  theme_minimal() +
  theme(plot.title = element_text(hjust = 0.5),
        axis.text.y = element_blank(),
        axis.title.y = element_blank()) 

all_dc_17 <- ggarrange(dc1_17p, dc2_17p, dc3_17p, dc4_17p, dc5_17p, dc6_17p, 
                       dc7_17p, dc8_17p, dc9_17p, dc10_17p, legend = "bottom",
                       ncol = 1, common.legend = T, labels = c("B", "", "", "", "", "", "", "", "", ""))
```

```{r all_dc19_brix}
dc1_19p <- ggplot(dc19, aes(x = DC1, y = 0, color = brix, label = ava_id)) +
  geom_jitter(alpha = .8) +
  xlim(c(-0.51, .51)) +
  scale_color_viridis_c() +
  theme_minimal() +
  theme(plot.title = element_text(hjust = 0.5),
        axis.text.y = element_blank(),
        axis.title.y = element_blank()) 

dc2_19p <- ggplot(dc19, aes(x = DC2, y = 0, color = brix, label = ava_id)) +
  geom_jitter(alpha = .8) +
  scale_color_viridis_c() +
  xlim(c(-0.51, .51)) +
  theme_minimal() +
  theme(plot.title = element_text(hjust = 0.5),
        axis.text.y = element_blank(),
        axis.title.y = element_blank()) 

dc3_19p <- ggplot(dc19, aes(x = DC3, y = 0, color = brix, label = ava_id)) +
  geom_jitter(alpha = .8) +
  scale_color_viridis_c() +
  xlim(c(-0.51, .51)) +
  theme_minimal() +
  theme(plot.title = element_text(hjust = 0.5),
        axis.text.y = element_blank(),
        axis.title.y = element_blank())

dc4_19p <- ggplot(dc19, aes(x = DC4, y = 0, color = brix, label = ava_id)) +
  geom_jitter(alpha = .8) +
  scale_color_viridis_c() +
  xlim(c(-0.51, .51)) +
  theme_minimal() +
  theme(plot.title = element_text(hjust = 0.5),
        axis.text.y = element_blank(),
        axis.title.y = element_blank()) 

dc5_19p <-ggplot(dc19, aes(x = DC5, y = 0, color = brix, label = ava_id)) +
  geom_jitter(alpha = .8) +
  xlim(c(-0.51, .51)) +
  scale_color_viridis_c() +
  theme_minimal() +
  theme(plot.title = element_text(hjust = 0.5),
        axis.text.y = element_blank(),
        axis.title.y = element_blank()) 

dc6_19p <-ggplot(dc19, aes(x = DC6, y = 0, color = brix, label = ava_id)) +
  geom_jitter(alpha = .8) +
  xlim(c(-0.51, .51)) +
  scale_color_viridis_c() +
  theme_minimal() +
  theme(plot.title = element_text(hjust = 0.5),
        axis.text.y = element_blank(),
        axis.title.y = element_blank())

dc7_19p <-ggplot(dc19, aes(x = DC7, y = 0, color = brix, label = ava_id)) +
  geom_jitter(alpha = .8) +
  xlim(c(-0.51, .51)) +
  scale_color_viridis_c() +
  theme_minimal() +
  theme(plot.title = element_text(hjust = 0.5),
        axis.text.y = element_blank(),
        axis.title.y = element_blank()) 

dc8_19p <-ggplot(dc19, aes(x = DC8, y = 0, color = brix, label = ava_id)) +
  geom_jitter(alpha = .8) +
  xlim(c(-0.51, .51)) +
  scale_color_viridis_c() +
  theme_minimal() +
  theme(plot.title = element_text(hjust = 0.5),
        axis.text.y = element_blank(),
        axis.title.y = element_blank()) 

dc9_19p <-ggplot(dc19, aes(x = DC9, y = 0, color = brix, label = ava_id)) +
  geom_jitter(alpha = .8) +
  xlim(c(-0.51, .51)) +
  scale_color_viridis_c() +
  theme_minimal() +
  theme(plot.title = element_text(hjust = 0.5),
        axis.text.y = element_blank(),
        axis.title.y = element_blank()) 

dc10_19p <-ggplot(dc19, aes(x = DC10, y = 0, color = brix, label = ava_id)) +
  geom_jitter(alpha = .8) +
  xlim(c(-0.51, .51)) +
  scale_color_viridis_c() +
  theme_minimal() +
  theme(plot.title = element_text(hjust = 0.5),
        axis.text.y = element_blank(),
        axis.title.y = element_blank()) 
all_dc_19 <- ggarrange(dc1_19p, dc2_19p, dc3_19p, dc4_19p, dc5_19p, dc6_19p,
                       dc7_19p, dc8_19p, dc9_19p, dc10_19p, ncol = 1, legend = "bottom",
                       common.legend = T, labels = c("B", "", "", "", "", "", "", "", "", ""))
```

```{r all_dc17_ava}
dc1_17p <- ggplot(dc17, aes(x = DC1, y = 0, color = ava, label = ava_id)) +
  geom_jitter(alpha = .8) +
  scale_color_manual(values = c("OR"=or,  "AV" = av, "SNC" = snc, "RRV" = rrv, "CRN" = crn,
                                "AS" = as, "SMV" = smv, "SRH" = srh)) +
  xlim(c(-0.3, .3)) +
  theme_minimal() +
  theme(plot.title = element_text(hjust = 0.5),
        axis.text.y = element_blank(),
        axis.title.y = element_blank()) 

dc2_17p <- ggplot(dc17, aes(x = DC2, y = 0, color = ava, label = ava_id)) +
  geom_jitter(alpha = .8) +
  scale_color_manual(values = c("OR"=or,  "AV" = av, "SNC" = snc, "RRV" = rrv, "CRN" = crn,
                                "AS" = as, "SMV" = smv, "SRH" = srh)) +
  xlim(c(-0.3, .3)) +
  theme_minimal() +
  theme(plot.title = element_text(hjust = 0.5),
        axis.text.y = element_blank(),
        axis.title.y = element_blank()) 

dc3_17p <- ggplot(dc17, aes(x = DC3, y = 0, color = ava, label = ava_id)) +
  geom_jitter(alpha = .8) +
  scale_color_manual(values = c("OR"=or,  "AV" = av, "SNC" = snc, "RRV" = rrv, "CRN" = crn,
                                "AS" = as, "SMV" = smv, "SRH" = srh)) +
  xlim(c(-0.3, .3)) +
  theme_minimal() +
  theme(plot.title = element_text(hjust = 0.5),
        axis.text.y = element_blank(),
        axis.title.y = element_blank())

dc4_17p <- ggplot(dc17, aes(x = DC4, y = 0, color = ava, label = ava_id)) +
  geom_jitter(alpha = .8) +
  scale_color_manual(values = c("OR"=or,  "AV" = av, "SNC" = snc, "RRV" = rrv, "CRN" = crn,
                                "AS" = as, "SMV" = smv, "SRH" = srh)) +
  xlim(c(-0.3, .3)) +
  theme_minimal() +
  theme(plot.title = element_text(hjust = 0.5),
        axis.text.y = element_blank(),
        axis.title.y = element_blank()) 

dc5_17p <-ggplot(dc17, aes(x = DC5, y = 0, color = ava, label = ava_id)) +
  geom_jitter(alpha = .8) +
  xlim(c(-0.3, .3)) +
  scale_color_manual(values = c("OR"=or,  "AV" = av, "SNC" = snc, "RRV" = rrv, "CRN" = crn,
                                "AS" = as, "SMV" = smv, "SRH" = srh)) +
  theme_minimal() +
  theme(plot.title = element_text(hjust = 0.5),
        axis.text.y = element_blank(),
        axis.title.y = element_blank()) 

dc6_17p <-ggplot(dc17, aes(x = DC6, y = 0, color = ava, label = ava_id)) +
  geom_jitter(alpha = .8) +
  xlim(c(-0.3, .3)) +
  scale_color_manual(values = c("OR"=or,  "AV" = av, "SNC" = snc, "RRV" = rrv, "CRN" = crn,
                                "AS" = as, "SMV" = smv, "SRH" = srh)) +
  theme_minimal() +
  theme(plot.title = element_text(hjust = 0.5),
        axis.text.y = element_blank(),
        axis.title.y = element_blank())

dc7_17p <-ggplot(dc17, aes(x = DC7, y = 0, color = ava, label = ava_id)) +
  geom_jitter(alpha = .8) +
  xlim(c(-0.3, .3)) +
  scale_color_manual(values = c("OR"=or,  "AV" = av, "SNC" = snc, "RRV" = rrv, "CRN" = crn,
                                "AS" = as, "SMV" = smv, "SRH" = srh)) +
  theme_minimal() +
  theme(plot.title = element_text(hjust = 0.5),
        axis.text.y = element_blank(),
        axis.title.y = element_blank()) 

dc8_17p <-ggplot(dc17, aes(x = DC8, y = 0, color = ava, label = ava_id)) +
  geom_jitter(alpha = .8) +
  xlim(c(-0.3, .3)) +
  scale_color_manual(values = c("OR"=or,  "AV" = av, "SNC" = snc, "RRV" = rrv, "CRN" = crn,
                                "AS" = as, "SMV" = smv, "SRH" = srh)) +
  theme_minimal() +
  theme(plot.title = element_text(hjust = 0.5),
        axis.text.y = element_blank(),
        axis.title.y = element_blank()) 

dc9_17p <-ggplot(dc17, aes(x = DC9, y = 0, color = ava, label = ava_id)) +
  geom_jitter(alpha = .8) +
  xlim(c(-0.3, .3)) +
  scale_color_manual(values = c("OR"=or,  "AV" = av, "SNC" = snc, "RRV" = rrv, "CRN" = crn,
                                "AS" = as, "SMV" = smv, "SRH" = srh)) +
  theme_minimal() +
  theme(plot.title = element_text(hjust = 0.5),
        axis.text.y = element_blank(),
        axis.title.y = element_blank()) 

dc10_17p <-ggplot(dc17, aes(x = DC10, y = 0, color = ava, label = ava_id)) +
  geom_jitter(alpha = .8) +
  xlim(c(-0.3, .3)) +
  scale_color_manual(values = c("OR"=or,  "AV" = av, "SNC" = snc, "RRV" = rrv, "CRN" = crn,
                                "AS" = as, "SMV" = smv, "SRH" = srh)) +
  theme_minimal() +
  theme(plot.title = element_text(hjust = 0.5),
        axis.text.y = element_blank(),
        axis.title.y = element_blank()) 

all_dc_17_ava <- ggarrange(dc1_17p, dc2_17p, dc3_17p, dc4_17p, dc5_17p, dc6_17p, 
                           dc7_17p, dc8_17p, dc9_17p, dc10_17p, legend = "bottom",
                           ncol = 1, common.legend = T, labels = c("A", "", "", "", "", "", "", "", "", ""))
```

```{r all_dc19_ava}
dc1_19p <- ggplot(dc19, aes(x = DC1, y = 0, color = ava, label = ava_id)) +
  geom_jitter(alpha = .8) +
  xlim(c(-0.51, .51)) +
  theme_minimal() +
  theme(plot.title = element_text(hjust = 0.5),
        axis.text.y = element_blank(),
        axis.title.y = element_blank()) +
  scale_color_manual(values = c("OR"=or,  "AV" = av, "SNC" = snc, "RRV" = rrv, "CRN" = crn,
          "AS" = as, "SMV" = smv, "SRH" = srh))

dc2_19p <- ggplot(dc19, aes(x = DC2, y = 0, color = ava, label = ava_id)) +
  geom_jitter(alpha = .8) +
  scale_color_manual(values = c("OR"=or,  "AV" = av, "SNC" = snc, "RRV" = rrv, "CRN" = crn,
                                "AS" = as, "SMV" = smv, "SRH" = srh)) +
  xlim(c(-0.51, .51)) +
  theme_minimal() +
  theme(plot.title = element_text(hjust = 0.5),
        axis.text.y = element_blank(),
        axis.title.y = element_blank()) 

dc3_19p <- ggplot(dc19, aes(x = DC3, y = 0, color = ava, label = ava_id)) +
  geom_jitter(alpha = .8) +
  scale_color_manual(values = c("OR"=or,  "AV" = av, "SNC" = snc, "RRV" = rrv, "CRN" = crn,
                                "AS" = as, "SMV" = smv, "SRH" = srh)) +
  xlim(c(-0.51, .51)) +
  theme_minimal() +
  theme(plot.title = element_text(hjust = 0.5),
        axis.text.y = element_blank(),
        axis.title.y = element_blank())

dc4_19p <- ggplot(dc19, aes(x = DC4, y = 0, color = ava, label = ava_id)) +
  geom_jitter(alpha = .8) +
  scale_color_manual(values = c("OR"=or,  "AV" = av, "SNC" = snc, "RRV" = rrv, "CRN" = crn,
                                "AS" = as, "SMV" = smv, "SRH" = srh)) +
  xlim(c(-0.51, .51)) +
  theme_minimal() +
  theme(plot.title = element_text(hjust = 0.5),
        axis.text.y = element_blank(),
        axis.title.y = element_blank()) 

dc5_19p <-ggplot(dc19, aes(x = DC5, y = 0, color = ava, label = ava_id)) +
  geom_jitter(alpha = .8) +
  xlim(c(-0.51, .51)) +
  scale_color_manual(values = c("OR"=or,  "AV" = av, "SNC" = snc, "RRV" = rrv, "CRN" = crn,
                                "AS" = as, "SMV" = smv, "SRH" = srh)) +
  theme_minimal() +
  theme(plot.title = element_text(hjust = 0.5),
        axis.text.y = element_blank(),
        axis.title.y = element_blank()) 

dc6_19p <-ggplot(dc19, aes(x = DC6, y = 0, color = ava, label = ava_id)) +
  geom_jitter(alpha = .8) +
  xlim(c(-0.51, .51)) +
  scale_color_manual(values = c("OR"=or,  "AV" = av, "SNC" = snc, "RRV" = rrv, "CRN" = crn,
                                "AS" = as, "SMV" = smv, "SRH" = srh)) +
  theme_minimal() +
  theme(plot.title = element_text(hjust = 0.5),
        axis.text.y = element_blank(),
        axis.title.y = element_blank())

dc7_19p <-ggplot(dc19, aes(x = DC7, y = 0, color = ava, label = ava_id)) +
  geom_jitter(alpha = .8) +
  xlim(c(-0.51, .51)) +
  scale_color_manual(values = c("OR"=or,  "AV" = av, "SNC" = snc, "RRV" = rrv, "CRN" = crn,
                                "AS" = as, "SMV" = smv, "SRH" = srh)) +
  theme_minimal() +
  theme(plot.title = element_text(hjust = 0.5),
        axis.text.y = element_blank(),
        axis.title.y = element_blank()) 

dc8_19p <-ggplot(dc19, aes(x = DC8, y = 0, color = ava, label = ava_id)) +
  geom_jitter(alpha = .8) +
  xlim(c(-0.51, .51)) +
  scale_color_manual(values = c("OR"=or,  "AV" = av, "SNC" = snc, "RRV" = rrv, "CRN" = crn,
                                "AS" = as, "SMV" = smv, "SRH" = srh)) +
  theme_minimal() +
  theme(plot.title = element_text(hjust = 0.5),
        axis.text.y = element_blank(),
        axis.title.y = element_blank()) 

dc9_19p <-ggplot(dc19, aes(x = DC9, y = 0, color = ava, label = ava_id)) +
  geom_jitter(alpha = .8) +
  xlim(c(-0.51, .51)) +
  scale_color_manual(values = c("OR"=or,  "AV" = av, "SNC" = snc, "RRV" = rrv, "CRN" = crn,
                                "AS" = as, "SMV" = smv, "SRH" = srh)) +
  theme_minimal() +
  theme(plot.title = element_text(hjust = 0.5),
        axis.text.y = element_blank(),
        axis.title.y = element_blank()) 

dc10_19p <-ggplot(dc19, aes(x = DC10, y = 0, color = ava, label = ava_id)) +
  geom_jitter(alpha = .8) +
  xlim(c(-0.51, .51)) +
  scale_color_manual(values = c("OR"=or,  "AV" = av, "SNC" = snc, "RRV" = rrv, "CRN" = crn,
                                "AS" = as, "SMV" = smv, "SRH" = srh)) +
  theme_minimal() +
  theme(plot.title = element_text(hjust = 0.5),
        axis.text.y = element_blank(),
        axis.title.y = element_blank()) 
all_dc_19_ava <- ggarrange(dc1_19p, dc2_19p, dc3_19p, dc4_19p, dc5_19p, dc6_19p,
                           dc7_19p, dc8_19p, dc9_19p, dc10_19p, ncol = 1, legend = "bottom",
                           common.legend = T, labels = c("A", "", "", "", "", "", "", "", "", ""))
```

```{r allDCs2017, fig.height= 9, fig.cap = "One-dimensional plots of diffusion components 1-10 for the 2017 vintage colored by **A** AVA and **B** brix. Diffusion components capture different latent structure in the dataset. Each component represents a different relationship among samples."}
ggarrange(all_dc_17_ava, all_dc_17, legend = "bottom")
```

```{r allDCs2019, fig.height= 9, fig.cap = "One-dimensional plots of diffusion components 1-10 for the 2019 vintage colored by **A** AVA and **B** brix. Diffusion components capture different latent structure in the dataset. Each component represents a different relationship among samples."}
ggarrange(all_dc_19_ava, all_dc_19, legend = "bottom")
```

```{r numdiffex}
num_diffex <- data.frame(diffusion_component = c("DC1", "DC2", "DC3", "DC4", 
                                                 "DC5", "DC6", "DC7", "DC8",
                                                 "DC9", "DC10"),
                         induced = c(length(filter(dc1_17, logFC > 3)$ORF),
                                     length(filter(dc2_17, logFC > 3)$ORF),
                                     length(filter(dc3_17, logFC > 3)$ORF),
                                     length(filter(dc4_17, logFC > 3)$ORF),
                                     length(filter(dc5_17, logFC > 3)$ORF),
                                     length(filter(dc6_17, logFC > 3)$ORF),
                                     length(filter(dc7_17, logFC > 3)$ORF),
                                     length(filter(dc8_17, logFC > 3)$ORF),
                                     length(filter(dc9_17, logFC > 3)$ORF),
                                     length(filter(dc10_17, logFC > 3)$ORF)),
                         repressed = c(length(filter(dc1_17, logFC < -3)$ORF),
                                       length(filter(dc2_17, logFC < -3)$ORF),
                                       length(filter(dc3_17, logFC < -3)$ORF),
                                       length(filter(dc4_17, logFC < -3)$ORF),
                                       length(filter(dc5_17, logFC < -3)$ORF),
                                       length(filter(dc6_17, logFC < -3)$ORF),
                                       length(filter(dc7_17, logFC < -3)$ORF),
                                       length(filter(dc8_17, logFC < -3)$ORF),
                                       length(filter(dc9_17, logFC < -3)$ORF),
                                       length(filter(dc10_17, logFC < -3)$ORF)),
                         induced = c(length(filter(dc1_19, logFC > 3)$ORF),
                                     length(filter(dc2_19, logFC > 3)$ORF),
                                     length(filter(dc3_19, logFC > 3)$ORF),
                                     length(filter(dc4_19, logFC > 3)$ORF),
                                     length(filter(dc5_19, logFC > 3)$ORF),
                                     length(filter(dc6_19, logFC > 3)$ORF),
                                     length(filter(dc7_19, logFC > 3)$ORF),
                                     length(filter(dc8_19, logFC > 3)$ORF),
                                     length(filter(dc9_19, logFC > 3)$ORF),
                                     length(filter(dc10_19, logFC > 3)$ORF)),
                         repressed = c(length(filter(dc1_19, logFC < -3)$ORF),
                                       length(filter(dc2_19, logFC < -3)$ORF),
                                       length(filter(dc3_19, logFC < -3)$ORF),
                                       length(filter(dc4_19, logFC < -3)$ORF),
                                       length(filter(dc5_19, logFC < -3)$ORF),
                                       length(filter(dc6_19, logFC < -3)$ORF),
                                       length(filter(dc7_19, logFC < -3)$ORF),
                                       length(filter(dc8_19, logFC < -3)$ORF),
                                       length(filter(dc9_19, logFC < -3)$ORF),
                                       length(filter(dc10_19, logFC < -3)$ORF))
)

kable(num_diffex, "latex", booktabs = T, col.names = c("diffusion component", "induced", "repressed", "induced", "repressed"),
      caption = 'Number of significantly differentially expressed genes for the top diffusion components for the 2017 and 2019 vintages') %>%
  kable_styling(font_size = 9) %>%
  add_header_above(c(" " = 1, "2017" = 2, "2019" = 2), align = "l", escape = F)
```

```{r try_all_by_all_dc_corr, eval = F}

# brix, hr, initial_brix, initial_ph,
# initial_nopa, initial_nh3, initial_MA, lon, lat, total_gdd_c, total_ppt
dc1_17_all <- dc_contrast(dc_fit = dc_fit17, coeff = 2,
                          adj.p.filt = 1, filter_logFC = F) %>%
  arrange(ORF) %>%
  select(ORF, COMMON, dc1_17_lfc = logFC_adj)
dc2_17_all <- dc_contrast(dc_fit = dc_fit17, coeff = 3,
                          adj.p.filt = 1, filter_logFC = F)  %>%
  arrange(ORF) %>%
  select(ORF, COMMON, dc2_17_lfc = logFC_adj)
dc3_17_all <- dc_contrast(dc_fit = dc_fit17, coeff = 4,
                         adj.p.filt = 1, filter_logFC = F)  %>%
  arrange(ORF) %>%
  select(ORF, COMMON, dc3_17_lfc = logFC_adj)
dc4_17_all <- dc_contrast(dc_fit = dc_fit17, coeff = 5,
                          adj.p.filt = 1, filter_logFC = F)  %>%
  arrange(ORF) %>%
  select(ORF, COMMON, dc4_17_lfc = logFC_adj)
dc5_17_all <- dc_contrast(dc_fit = dc_fit17, coeff = 6,
                          adj.p.filt = 1, filter_logFC = F)  %>%
  arrange(ORF) %>%
  select(ORF, COMMON, dc5_17_lfc = logFC_adj)
dc6_17_all <- dc_contrast(dc_fit = dc_fit17, coeff = 7,
                          adj.p.filt = 1, filter_logFC = F) %>%
  arrange(ORF) %>%
  select(ORF, COMMON, dc6_17_lfc = logFC_adj)
dc7_17_all <- dc_contrast(dc_fit = dc_fit17, coeff = 8,
                          adj.p.filt = 1, filter_logFC = F)  %>%
  arrange(ORF) %>%
  select(ORF, COMMON, dc7_17_lfc = logFC_adj)
dc8_17_all <- dc_contrast(dc_fit = dc_fit17, coeff = 9,
                          adj.p.filt = 1, filter_logFC = F)  %>%
  arrange(ORF) %>%
  select(ORF, COMMON, dc8_17_lfc = logFC_adj)
dc9_17_all <- dc_contrast(dc_fit = dc_fit17, coeff = 10,
                          adj.p.filt = 1, filter_logFC = F)  %>%
  arrange(ORF) %>%
  select(ORF, COMMON, dc9_17_lfc = logFC_adj)
dc10_17_all <- dc_contrast(dc_fit = dc_fit17, coeff = 11,
                          adj.p.filt = 1, filter_logFC = F)  %>%
  arrange(ORF) %>%
  select(ORF, COMMON, dc10_17_lfc = logFC_adj)

## 2019
dc1_19_all <- dc_contrast(dc_fit = dc_fit19, coeff = 2,
                          adj.p.filt = 1, filter_logFC = F) %>%
  arrange(ORF) %>%
  select(ORF, COMMON, dc1_19_lfc = logFC_adj)
dc2_19_all <- dc_contrast(dc_fit = dc_fit19, coeff = 3,
                          adj.p.filt = 1, filter_logFC = F)  %>%
  arrange(ORF) %>%
  select(ORF, COMMON, dc2_19_lfc = logFC_adj)
dc3_19_all <- dc_contrast(dc_fit = dc_fit19, coeff = 4,
                         adj.p.filt = 1, filter_logFC = F)  %>%
  arrange(ORF) %>%
  select(ORF, COMMON, dc3_19_lfc = logFC_adj)
dc4_19_all <- dc_contrast(dc_fit = dc_fit19, coeff = 5,
                          adj.p.filt = 1, filter_logFC = F)  %>%
  arrange(ORF) %>%
  select(ORF, COMMON, dc4_19_lfc = logFC_adj)
dc5_19_all <- dc_contrast(dc_fit = dc_fit19, coeff = 6,
                          adj.p.filt = 1, filter_logFC = F)  %>%
  arrange(ORF) %>%
  select(ORF, COMMON, dc5_19_lfc = logFC_adj)
dc6_19_all <- dc_contrast(dc_fit = dc_fit19, coeff = 7,
                          adj.p.filt = 1, filter_logFC = F) %>%
  arrange(ORF) %>%
  select(ORF, COMMON, dc6_19_lfc = logFC_adj)
dc7_19_all <- dc_contrast(dc_fit = dc_fit19, coeff = 8,
                          adj.p.filt = 1, filter_logFC = F)  %>%
  arrange(ORF) %>%
  select(ORF, COMMON, dc7_19_lfc = logFC_adj)
dc8_19_all <- dc_contrast(dc_fit = dc_fit19, coeff = 9,
                          adj.p.filt = 1, filter_logFC = F)  %>%
  arrange(ORF) %>%
  select(ORF, COMMON, dc8_19_lfc = logFC_adj)
dc9_19_all <- dc_contrast(dc_fit = dc_fit19, coeff = 10,
                          adj.p.filt = 1, filter_logFC = F)  %>%
  arrange(ORF) %>%
  select(ORF, COMMON, dc9_19_lfc = logFC_adj)
dc10_19_all <- dc_contrast(dc_fit = dc_fit19, coeff = 11,
                          adj.p.filt = 1, filter_logFC = F)  %>%
  arrange(ORF) %>%
  select(ORF, COMMON, dc10_19_lfc = logFC_adj)

## COMBINE
dc_lfc_all <- list(dc1_17_all, dc2_17_all, dc3_17_all, dc4_17_all,
                   dc5_17_all, dc6_17_all, dc7_17_all, dc8_17_all,
                   dc9_17_all, dc10_17_all, dc1_19_all, dc2_19_all, 
                   dc3_19_all, dc4_19_all, dc5_19_all, dc6_19_all, 
                   dc7_19_all, dc8_19_all, dc9_19_all, dc10_19_all) %>% 
  purrr::reduce(left_join, by = c("ORF", "COMMON")) %>%
  select(-ORF) %>%
  column_to_rownames("COMMON")
corr_all <- cor(dc_lfc_all) 
is.na(corr_all) <- abs(corr_all) < 0.5
is.na(r) <- abs(r) < 0.6
corr_all
ggcorrplot(corr_all, type = "upper", show.diag = T, lab = T,
           digits = 2, lab_size = 3, tl.cex = 10, 
           lab_col = "white", show.legend = T,  legend.title =  "",
           colors = brewer.pal(n = 3, name = "RdYlBu")) + 
  theme(legend.position = "bottom") 



ggcorrplot(cor(dc_lfc_all), type = "upper", show.diag = F, lab = T,
           sig.level = 0.0000001, p.mat = cor_pmat(dc_lfc_all), insig = "blank", 
           hc.order = T, digits = 2, lab_size = 2, tl.cex = 9, 
           lab_col = "white", show.legend = T,  legend.title =  "",
           colors = brewer.pal(n = 3, name = "RdYlBu")) + 
  theme(legend.position = "bottom")
```



## Diffusion mapping differentiate vineyard-site specific differences in fermentation

In 2019, we altered the sampling scheme to capture early fermentation more, as adaptation to the wine environment may be different for different fermentations and show what type of stress the cells are under. 

Diffusion components 2 and 4 separate early fermentation samples (**Figure \@ref(fig:v2019dc2dc3)**). 
Sample stake at 2hr, 6hr, and 16hr form distinct clusters, where DC2 separates all 3 time points and DC4 separates the 6hr samples from teh other timepoints. 
Along DC2, genes *TIR1*, *TIR2*, *TIR3*, *DAN1*, and *TIR4* are the top 5 repressed genes, indicating gene expression for cell wall remodelling is highest at 16 hrs post inoculation (**Table \@ref(tab:kable19most)**).
Pathways like ribosome biogenesis, RNA transport, RNA degradation, and basal transcription factors are induced along DC2, while biosynthesis of amino acids, carbon metabolism, and steroid biosythesis are enriched in repressed genes (**Figure \@ref(fig:DC2viz2019)**), reflecting the stressful transition yeast make upon inoculation.

Along DC4, repressed genes are enriched for proteasome, yeast cell cycle, DNA replication, mismatch repair, terpenoid biosynthesis, and one carbon pool by folate. 
These pathways are more highly expressed in the 6hr samples. 
Samples from both *OR2* and *SRH1* are accelerated along the transition from 2hr to 6hr, and 6hr to 16hr. However, overall, there is high similarity among sites in DC2 and DC4. 
Diffusion component 2 separates 2 hour from 16 hours samples, indicating that this is the largest source of variation after metabolic remodeling that occurs other than decrease in brix. 


```{r v2019dc2dc3, fig.height = 3, fig.cap = "Diffusion component 2 and diffusion component 3 capture variation in early fermentation. While late-fermentation samples cluster at the origin, early fermentation samples for three clusters."}
by_hour <- ggplot(dc19, aes(x = DC1, y = DC2, color = factor(hr, levels = c("0", "2", "6", "16", "64", "112")), label = ava)) +
  geom_point(alpha = .8) +
  scale_color_brewer(palette = "Paired", name = "hour") + 
  theme_minimal() +
  theme(plot.title = element_text(hjust = 0.5)) 

by_ava <- ggplot(dc19, aes(x = DC2, y = DC3, color = ava, label = hr)) +
  geom_point(alpha = .8) +
  scale_color_manual(values = c("OR"=or,  "AV" = av, "SNC" = snc, "RRV" = rrv, "CRN" = crn,
                                "AS" = as, "SMV" = smv, "SRH" = srh)) + 
  theme_minimal() +
  theme(plot.title = element_text(hjust = 0.5)) 

by_hour_67 <- ggplot(dc19, aes(x = DC4, y = DC10, color = factor(hr, levels = c("0", "2", "6", "16", "64", "112")), label = ava)) +
  geom_point(alpha = .8) +
  scale_color_brewer(palette = "Paired", name = "hour") + 
  theme_minimal() +
  theme(plot.title = element_text(hjust = 0.5)) 

by_ava_67 <- ggplot(dc19, aes(x = DC4, y = DC10, color = ava, label = ava)) +
  geom_point() +
  theme_minimal() +
  scale_color_manual(values = c("OR"=or,  "AV" = av, "SNC" = snc, "RRV" = rrv, "CRN" = crn,
                                "AS" = as, "SMV" = smv, "SRH" = srh)) + 
  theme(plot.title = element_text(hjust = 0.5)) 

#ggarrange(by_hour_67, by_ava_67, ncol = 2, legend = "bottom", labels = c("A", "B"))

ggarrange(by_hour, by_ava, ncol = 2, legend = "bottom", labels = c("A", "B"))
```


Conversely, diffusion component 6 and 7 capture separation in early fermentation samples. Along DC6, samples from sites *SMV1*, *SMV2* (1), and to a lesser extent *RRV3* and *AV1* separate from other 2hr samples. 
Only one gene is induced (*PDC5*), and two genes are repressed (*ARG1* and *RPS14B*).
*PDC5* is a minor isoform of pyruvate decarboxylase, and is a key enzyme in alcoholic fermentation. 
It is also involved in amino acid catabolism.
*ARG1* is part of the arginine biosynthesis pathway, while *RPS14B* is required for ribosome assembly. 
This indicates that...

Along DC7, samples from the 2hr, 6hr, and 16hr time points separate. The *SRH1* 2hr and 6hr samples separate in one direction the most, followed by *OR2* 2hr and 6hr and *SMV2* 16hr. 
On the other side *CRN1* and *AS1* separate. 
Only GO term transmembrane transport is enriched in induced genes, and only intracellular mRNA localization is enriched in repressed genes (**Figure S14**).

Diffusion component 3 separates inoculum from all other samples (see **Figure 4B**). 
Inoculum samples are enriched for metabolic pathways like carbon metabolism, TCA cycle, pyruvate metabolism, and glyoxylate metabolism, among others (**Figure S10**). 
Conversely,  all wine samples are enriched for ribosome, biosynthesis of amino acids, glcolysis/gluconeogenesis, steroid biosynthesis, and terpenoid backbone (**Figure S10**). 

Late fermentation samples separate in diffusion component 5. 
Given that two components dominated by changes in early fermentation occur before this, this indicates that there is more variation in the early fermentation response than there is in late fermentation.
While samples largely separate by brix, with dry fermentations segregating to one side, samples from site *AS2* break this trend. 
Induced genes in dry samples are enriched for protein processing in the ER, and repressed genes are enriched for pathways like ribosomes, carbon metabolism, purine metabolism, and fatty acid metabolism (**Figure S12**)


```{r v2019dc8dc10, fig.height = 3, fig.cap = "Diffusion component 8 and diffusion component 10 capture variation in late fermentation. While early fermentation samples cluster at the origin, late fermentation samples, especially those from regions AS, AV, and SNC, form distinct clusters."}
by_hour <- ggplot(dc19, aes(x = DC8, y = DC10, color = factor(hr, levels = c("0", "2", "6", "16", "64", "112")), label = ava_id)) +
  geom_point() +
  scale_color_brewer(palette = "Paired", name = "hour") + 
  theme_minimal() +
  theme(plot.title = element_text(hjust = 0.5)) 

by_ava <- ggplot(dc19, aes(x = DC8, y = DC10, color = ava, label = ava_id)) +
  geom_point() +
  scale_color_manual(values = c("OR"=or,  "AV" = av, "SNC" = snc, "RRV" = rrv, "CRN" = crn,
                                  "AS" = as, "SMV" = smv, "SRH" = srh)) + 
  theme_minimal() +
  theme(plot.title = element_text(hjust = 0.5)) 

ggarrange(by_hour, by_ava, ncol = 2, legend = "bottom", labels = c("A", "B"))
```

## Diffusion mapping works on other datasets with "developmental" trajectories

OR

## Diffusion mapping reveals a coordinated response to hypoxia

```{r diffusion_mapping_hypoxia}
hypoxia_info <- read_csv("hypoxia/hypoxia_info.csv")
files <- list.files("hypoxia/GSE115171_RAW", pattern = ".txt.gz$", recursive = T, full.names = T)
files <- c(files, list.files("hypoxia/GSE85595_RAW", pattern = "tab.gz", recursive = T, full.names = T))
counts <- files %>%
  purrr::set_names() %>% 
  map_dfr(function(x) read_tsv(x, col_names = c("gene", "count")), .id = "source") %>%
  mutate(source = gsub("\\.tabular\\.txt\\.gz", "", source)) %>%
  mutate(source = gsub("\\.tab.gz", "", source)) %>%
  mutate(source = gsub(".*\\/", "", source)) %>%
  filter(grepl(pattern = "^Y", x = gene)) %>%
  pivot_wider(id_cols = gene, names_from = source, values_from = count) %>%
  as.data.frame() %>%
  column_to_rownames("gene")

m <- counts
m <- sweep(m, 2, colSums(m), `/`)
m <- m[grepl("^Y", rownames(m)), ]                    # subset to mRNA
m <- t(m)                                             # transpose counts

# build diffusion map
dm_hyp <- DiffusionMap(m, k = 41, n_eigs = 334)

# create a df with each DC that occurs after a large jump and record metadata
dc_hyp <- data.frame(DC1 = dm_hyp$DC1, DC2 = dm_hyp$DC2, DC3 = dm_hyp$DC3, 
                     DC4 = dm_hyp$DC4, DC5 = dm_hyp$DC5, DC7 = dm_hyp$DC7) %>%
  rownames_to_column("sample") %>%
  mutate(sample = gsub("GSM......._", "", sample)) %>%
  left_join(hypoxia_info, by = "sample")
```

The diffusion map constructed from the hypoxia gene expression study shows time-ordered relationships between samples.
DC1 captures gene expression at 30 min-post 100% nitrogen exposure, while DC2 captures a separate transition that occurs during long-term exposure to nitrogen (t > 120min). 
Given that the 30min samples differentiate in the first diffusion component, this indicates that the short-term response to hypoxia is stronger and different from the long-term response.

The majority of mutants follow the same trajectory within the diffusion map as wild type strain during time-course exposure to nitrogen. 
Many of the mutants in this study are transcription factor deletants (e.g. rox1, hap1, mga2, upc2, ecm22, mot3, dsc1, hap4, hap2, hap3, hap5, and combinations thereof). 
The conserved hypoxic response trajectory through the diffusion map supports redundant regulation of the hypoxic response for most deletants.

However, there are two groups of outliers in this response. 
First, the deletants  upc2, mg2, and ecm22 are differentiated along DC2 at early time points and do not experience a peak response at 30 min. 
This indicates that these samples have gene expression profiles that resemble late-stage hypoxic response throughout nitrogen exposure.
*UPC2* and *ECM22* both induce sterol biosynthetic genes upon sterol depletion, while *MGA2* regulates *OLE1* transcription, a gene that codes for a key protein in oleate biosynthesis.
Both pathways require oxygen.
The shift in these time course samples away from the 30 min peak and toward late-stage samples indicates that gene expression related to these transcription factors is important from 0-30 min in the hypoxic response.

The hap1 mutant, and to a lesser extent the hap2_hap3_hap4_hap5 mutant, also deviate from the wild type trajectory.
While hap1 resembles wild type at t = 0, it does not migrate along DC1 toward the 30 min hypoxic peak. 
After 30 min of nitrogen exposure, it is most similar to the upc2/mga2/ecm22 deletant cluster.
During 120-180 min, hap1 clusters with other long-exposure samples.
Then at 240 min hap1 is back with the time zero samples. 
The hap1 mutant was sequenced in biological duplicate and have high concordance between samples, indicating this is likely a real biological process.
*HAP1* regulates gene expression in response to heme and oxygen levels by derepressing *ROX1*, a repressor for a set of hypoxic genes [@zitomer1997regulation]. 
In the absence of *HAP1*, *ROX1* would not be derepressed and the hypoxic gene set would not be expressed in response to anaerobiosis. 
The hap1 deletion appears to be important in both peak hypoxic response and maintenance of long-term hypoxic gene expression. 

In contrast, the rox1 and mot3 deletants maintain a gene expression profile shifted along DC1 toward the peak hypoxic response through nitrogen exposure. 
Further, the rox1, mot3, and rox1_mot3 deletants have the most exaggerated hypoxic peak on DC1. 
*ROX1* represses a set of hypoxic genes, while *MOT3* leads to a larger 30 min hypoxic peak than *ROX1* or *MOT3* deletion alone, suggesting *ROX1* and *MOT3* perform some individual repression of the hypoxic gene set.
However, it is possible that peak 30min hypoxic response was not observed in all samples. 


```{r DMhypoxia, fig.cap="Diffusion map of wildtype and 22 mutant strains exposed to 100% nitrogen for 0-240 min. Wildtype cells are crosses, while all mutants appear as points."}
dc_hyp$time <- as.factor(dc_hyp$time)
scat1 <- ggplot(data=subset(dc_hyp, deletion != "HAP1"), aes(x = DC1, y = DC2, color = time, label = deletion)) +
  geom_point() +
  scale_color_viridis_d(option = "plasma") +
  theme_minimal() +
  theme(plot.title.position = "plot") +
  ggtitle("GSE115171 & GSE85595") 

# add WT
scat1 + 
  geom_point(data=subset(dc_hyp, deletion == "HAP1"), aes(x = DC1, y = DC2), shape = 3)
```



We next performed differential expression along DC1 and DC2 to determine which genes were causing separation along each diffusion component.

```{r diffex}
d0 <- DGEList(counts)                          # initialize the DGE object
d <- calcNormFactors(d0)                       # calculate normalization factors
dc_hyp_mm <- model.matrix(~dc_hyp$DC1 + ~dc_hyp$DC2 + ~dc_hyp$DC3 + dc_hyp$DC4 + dc_hyp$DC5)
dc_hyp_y <- voom(d, dc_hyp_mm, plot = F)
dc_hyp_fit <- lmFit(dc_hyp_y, dc_hyp_mm)

dc1_hyp <- dc_contrast(dc_fit = dc_hyp_fit, coeff = 2)
dc2_hyp <- dc_contrast(dc_fit = dc_hyp_fit, coeff = 3)
```


```{r DC1_viz, fig.height = 5, fig.cap= "Gene set enrichment for differentially expressed genes along diffusion component 1. Only the top 10 enriched categories are displayed for each enrichment analysis. **A** KEGG pathways that are enriched in induced genes. **B** KEGG pathways that are enriched in repressed genes. **C** Gene Ontologies in category *biological processes* that are enriched in induced genes. **D** Gene Ontologies in category *biological processes* that are enriched in repressed genes."}
dc_viz(dc1_hyp)
#dc_top_diffex_plot(dc = dc_hyp, dc_num = "DC1", dc_con = dc1_hyp, color_by = "hr")
```

DC1 separates a peak hypoxic response that is dominated by samples at t = 30min of nitrogen exposure. 
The top differentially expressed genes along DC1 are shown in **Table 1**, while **Figure 2** depicts gene set enrichment for induced and repressed genes. 

**WHAT IS CAUSING THE 30 MIN PEAK ON DC1?** 
We expected this to be *ROX1*/*MOT3* regulated genes because the rox1 mot3 deletants fall so far out on DC1 (see **Figure S1**), but *ROX1*/*MOT3* is solidly captured in DC2 (see 2 next paragraphs/**Table 1**).

DC2 separates hypoxic response by time exposed to 100% nitrogen. 
The top differentially expressed genes along DC2 are shown in **Table 1**, while **Figure 3** depicts gene set enrichment for induced and repressed genes. 

Unexpectedly, *ROX1* and *MOT3*, as well as genes repressed by *ROX1* and *MOT3* like *DAN1*, *TIR* genes, *ANB1*, and *HEM13*, are significantly differentially expressed along DC2.
While we anticipated that these classic hypoxia response genes would change throughout the time course, we expected these specific differences to segregate to DC1 given that rox1 and mot3 deletants drive peak separation in DC1.


```{r most_dc_hyp1, include = T}
most1 <- dc_top_table(dc1_hyp)
dc1_hyp_up <- dc1_hyp %>%
  filter(logFC_adj > 0)
dc1_hyp_down <- dc1_hyp %>%
  filter(logFC_adj < 0)
```





```{r DC2_viz, fig.height = 5, fig.cap= "Gene set enrichment for differentially expressed genes along diffusion component 1. Only the top 10 enriched categories are displayed for each enrichment analysis. **A** KEGG pathways that are enriched in induced genes. **B** KEGG pathways that are enriched in repressed genes. **C** Gene Ontologies in category *biological processes* that are enriched in induced genes. **D** Gene Ontologies in category *biological processes* that are enriched in repressed genes."}

dc_viz(dc2_hyp)
```


```{r most_dc_hyp2, include = T}
most2 <- dc_top_table(dc2_hyp)

dc2_hyp_up <- dc2_hyp %>%
  filter(logFC_adj > 0)
dc2_hyp_down <- dc2_hyp %>%
  filter(logFC_adj < 0)
```

```{r kable_most}
kable(cbind(most1, most2), "latex", booktabs = T,
      caption = 'Top 15 induced and repressed genes for diffusion components 1 and 2.') %>%
  kable_styling(font_size = 9) %>%
  add_header_above(c("Diffusion component 1" = 3, "diffusion component 2" = 3), 
                   align = "l", escape = F)
```


### Diffusion mapping captures genes involved in oxygen regulation

We next investigated whether the genes identified as differentially expressed along DC1 and DC2 matched oxygen-regulated genes identified by seven previous microarray and RNA-sequencing studies. 
These genes were collected in the 2017 publication by (**supplement table S1**) [@bendjilali2017time]. 

```{r oxygen_regulated_genes}
url <- "https://www.g3journal.org/highwire/filestream/473070/field_highwire_adjunct_files/15/TableS3.xlsx"
download.file(url = url, destfile = "hypoxia/inputs/tabs3_oxygen_regulated_genes.xlsx")
ox_reg <- readxl::read_excel(path = "hypoxia/inputs/tabs3_oxygen_regulated_genes.xlsx", 
                             col_names = c('ORF', 'COMMON', 'description', 'Becerra_2002',
                                           'Hickman_2007', 'Hickman_2011', 'Kwast_2002', 
                                           'Lai_2005', 'Lai_2006', 'terLinde_1999', 
                                           'terLinde_2002', 'hypoxic', 'aerobic', 'total'),
                             sheet = 1, skip = 1)

hypoxic1 <- ox_reg %>%
  group_by(hypoxic) %>%
  tally()

aerobic1 <- ox_reg %>%
  group_by(aerobic) %>%
  tally()
```

Of the 11 genes that were characterized as oxygen-regulated in six or seven microarray studies, we identified 10 in the first two diffusion components.
The one aerobic gene not identified, *CYC1*, was filtered prior to differential expression because it was not highly expressed enough across all samples 
(mean = `r round(mean(d0["YJR048W", ][["counts"]]), digits = 2)`, sd = `r round(sd(d0["YJR048W", ][["counts"]]), digits = 2)`).

```{r test_hypoxia_intersect_microarray}
hypoxic_genes <- ox_reg %>%
  filter(hypoxic >= 6) %>%
  dplyr::select(ORF, COMMON)

hypoxic_genes_dc1_hyp <- hypoxic_genes[hypoxic_genes$ORF %in% dc1_hyp$ORF, ]
hypoxic_genes_dc2_hyp <- hypoxic_genes[hypoxic_genes$ORF %in% dc1_hyp$ORF, ]
# table(hypoxic_genes_dc1_hyp $ORF %in% hypoxic_genes_dc2_hyp$ORF)
# table(hypoxic_genes$ORF %in% dc2_hyp$ORF)
# table(hypoxic_genes$ORF %in% dc1_hyp$ORF)
```

```{r test_aerobic_intersect_microarray}
aerobic_genes <- ox_reg %>%
  filter(aerobic >= 6) %>%
  dplyr::select(ORF, COMMON)

aerobic_genes_dc1_hyp <- aerobic_genes[aerobic_genes$ORF %in% dc1_hyp$ORF, ]
aerobic_genes_dc2_hyp <- aerobic_genes[aerobic_genes$ORF %in% dc2_hyp$ORF, ]
# table(aerobic_genes_dc1_hyp$ORF %in% aerobic_genes_dc2_hyp$ORF)
# table(aerobic_genes$ORF %in% dc1_hyp$ORF)
# table(aerobic_genes$ORF %in% dc2_hyp$ORF)
```

[@bendjilali2017time] further characterized genes as hypoxic or aerobic using differential expression on their wild type time series data set. 
To understand whether diffusion mapping captured the same oxygen-regulated genes, we intersected the hypoxic and aerobic genes with genes that were differentially expressed along DC1 and DC2. 
Two hundred forty six of 291 (84.5%) genes predicted as aerobic by [@bendjilali2017time] were significantly differentially expressed, while 418 of 519 (80.5%) genes predicted as hypoxic were significantly differentially expressed. 
Further, if we divide the diffusion map into four quadrants, the majority of aerobic genes segregate to the same quadrant as time = 0-5min samples, while the majority of hypoxic genes segregate to quadrants with later time points.

```{r}
url <- 'https://www.g3journal.org/highwire/filestream/473070/field_highwire_adjunct_files/13/TableS1.xlsx'
download.file(url = url, destfile = "hypoxia/inputs/tabs1_rnaseq_res.xlsx")
rnaseq_oxreg <- readxl::read_excel(path = "hypoxia/inputs/tabs1_rnaseq_res.xlsx", sheet = 1)%>%
  filter(grepl("^Y", gene))

hypoxic <- rnaseq_oxreg %>%
  filter(oxygen.regulated == "hypoxic") 

aerobic <- rnaseq_oxreg %>%
  filter(oxygen.regulated == "aerobic")

oxreg_int <- data.frame(intersection = c("DC1 induced", "DC1 repressed", "DC2 induced", "DC2 repressed", "total"),
                        hypoxic= c(table(hypoxic$gene %in% dc1_hyp_up$ORF)[2],
                                    table(hypoxic$gene %in% dc2_hyp_down$ORF)[2],
                                    table(hypoxic$gene %in% dc1_hyp_up$ORF)[2],
                                    table(hypoxic$gene %in% dc2_hyp_down$ORF)[2],
                                    table(hypoxic$gene %in% c(dc1_hyp_up$ORF, dc1_hyp_down$ORF, dc2_hyp_down$ORF, dc2_hyp_up$ORF))[2]), 
                        aerobic = c(table(aerobic$gene %in% dc1_hyp_up$ORF)[2],
                                    table(aerobic$gene %in% dc1_hyp_down$ORF)[2],
                                    table(aerobic$gene %in% dc2_hyp_up$ORF)[2],
                                    table(aerobic$gene %in% dc2_hyp_down$ORF)[2],
                                    table(aerobic$gene %in% c(dc1_hyp_up$ORF, dc1_hyp_down$ORF, dc2_hyp_down$ORF, dc2_hyp_up$ORF))[2]))
kable(oxreg_int, "latex", booktabs = T, col.names = c("Intersection", "Hypoxic (519 predicted)", "Aerobic (291 predicted)"),
      caption = 'Intersection between oxygen regulation genes predicted by Bendjilali et al. 2017 and those predicted by this study.') %>%
  kable_styling(font_size = 9)
```

### Overlap between diffusion components

Most genes are differentially expressed in a single component. 
However, some genes are differentially expressed in both diffusion components. 
This likely indicates that these genes play a role in both the transcriptional remodeling that occurs at 30 mins, and in remodeling and maintenance that occur at longer time scales.

```{r upset_plot_hypoxia}
diffex_list <- list(DC1_induced = dc1_hyp_up$ORF,
                    DC1_repressed = dc1_hyp_down$ORF,
                    DC2_induced = dc2_hyp_up$ORF,
                    DC2_repressed = dc2_hyp_down$ORF)
upset_plt<- upset(fromList(diffex_list))
```

```{r dc_hyp_int}
dc2_hyp_in_dc1_hyp <- dc2_hyp[dc2_hyp$ORF %in% dc1_hyp$ORF, ] 

dc1_hyp_in_dc2_hyp <- dc1_hyp[dc1_hyp$ORF %in% dc2_hyp$ORF, ]
dc1_hyp_in_dc2_hyp <- dc1_hyp_in_dc2_hyp[order(match(dc1_hyp_in_dc2_hyp$ORF, dc2_hyp_in_dc1_hyp$ORF)), ]
stopifnot(all.equal(dc1_hyp_in_dc2_hyp$ORF, dc2_hyp_in_dc1_hyp$ORF))

dc_hyp_int <- data.frame(dc1_hyp = dc1_hyp_in_dc2_hyp$logFC, dc2_hyp = dc2_hyp_in_dc1_hyp$logFC, ORF = dc2_hyp_in_dc1_hyp$ORF)
# lm(dc1_hyp ~ dc2_hyp, data = dc_hyp_int)
dc_hyp_int_plt <- ggplot(dc_hyp_int, aes(x = dc1_hyp, y = dc2_hyp, label = ORF)) +
  geom_point(alpha = .05) +
  xlim(c(-40, 41)) + ylim(c(-40, 41)) + 
  labs(x = "Log2FC DC1", y = "log2FC DC2") +
  theme_minimal()
```

```{r, fig.height=3, fig.cap="Some genes are expressed in diffusion component (DC) 1 and DC2. **A** Upset plot showing intersections of genes that are significantly induced and repressed in DC1 and DC2. The majority of genes are significantly differentially expressed in a single diffusion component. **B** Scatter plot of log2FC values for genes are significantly differentially expressed along both diffusion components. Color is fully saturated when 20 points overlap."}
plot_grid(as.grob(upset_plt), dc_hyp_int_plt, ncol = 2, labels = c("A", "B"))
```

# Discussion

In this study, we paired diffusion mapping with differential expression to capture primary fermentation of pinot noir wine measured with RNA-sequencing of musts from 15 vineyard sites. 
The time-ordered clustering of replicates and samples indicates that diffusion mapping captures time-varying patterns in this expression data set. 

Benefits of this method:

+ unbalanced sample designs (e.g. no replicates), because it uses linear regression over a continuous variable.
+ however, replication is still a good idea to make sure that a sample is not anomalous
+ uneven time series
+ compared to other methods, isolates time-varying responses in a component
+ easy to interpret a lot of samples given their relative distance to one another  


<!-- As model organism, the genes and pathways of *S. cerevisiae* are well annotated  -->
<!-- While diffusion mapping is  -->

# Methods

## Fermentation

## Sequencing and preprocessing

+ UC Davis DNA tech core
+ trimming, mapping, 2019 umi barcodes, htseq

We normalized read counts based on total number of reads per sample (library size) prior to constructing the diffusion map. 
As input to differential expression, we used raw sequencing counts and performed filtering and normalization with the limma package using the `calcNormFactors()` function [@limma2015].

## Construction of Diffusion Maps

Prior to diffusion map construction, we filtered gene counts to non-mitochondrial mRNA. 
We built the diffusion map using the Bioconductor package destiny [@destiny2015].
The destiny diffusion map algorithm accounts for noise types inherent in RNA-sequencing data such as missing values and uncertainties in measurements.

To select number of neighbors (k-size) in the construction of the diffusion map, we tested all values of *k* from 1 - (*n* - 2). 
We selected the k-size that produced a graph with one component and that maximized the difference between sequential eigenvalues when eigenvalues were ordered. 
We selected k-sizes of 31 for the 2017 vintage and 41 for the 2019 vintage.

## Differential expression

To determine which genes drove separation of samples along each component, we used differential expression to correlate each gene with diffusion component values.
We used limma to fit a linear regression model to each gene [@limma2015].
Using this model, the log~2~ fold change is the slope of the line for each unit increase in the diffusion component. 
We normalized log~2~ fold change values by calculating the length of the diffusion component (max - minimum) and multiplying all log~2~ fold change values by this amount. 
We analyzed log~2~ fold change values greater than two, i.e. genes with a log~2~ fold change of at least 2 between the most separated samples along a diffusion component.

+ clusterProfiler

## Comparison against differential expression with brix

# References

<div id="refs"></div>


\newpage
# Supplementary material {-}

\beginsupplement

```{r brix_upset17}
brix_list_17 <- list(dc1_17_induced = filter(dc1_17, logFC > 3)$ORF,
                     dc1_17_repressed = filter(dc1_17, logFC < -3)$ORF,
                     brix_17_induced = filter(brix_17, logFC > 0)$ORF,
                     brix_17_repressed = filter(brix_17, logFC < -0)$ORF
)
upset_brix_17 <- UpSetR::upset(fromList(brix_list_17), nsets = 8, nintersects = 200, 
                       order.by = "freq", keep.order = T)
```

```{r brix_upset19}
brix_list_19 <- list(dc1_19_induced = filter(dc1_19, logFC > 3)$ORF,
                     dc1_19_repressed = filter(dc1_19, logFC < -3)$ORF,
                     brix_19_induced = filter(brix_19, logFC > 0)$ORF,
                     brix_19_repressed = filter(brix_19, logFC < 0)$ORF
)
upset_brix_19 <- UpSetR::upset(fromList(brix_list_19), nsets = 8, nintersects = 200, 
                               order.by = "freq", keep.order = T)
```

```{r dc1_upset}
dc1_list <- list(dc1_17_induced = filter(dc1_17, logFC > 3)$ORF,
                 dc1_17_repressed = filter(dc1_17, logFC < -3)$ORF,
                 dc1_19_induced = filter(dc1_19, logFC > 3)$ORF,
                 dc1_19_repressed = filter(dc1_19, logFC < -3)$ORF)
upset_dc1 <- UpSetR::upset(fromList(dc1_list), nsets = 8, nintersects = 200, 
                   order.by = "freq", keep.order = T)
```

```{r brix_upset}
brix_list <- list(brix_17_induced = filter(brix_17, logFC > 0)$ORF,
                  brix_17_repressed = filter(brix_17, logFC < -0)$ORF,
                  brix_19_induced = filter(brix_19, logFC > 0)$ORF,
                  brix_19_repressed = filter(brix_19, logFC < 0)$ORF)
upset_brix <- UpSetR::upset(fromList(brix_list), nsets = 8, nintersects = 200, 
                   order.by = "freq", keep.order = T)
```

```{r upsetgrid, fig.height= 5, fig.cap="Upset plot depicting intersections between differentially expressed genes using diffusion component 1 or brix as continuous variable. **A.** Intersection of differentially expressed genes from the 2017 vintage. **B.** Intersection of genes from the 2019 vintage. **C.** Intersection of genes differentially expressed along diffusion component 1 from the 2017 and 2019 vintages. **D** Intersection of genes differentially expressed relative to change in brix from the 2017 and 2019 vintages."}

plot_grid(as.grob(upset_brix_17), as.grob(upset_brix_19), 
          as.grob(upset_dc1), as.grob(upset_brix),
          ncol = 2, labels = c("A", "B", "C", "D"))
```

```{r annotatedplts17, fig.height = 4, fig.width = 6, eval = T, fig.cap="**A**, **B**"}
ggarrange(dc_top_diffex_plot(dc = dc17, dc_num = "DC1", dc_con = dc1_17, color_by = "brix"),
          dc_top_diffex_plot(dc = dc17, dc_num = "DC2", dc_con = dc2_17, color_by = "ava"),
          nrow = 2, labels = c("A", "B"))
```

```{r annotatedplts19, fig.height = 8, fig.width = 6, eval = T, fig.cap="**A**, **B**"}
ggarrange(dc_top_diffex_plot(dc = dc19, dc_num = "DC1", dc_con = dc1_19, color_by = "brix"),
          dc_top_diffex_plot(dc = dc19, dc_num = "DC2", dc_con = dc2_19, color_by = "hr"),
          dc_top_diffex_plot(dc = dc19, dc_num = "DC3", dc_con = dc3_19, color_by = "hr"),
          dc_top_diffex_plot(dc = dc19, dc_num = "DC4", dc_con = dc4_19, color_by = "hr"),
          nrow = 4, labels = c("A", "B", "C", "D"))
```

```{r annotatedplts192, fig.height = 8, fig.width=6, eval = T, fig.cap="**A**, **B**"}
ggarrange(dc_top_diffex_plot(dc = dc19, dc_num = "DC5", dc_con = dc5_19, color_by = "ava"),
          dc_top_diffex_plot(dc = dc19, dc_num = "DC6", dc_con = dc6_19, color_by = "ava"),
          dc_top_diffex_plot(dc = dc19, dc_num = "DC7", dc_con = dc7_19, color_by = "ava"),
          dc_top_diffex_plot(dc = dc19, dc_num = "DC8", dc_con = dc8_19, color_by = "ava"),
          nrow = 4, labels = c("A", "B", "C", "D"))
```

```{r annotatedplts193, fig.height = 4, fig.cap="**A**, **B**"}
ggarrange(dc_top_diffex_plot(dc = dc19, dc_num = "DC9", dc_con = dc9_19, color_by = "ava"),
          dc_top_diffex_plot(dc = dc19, dc_num = "DC10", dc_con = dc10_19, color_by = "ava"),
          nrow = 2, labels = c("A", "B"))
```

## Enrichment analysis 

```{r DC1viz2017, fig.height = 7.9, fig.cap= "Gene set enrichment for differentially expressed genes along diffusion component 1 in the 2017 vintage. All enriched categories with p < 0.05 after Bonferroni correction are shown. Pathways on the right side of the figure are induced in samples with a high diffusion component value, while pathways on the left of the figure are induced in samples with a low diffusion component value. Log~2~ fold change values reflect the total change in expression across the diffusion component."}
dc_top_enrich_plot(dc = dc17, dc_num = "DC1", dc_con = dc1_17, color_by = "brix", show_genes = F)
```

```{r DC2viz2017, fig.height = 7.5, fig.cap= "Gene set enrichment for differentially expressed genes along diffusion component 2 in the 2017 vintage. All enriched categories with p < 0.05 after Bonferroni correction are shown. Pathways on the right side of the figure are induced in samples with a high diffusion component value, while pathways on the left of the figure are induced in samples with a low diffusion component value. Log~2~ fold change values reflect the total change in expression across the diffusion component."}
dc_top_enrich_plot(dc = dc17, dc_num = "DC2", dc_con = dc2_17, color_by = "brix", show_genes = F)
```

```{r DC1viz2019, fig.height = 8, fig.cap= "Gene set enrichment for differentially expressed genes along diffusion component 1 in the 2019 vintage. All enriched categories with p < 0.05 after Bonferroni correction are shown. Pathways on the right side of the figure are induced in samples with a high diffusion component value, while pathways on the left of the figure are induced in samples with a low diffusion component value. Log~2~ fold change values reflect the total change in expression across the diffusion component."}
dc_top_enrich_plot(dc = dc19, dc_num = "DC1", dc_con = dc1_19, color_by = "brix", show_genes = F)
```

```{r DC2viz2019, fig.height = 8, fig.cap= "Gene set enrichment for differentially expressed genes along diffusion component 2 in the 2019 vintage. All enriched categories with p < 0.05 after Bonferroni correction are shown. Pathways on the right side of the figure are induced in samples with a high diffusion component value, while pathways on the left of the figure are induced in samples with a low diffusion component value. Log~2~ fold change values reflect the total change in expression across the diffusion component."}
dc_top_enrich_plot(dc = dc19, dc_num = "DC2", dc_con = dc2_19, color_by = "hr", show_genes = F)
```

```{r DC3viz2019, fig.height = 3.7, fig.cap= "Gene set enrichment for differentially expressed genes along diffusion component 3 in the 2019 vintage. All enriched categories with p < 0.05 after Bonferroni correction are shown. Pathways on the right side of the figure are induced in samples with a high diffusion component value, while pathways on the left of the figure are induced in samples with a low diffusion component value. Log~2~ fold change values reflect the total change in expression across the diffusion component."}
dc_top_enrich_plot(dc = dc19, dc_num = "DC3", dc_con = dc3_19, color_by = "hr", show_genes = F)
```

```{r DC4viz2019, fig.height = 4.1, fig.cap="Gene set enrichment for differentially expressed genes along diffusion component 4 in the 2019 vintage. All enriched categories with p < 0.05 after Bonferroni correction are shown. Pathways on the right side of the figure are induced in samples with a high diffusion component value, while pathways on the left of the figure are induced in samples with a low diffusion component value. Log~2~ fold change values reflect the total change in expression across the diffusion component."}
dc_top_enrich_plot(dc = dc19, dc_num = "DC4", dc_con = dc4_19, color_by = "hr", show_genes = F)
```

```{r DC5viz2019, fig.height = 2, fig.cap="Gene set enrichment for differentially expressed genes along diffusion component 5 in the 2019 vintage. All enriched categories with p < 0.05 after Bonferroni correction are shown. Pathways on the right side of the figure are induced in samples with a high diffusion component value, while pathways on the left of the figure are induced in samples with a low diffusion component value. Log~2~ fold change values reflect the total change in expression across the diffusion component."}
dc_top_enrich_plot(dc = dc19, dc_num = "DC5", dc_con = dc5_19, color_by = "hr", show_genes = F)
```

```{r DC6viz2019, fig.height = 2, fig.cap="Gene set enrichment for differentially expressed genes along diffusion component 6 in the 2019 vintage. All enriched categories with p < 0.05 after Bonferroni correction are shown. Pathways on the right side of the figure are induced in samples with a high diffusion component value, while pathways on the left of the figure are induced in samples with a low diffusion component value. Log~2~ fold change values reflect the total change in expression across the diffusion component."}
dc_top_enrich_plot(dc = dc19, dc_num = "DC6", dc_con = dc6_19, color_by = "ava", show_genes = F)
```

```{r DC7viz2019, eval = F, fig.height = 7, fig.cap="Gene set enrichment for differentially expressed genes along diffusion component 7 in the 2019 vintage. All enriched categories with p < 0.05 after Bonferroni correction are shown. Pathways on the right side of the figure are induced in samples with a high diffusion component value, while pathways on the left of the figure are induced in samples with a low diffusion component value. Log~2~ fold change values reflect the total change in expression across the diffusion component."}
dc_top_enrich_plot(dc = dc19, dc_num = "DC7", dc_con = dc7_19, color_by = "ava", show_genes = F)
```

```{r DC8viz2019, fig.height = 3.7, fig.cap="Gene set enrichment for differentially expressed genes along diffusion component 8 in the 2019 vintage. All enriched categories with p < 0.05 after Bonferroni correction are shown. Pathways on the right side of the figure are induced in samples with a high diffusion component value, while pathways on the left of the figure are induced in samples with a low diffusion component value. Log~2~ fold change values reflect the total change in expression across the diffusion component."}
dc_top_enrich_plot(dc = dc19, dc_num = "DC8", dc_con = dc8_19, color_by = "ava", show_genes = F)
```

```{r DC9viz2019, eval = F, fig.height = 3, fig.cap="Gene set enrichment for differentially expressed genes along diffusion component 9 in the 2019 vintage. All enriched categories with p < 0.05 after Bonferroni correction are shown. Pathways on the right side of the figure are induced in samples with a high diffusion component value, while pathways on the left of the figure are induced in samples with a low diffusion component value. Log~2~ fold change values reflect the total change in expression across the diffusion component."}
dc_top_enrich_plot(dc = dc19, dc_num = "DC9", dc_con = dc9_19, color_by = "ava", show_genes = F)
```

```{r DC10viz2019, fig.height = 4.1, fig.cap="Gene set enrichment for differentially expressed genes along diffusion component 10 in the 2019 vintage. All enriched categories with p < 0.05 after Bonferroni correction are shown. Pathways on the right side of the figure are induced in samples with a high diffusion component value, while pathways on the left of the figure are induced in samples with a low diffusion component value. Log~2~ fold change values reflect the total change in expression across the diffusion component."}
dc_top_enrich_plot(dc = dc19, dc_num = "DC10", dc_con = dc10_19, color_by = "ava", show_genes = F)
```


```{r test, eval = F, include = F, eval = F}
gene <- 'YPR065W' # ROX1
gene <- 'YMR070W' # MOT3
gene <- "YLR228C" # ECM22
gene <- 'YDR213W' # UPC2
gene <- 'YIR033W' # MGA2
gene <- 'YGL055W' # OLE1
gene <- 'YJR150C' # DAN1
gene <- 'YDR044W' # HEM13
Y <- dc_y$E[gene, ]
dc$Y = Y
ggplot(dc, aes(y = Y, x = DC2)) +
  geom_point(color = "grey", alpha = .7) +
  ggtitle(paste(gene, ", ", 
                dc2_top %>% filter(ORF == gene) %>% dplyr::select(COMMON),
                ": Log2FC = ", 
                dc2_top %>% filter(ORF == gene) %>% dplyr::select(logFC) %>% mutate(logFC = round(logFC, digits = 3))))+
  theme_minimal() +
  labs(y = "normalized log expression value") +
  #geom_text(x = -.05, y = 12.4, label=lm_eqn(dc, 'Y', 'DC2'), parse=T) +
  geom_smooth(method='lm')

```



```{r, eval = F}
tmp <- ggplot(dc17, aes(x = DC2, y = DC3, color = brix, label = ava_id)) +
  geom_jitter() +
  scale_color_viridis_c() +
  #scale_color_tableau(palette = "Tableau 20") +
  theme_minimal() +
  theme(plot.title = element_text(hjust = 0.5)) +
  ggtitle("2017") 
ggplotly(tmp)
tmp <- ggplot(dc17, aes(x = DC2, y = DC3, color = ava, label = brix)) +
  geom_jitter() +
  #scale_color_viridis_c() +
  scale_color_tableau(palette = "Tableau 20") +
  theme_minimal() +
  theme(plot.title = element_text(hjust = 0.5)) +
  ggtitle("2017")
tmp
ggplotly(tmp)
# DC1: brix
# DC2: late ferm; brix
# DC3: ?
# DC4: ?
# DC5: ?
# DC7: ?

p <- ggplot(dc19, aes(x = DC6, y = DC7, color = brix, label = ava_id)) +
  geom_jitter() +
  scale_color_viridis_c() +
  #scale_color_tableau(palette = "Tableau 20") +
  theme_minimal() +
  theme(plot.title = element_text(hjust = 0.5)) +
  ggtitle("2019") 
ggplotly(p)
# DC1: brix
# DC2: early fermentation differences
# DC3: Wine from inoculum samples
# DC4: early ferm
# DC5: late ferm; brix
# DC6: early ferm
# DC7: early ferm

```