---
title: "Mapping trajectories of global shifts in gene expression"
always_allow_html: yes
author: 
- Taylor Reiter
- Rachel Montpetit
- ...
- C. Titus Brown
- Ron Runnebaum
- Ben Montpetit
date: "`r format(Sys.time(), '%d %B, %Y')`"
bibliography: ../2020-pn-tmp/bibliography.bib
graphics: yes
figsintext: yes
geometry: "left=1in,right=1in,top=1in,bottom=1in"
header-includes:
- \usepackage{verbatim}
- \newcommand{\comm}[1]{}
- \usepackage{float} \floatplacement{figure}{H}
- \newcommand{\beginsupplement}{\setcounter{table}{0}  \renewcommand{\thetable}{S\arabic{table}}
  \setcounter{figure}{0} \renewcommand{\thefigure}{S\arabic{figure}}}
- \usepackage[font={small,it}, labelfont={bf}]{caption}
- \usepackage{setspace}\singlespacing
keep_tex: yes
output:
  bookdown::word_document2:
    fig_caption: yes
    toc: false
---
  
```{r setup, include=FALSE, eval = T}
knitr::opts_chunk$set(echo = F, eval = T, message = F, warning = F, cache = T, 
                      include = T, fig.keep = 'high', fig.path = "Rmd_fig_out/",
                      dpi=400, fig.width = 6.5)
```

```{r libs, eval = T}
library(dplyr)
library(purrr)
library(tidyr)
library(tibble)
library(readr)
library(purrr)
library(stringr)
library(ggplot2)
library(ggthemes)
library(ggpubr)
library(ggrepel)
library(ggplotify)
library(ggcorrplot)
library(limma)
library(edgeR)
library(clusterProfiler)
library(org.Sc.sgd.db)
library(kableExtra)
library(gridExtra)
library(UpSetR)
library(cowplot)
library(knitr)
library(RColorBrewer)
set.seed(1)
```

```{r diffusion_mapping_functions}
## as implemented here: https://doi.org/10.6084/m9.figshare.12864011.v4
## standardize columns func
norm_mat <- function(mat){
  ms <- apply(mat, 2, function(col) (col - mean(col)) / sd(col))
  ms
}

## compute euclid dist and standrdize to similarities
get_euc <- function(mat, n.threads = 1, alt = 'euclidean'){
  if(n.threads > 1){
    eu <- parDist(mat, method = alt, threads = n.threads) %>% as.matrix()
    ieu <- 1 / eu
    diag(ieu) <- 0
  }
  
  if(n.threads == 1){
    eu <- dist(mat, method = alt) %>% as.matrix()
    ieu <- 1 / eu
    diag(ieu) <- 0
  }

  ieu
}

## function to threshold the normalized distance mat
threshold <- function(mat, top_k = 10){
  thr <- mat
  tnr <- nrow(thr)
  ## set similarities outside of the top k to 0
  for(i in 1:tnr){
    ## rank entries in each row in reverse order (so largest value == rank 1), 
    ## and set entries that are outside of 1:k to 0
    thr[i, !rank(-thr[i, ], ties.method = 'random') %in% 1:top_k] <- 0
  }
  
  for(i in 1:tnr){
    for(j in 1:tnr){
      if(thr[i, j] == 0 & thr[j, i] != 0){
        thr[i, j] <- thr[j, i]
      }
    }
  }
  thr
}

## function to calculate norm laplacian
get_laplac <- function(mat){
  L <- -mat
  S <- rowSums(mat)
  nL <- L / S
  diag(nL) <- 1
  nL
}
```

```{r processing_functions}
orf_to_common <- function(keys){
  common <- AnnotationDbi::select(org.Sc.sgd.db,
                                  keys = keys,
                                  columns=c("COMMON"),
                                  keytype="ORF")
  return(common)
}
orf_to_description <- function(keys){
  description <- AnnotationDbi::select(org.Sc.sgd.db::org.Sc.sgd.db,
                                  keys = keys,
                                  columns=c("DESCRIPTION"),
                                  keytype="ORF")
  return(description)
}
lm_eqn <- function(df, y, x){
    formula = as.formula(sprintf('%s ~ %s', y, x))
    m <- lm(formula, data=df);
    # formating the values into a summary string to print out
    # ~ give some space, but equal size and comma need to be quoted
    eq <- substitute(italic(target) == a + b %.% italic(input)*","~~italic(r)^2~"="~r2*","~~p~"="~italic(pvalue), 
         list(target = y,
              input = x,
              a = format(as.vector(coef(m)[1]), digits = 2), 
              b = format(as.vector(coef(m)[2]), digits = 2), 
             r2 = format(summary(m)$r.squared, digits = 3),
             # getting the pvalue is painful
             pvalue = format(summary(m)$coefficients[2,'Pr(>|t|)'], digits=1)
            )
          )
    as.character(as.expression(eq));                 
}

dc_contrast <- function(dc_fit, coeff, adj.p.filt = 0.01, filter_logFC = T){
  # Test diffusion component coefficient
  dc_con <- contrasts.fit(dc_fit, coefficients = coeff)   
  dc_b <- eBayes(dc_con)
  dc_top <- topTable(dc_b, sort.by = "logFC", number = Inf)
  dc_top <- dc_top[dc_top$adj.P.Val < adj.p.filt, ]
  
  # Translate ORF to Common
  dc_top$ORF <- rownames(dc_top)
  commons <- orf_to_common(dc_top$ORF)
  # subset to first occurrence of ORF
  commons <- commons[match(unique(commons$ORF), commons$ORF), ]
  commons <- commons[ , -2]
  commons$COMMON <- ifelse(is.na(commons$COMMON), commons$ORF, commons$COMMON)
  commons <- unique(commons[ , ])
  dc_top <- left_join(dc_top, commons, by = c("ORF"))
  # normalize logfc based on min and max of DC
  normalizer <- max(dc_fit$design[ , coeff]) - min(dc_fit$design[ , coeff])
  dc_top <- dc_top %>%
    mutate(logFC_adj = logFC*normalizer)
  if(filter_logFC == F) {
    return(dc_top)
  } else {
    # Join Common to diffex results
    dc_top <- dc_top %>%
      filter(abs(logFC_adj) > 2)
    return(dc_top)
  }
}

dc_viz <- function(dc_con){
  # separate to induced and repressed
  up <- dc_con[dc_con$logFC > 0, ]
  down <- dc_con[dc_con$logFC < 0, ] 
  
  # KEGG enrichment
  ekegg_up <- enrichKEGG(up$ORF, "sce", pAdjustMethod = "bonferroni") 
  plt_rows <- ekegg_up@result %>%
    filter(p.adjust < .05) %>%
    nrow()
  plt1 <- dotplot(ekegg_up, showCategory = plt_rows, font.size = 5) + theme_minimal(base_size = 5)
  
  ekegg_down <- enrichKEGG(down$ORF, "sce", pAdjustMethod = "bonferroni")
  plt_rows <- ekegg_down@result %>%
    filter(p.adjust < .05) %>%
    nrow()
  plt2 <- dotplot(ekegg_down, showCategory = plt_rows, font.size = 5) + theme_minimal(base_size = 5)
  
  # GO enrichment
  ego_up <- enrichGO(up$COMMON, OrgDb = org.Sc.sgd.db, keyType = "GENENAME", 
                     ont = "ALL", pAdjustMethod = "bonferroni")
  plt_rows <- ego_up@result %>%
    filter(p.adjust < .05) %>%
    nrow()
  plt3 <- dotplot(ego_up, showCategory = plt_rows, font.size = 5) + theme_minimal(base_size = 5)
  
  ego_down <- enrichGO(down$COMMON, OrgDb = org.Sc.sgd.db, keyType = "GENENAME", 
                       ont = "ALL", pAdjustMethod = "bonferroni")
  plt_rows <- ego_down@result %>%
    filter(p.adjust < .05) %>%
    nrow()
  plt4 <- dotplot(ego_down, showCategory = plt_rows, font.size = 5) + theme_minimal(base_size = 5)
  
  # combine plots
  ggarrange(plt1, plt2, plt3, plt4, ncol = 2, nrow = 2, common.legend = T, 
            legend = "bottom", labels = c("A", "B", "C", "D"),
            heights = c(1, 2))
}

# this plot only makes sense when there are > ~10 genes per component that are
# differentially expressed
dc_top_enrich_plot <- function(dc, dc_num, dc_con, color_by = c("brix", "ava", "hr"), show_genes = F){
  # dc <- dc19
  # dc_num <- "DC3"
  # dc_con <- dc3_19
  # separate to induced and repressed -- note that filtering for 
  # log2fc is done in dc_contrast() when normalize_logFC = T. 
  # default is a normalizes logFC > abs(2) 
  # therefore, up down separates the positive and negative logFC values
  
  up <- dc_con[dc_con$logFC > 0, ]
  down <- dc_con[dc_con$logFC < 0, ] 
               
  # KEGG enrichment
  if(nrow(up) != 0){
    ekegg_up <- enrichKEGG(up$ORF, "sce", pAdjustMethod = "bonferroni") 
    ekegg_up_top <- ekegg_up@result %>%
      filter(p.adjust < 0.05) %>%
      dplyr::select(Description, geneID) %>%
      mutate(enrichment = "KEGG") 
  } else {
    ekegg_up_top <- data.frame("Description" = character(),
                               "geneID" = character(),
                               "enrichment" = character())
  }
  
  if(nrow(down) != 0){
    if(!identical(down$ORF, down$COMMON)) { # Filter out datasets where everything is unknown
      ekegg_down <- enrichKEGG(down$ORF, "sce", pAdjustMethod = "bonferroni")
      ekegg_down_top <- ekegg_down@result %>%
        filter(p.adjust < 0.05) %>%
        dplyr::select(Description, geneID) %>%
        mutate(enrichment = "KEGG")
    } else {
      ekegg_down_top <- data.frame("Description" = character(),
                                   "geneID" = character(),
                                   "enrichment" = character())
    }
  } else {
    ekegg_down_top <- data.frame("Description" = character(),
                                 "geneID" = character(),
                                 "enrichment" = character())
  }
  
  # GO enrichment
  if(nrow(up) != 0){
    ego_bp_up <- enrichGO(up$COMMON, OrgDb = org.Sc.sgd.db, keyType = "GENENAME", 
                          ont = "BP", pAdjustMethod = "bonferroni")
    ego_bp_up_top <- ego_bp_up@result %>%
      filter(p.adjust < 0.05) %>%
      dplyr::select(Description, geneID) %>%
      mutate(enrichment = "GO BP")
    
    ego_mf_up <- enrichGO(up$COMMON, OrgDb = org.Sc.sgd.db, keyType = "GENENAME", 
                          ont = "MF", pAdjustMethod = "bonferroni")
    ego_mf_up_top <- ego_mf_up@result%>%
      filter(p.adjust < 0.05) %>%
      dplyr::select(Description, geneID) %>%
      mutate(enrichment = "GO MF")
    
    ego_cc_up <- enrichGO(up$COMMON, OrgDb = org.Sc.sgd.db, keyType = "GENENAME", 
                          ont = "CC", pAdjustMethod = "bonferroni")
    ego_cc_up_top <- ego_cc_up@result %>%
      filter(p.adjust < 0.05) %>%
      dplyr::select(Description, geneID) %>%
      mutate(enrichment = "GO CC")
  } else {
    ego_cc_up_top <- ego_mf_up_top <- ego_bp_up_top <- data.frame("Description" = character(),
                                                                  "geneID" = character(),
                                                                  "enrichment" = character())
  }
  
  if(nrow(down) != 0){
    if(!identical(down$ORF, down$COMMON)) { # Filter out datasets where everything is unknown
      ego_bp_down <- enrichGO(down$COMMON, OrgDb = org.Sc.sgd.db, keyType = "GENENAME", 
                              ont = "BP", pAdjustMethod = "bonferroni")
      ego_bp_down_top <- ego_bp_down@result %>%
        filter(p.adjust < 0.05) %>%
        dplyr::select(Description, geneID) %>%
        mutate(enrichment = "GO BP")
      
      ego_mf_down <- enrichGO(down$COMMON, OrgDb = org.Sc.sgd.db, keyType = "GENENAME", 
                              ont = "MF", pAdjustMethod = "bonferroni")
      ego_mf_down_top <- ego_mf_down@result %>%
        filter(p.adjust < 0.05) %>%
        dplyr::select(Description, geneID) %>%
        mutate(enrichment = "GO MF")
      
      ego_cc_down <- enrichGO(down$COMMON, OrgDb = org.Sc.sgd.db, keyType = "GENENAME", 
                              ont = "CC", pAdjustMethod = "bonferroni")
      if(class(ego_cc_down) == "NULL") {
        message("No enriched GO CC Down, moving on!")
        # make empty df so obj is rbind compliant
        ego_cc_down_top <- data.frame("Description" = character(), 
                                      geneID = character(), 
                                      enrichment = character())
      } else {
        ego_cc_down_top <- ego_cc_down@result %>%
          filter(p.adjust < 0.05) %>%
          dplyr::select(Description, geneID) %>%
          mutate(enrichment = "GO CC")
      } 
    } else {
      ego_cc_down_top <- ego_mf_down_top <- ego_bp_down_top <- data.frame("Description" = character(),
                                                                          "geneID" = character(),
                                                                          "enrichment" = character())
    }
  }  else {
    ego_cc_down_top <- ego_mf_down_top <- ego_bp_down_top <- data.frame("Description" = character(),
                                                                        "geneID" = character(),
                                                                        "enrichment" = character())
  }
  
  # COMBINE ENRICHMENT TABLES
  if(show_genes == T){
    down_tbl <- rbind(ego_bp_down_top, ego_cc_down_top, ego_mf_down_top, ekegg_down_top) 
    up_tbl <- rbind(ego_bp_up_top, ego_cc_up_top, ego_mf_up_top, ekegg_up_top)
  } else if(show_genes == F) {
    down_tbl <- rbind(ego_bp_down_top, ego_cc_down_top, ego_mf_down_top, ekegg_down_top) %>%
      dplyr::select(-geneID, -enrichment, Pathway = Description)
    up_tbl <- rbind(ego_bp_up_top, ego_cc_up_top, ego_mf_up_top, ekegg_up_top) %>%
      dplyr::select(-geneID, -enrichment, Pathway = Description)
  }


  # PRUNE PATHWAY NAMES TO FIRST X WORDS
  up_tbl$Pathway <- gsub(" \\(.*\\)", "", up_tbl$Pathway)
  up_tbl$Pathway <-gsub(" SSU.* from tricistronic rRNA transcript", " rRNA", up_tbl$Pathway)
  down_tbl$Pathway <- gsub(" \\(.*\\)", "", down_tbl$Pathway)
  down_tbl$Pathway <-gsub(" SSU.* from tricistronic rRNA transcript", " rRNA", down_tbl$Pathway)
  
  # PLOT ANNOTATION TABLES
  if(nrow(up_tbl) == 0){
    # make an empty table if there are no results
    up_tbl <- ggtexttable(data.frame("No_sig_genes" = ""), rows = NULL, theme = ttheme(base_size = 5, base_style = "blank", padding = unit(c(.5, .5), "mm")))
  } else {
    up_tbl <- ggtexttable(up_tbl,rows = NULL, theme = ttheme(base_size = 5, base_style = "blank", padding = unit(c(.5, .5), "mm")))
  }
  
  if(nrow(down_tbl) == 0){
    # make an empty table if there are no results
    down_tbl <- ggtexttable(data.frame("No_sig_genes" = ""), rows = NULL, theme = ttheme(base_size = 5, base_style = "blank", padding = unit(c(.5, .5), "mm")))
  } else {
    down_tbl <- ggtexttable(down_tbl,rows = NULL, theme = ttheme(base_size = 5, base_style = "blank", padding = unit(c(.5, .5), "mm")))
  }
  
  # PLOT DC
  # prep DF for plotting. Rename DC* to DC so that the same innvocation can be used
  # for ggplot across all plots
  dc <- dc %>%
    dplyr::select(DC = all_of(dc_num), brix, ava_id, ava, hr, sample)
  
  if(color_by == "brix"){
    plt_dc <- ggplot(dc, aes(x = DC, y = 0, color = brix)) +
      geom_jitter() +
      scale_color_viridis_c() +      
      labs(x = dc_num) +
      theme_minimal() +
      theme(plot.title = element_text(hjust = 0.5),
            axis.text.y = element_blank(),
            axis.title.y = element_blank(),
            legend.position = "bottom") 
  } else if(color_by == "ava") {
    plt_dc <- ggplot(dc, aes(x = DC, y = 0, color = ava)) +
      geom_jitter() +
      theme_minimal() +
      labs(x = dc_num) +
      theme(plot.title = element_text(hjust = 0.5),
            axis.text.y = element_blank(),
            axis.title.y = element_blank(),
            legend.position = "bottom") +
      scale_color_manual(values = c("OR"=or,  "AV" = av, "SNC" = snc, "RRV" = rrv, "CRN" = crn,
                                    "AS" = as, "SMV" = smv, "SRH" = srh)) 
  } else if(color_by == "hr"){
    plt_dc <- ggplot(dc, aes(x = DC, y = 0, color = factor(hr, levels = c("0", "2", "6", "16", "64", "112")))) +
      geom_point() +
      scale_color_brewer(palette = "Paired", name = "hour") + 
      geom_jitter() +
      labs(x = dc_num) +
      theme_minimal() +
      theme(plot.title = element_text(hjust = 0.5),
            axis.text.y = element_blank(),
            axis.title.y = element_blank(),
            legend.position = "bottom")
    }

  # Plot chart and table into one object
  return(ggarrange(down_tbl, plt_dc, up_tbl, ncol = 3))
}

dc_top_diffex_plot <- function(dc, dc_num, dc_con, color_by = c("brix", "ava", "hr")){
  # dc <- dc19
  # dc_num <- "DC7"
  # dc_con <- dc7_19
  # color_by = "ava"
  # separate to induced and repressed -- note that filtering for 
  # log2fc is done in dc_contrast() when normalize_logFC = T. 
  # default is a normalizes logFC > abs(2) 
  # therefore, up down separates the positive and negative logFC values
  
  up_most <- dc_con %>%
    filter(logFC_adj > 0) %>%
    dplyr::select(COMMON, logFC_adj) %>%
    mutate(logFC = round(logFC_adj, digits = 1)) %>%
    dplyr::select(-logFC_adj) %>%
    head(n = 10)
  
  down_most <- dc_con %>%
    filter(logFC_adj < 0) %>%
    dplyr::select(COMMON, logFC_adj) %>%
    mutate(logFC = round(logFC_adj, digits = 1)) %>% 
    dplyr::select(-logFC_adj) %>%
    head(n = 10)
  
  # COMBINE ENRICHMENT TABLES
  # make an annotation table
  
  if(nrow(up_most) == 0){
    # make an empty table if there are no results
    up_tbl <- ggtexttable(data.frame("No_sig_genes" = ""), rows = NULL, theme = ttheme(base_size = 6, base_style = "classic", padding = unit(c(1, 1), "mm")))
  } else{
    up_tbl <- ggtexttable(up_most, rows = NULL, theme = ttheme(base_size = 6, base_style = "classic", padding = unit(c(1, 1), "mm")))
  }
  if(nrow(down_most) == 0){
    # make an empty table if there are no results
    down_tbl <- ggtexttable(data.frame("No_sig_genes" = ""), rows = NULL, theme = ttheme(base_size = 6, base_style = "classic", padding = unit(c(1, 1), "mm")))
  } else {
    down_tbl <- ggtexttable(down_most,rows = NULL, theme = ttheme(base_size = 6, base_style = "classic", padding = unit(c(1, 1), "mm")))
  }
  # PLOT DC
  # prep DF for plotting. Rename DC* to DC so that the same innvocation can be used
  # for ggplot across all plots
  dc <- dc %>%
    dplyr::select(DC = all_of(dc_num), brix, ava_id, ava, hr, sample)
  
  if(color_by == "brix"){
    plt_dc <- ggplot(dc, aes(x = DC, y = 0, color = brix)) +
      geom_jitter() +
      scale_color_viridis_c() +
      labs(x = dc_num) +
      theme_minimal() +
      theme(plot.title = element_text(hjust = 0.5),
            axis.text.y = element_blank(),
            axis.title.y = element_blank(),
            legend.position = "bottom") 
  } else if(color_by == "ava") {
    plt_dc <- ggplot(dc, aes(x = DC, y = 0, color = ava)) +
      geom_jitter() +
      theme_minimal() +
      labs(x = dc_num) +
      theme(plot.title = element_text(hjust = 0.5),
            axis.text.y = element_blank(),
            axis.title.y = element_blank(),
            legend.position = "bottom") +
      scale_color_manual(values = c("OR"=or,  "AV" = av, "SNC" = snc, "RRV" = rrv, "CRN" = crn,
                                    "AS" = as, "SMV" = smv, "SRH" = srh)) 
  } else if(color_by == "hr"){
    plt_dc <- ggplot(dc, aes(x = DC, y = 0, color = factor(hr, levels = c("0", "2", "6", "16", "64", "112")))) +
      geom_point() +
      scale_color_brewer(palette = "Paired", name = "hour") + 
      geom_jitter() +
      labs(x = dc_num) +
      theme_minimal() +
      theme(plot.title = element_text(hjust = 0.5),
            axis.text.y = element_blank(),
            axis.title.y = element_blank(),
            legend.position = "bottom")
  } else {
      message("Please try a legitimate color values, either brix, ava, or hr")
    }

  # Plot chart and table into one object
  return(ggarrange(down_tbl, plt_dc, up_tbl, ncol = 3))
}

dc_top_table <- function(dc_con){
  up_most <- dc_con %>%
    filter(logFC > 0) %>%
    dplyr::select(ORF, COMMON, logFC_adj) %>%
    mutate(logFC = round(logFC_adj, digits = 1)) %>%
    dplyr::select(-logFC_adj) %>%
    head(n = 15)
  
  down_most <- dc_con %>%
    filter(logFC < 0) %>%
    dplyr::select(ORF, COMMON, logFC_adj) %>%
    mutate(logFC = round(logFC_adj, digits = 1)) %>%
    dplyr::select(-logFC_adj) %>%
    head(n = 15)
  
  most <- rbind(up_most, down_most)
  
  # add blank rows to enable printing when fewer than 30 diffex genes
  if(30-nrow(most) == 0){
    return(most)
  } else {
    blank_rows_to_add <- 30-nrow(most)
    blank_rows <- data.frame(ORF = rep("", blank_rows_to_add),
                             COMMON = rep("", blank_rows_to_add),
                             logFC = rep("", blank_rows_to_add))
    most <- rbind(most, blank_rows)
    return(most)
  }
}
```

```{r colors_avas}
#  ordered south to north
srh = "#FFAABB" # pink
smv = "#EE8866" # orange
as  = "#EEDD88" # light yellow
snc = "#99DDFF" # light cyan 
crn = "#AAAA00" # olive
rrv = "#BBCC33" # pear 
av  = "#77AADD" # light blue
or  = "#DDDDDD" # pale grey
```

# Introduction

During wine fermentation, *Saccharomyces cerevisiae* transforms grape must into wine by metabolizing sugars and other nutrients to ethanol and aroma compounds.
*S. cerevisiae* dynamically responds to its environment throughout this process by remodeling its transcriptome to accommodate decreasing nutrient availability and rising ethanol concentrations (reviewed in @bisson2019gene).
*S. cerevisiae's* core gene expression program is consistent across diverse fermentation conditions [@rossignol2003genome; @marks2008dynamics; @rossouw2012effect] (CITE BRIX PAPER).
Yet, overexpression of some genes leads to quantifiable differences in wine outcomes [@rossouw2008linking], and differences in nutrient availability in grape must (e.g. nitrogen concentration) leads to changes in aroma compounds like esters and higher alcohols [@carrau2008production].
The chemical composition of grape must interacts with *S. cerevisiae* to determine primary fermentation outcomes.

Grape must is chemically diverse and is a reflection of biotic and abiotic pressures encountered by a grapevine during a growing season.
Remarkably, wines produced under similar vinification conditions from genetically identical grapes grown in different locations have diverse sensory outcomes [@cantu2020investigating], many of which are reproducible across years (e.g. vintages) [@roullier2014grape].
This phenomenon is referred to as *terroir* -- the environmental and anthropomorphic factors that contribute to wine outcomes.
After observing diverse sensory outcomes in wines where the only consistent variable between fermentations was vineyard location [@cantu2020investigating], we sought to uncover the quantifiable contribution of vineyard site to wine outcomes by using *S. cerevisiae* gene expression as a biosensor to detect differences between fermentations.
As a model organism used for basic science and biotechnology, the *S. cerevisiae* genome and transcriptome is well annotated and well studied in diverse environments, including wine [@cherry1998sgd; @gasch2000genomic; @causton2001remodeling; @rossignol2003genome; @marks2008dynamics; @rossouw2012effect; @bendjilali2017time]. 
Given this resource, high throughput gene expression surveys (microarray, RNA sequencing) have been used for many wine studies and revealed the causes of stuck and sluggish fermentations [@duc2017set], the trigger for entry into stationary phase [@rossignol2003genome; @marks2008dynamics], and the impact of inter-species interactions on *S. cerevisiae* metabolism in wine [@tronchoni2017early; @curiel2017different; @alonso2019dominance].
However, after performing time series RNA sequencing on many fermentations with the aim of using differential expression to identify differences in fermentations from different vineyard sites, we found that, using standard analysis methods, we only identified the dominant and consistent global shift in gene expression across fermentations and not gene expression patterns that differentiate fermentations from different vineyard sites [@conesa2006masigpro; @robinson2010edger; @love2014moderated; @limma2015; @jung2017timesvector; @abu2018clust].
Our primary challenge arose because fermentations proceeded at different rates, leading to asynchronous biological progression among sequenced samples with respect to absolute time. 
Samples need to be at the same stage of fermentation to be compared or else many genes will appear falsely differentially expressed [@rossouw2012effect; @bar2012studying].
This is a common problem in time series experiments with multiple groups and in some experimental systems there are strategies to combat this issue [@bar2012studying].
For example, in experiments that study the cell cycle, inhibitors arrest the cell cycle at the same stage across groups thereby enabling comparisons [@spellman1998comprehensive].
This experimental approach is not applicable to wine fermentations where nutrient availability is more important than stage of the cell cycle [@bisson2019gene].
 
However, a similar problem has been addressed for analysis of single cell RNA sequencing from differentiating cells.
As cells differentiate during development, absolute time may not reflect the extent of differentiation in an individual cell.
Pseudotime analysis reorders asynchronous cells from absolute time to stage in differentiation relative to other cells undergoing the same process.
In particular, diffusion maps have been used to reorder asynchronously shifted cells because they preserve relationships between samples [@destiny2015].

Diffusion mapping is a manifold learning technique that uses information from the *k* most similar samples to construct non-linear composites of the major sources of variation among samples  [@coifman2005geometric; @coifman2006diffusion].
Dimensionality reduction algorithms like principal component analysis (PCA) and t-distributed stochastic neighbor embedding (tSNE) are commonly applied to sequencing data to identify sources of variation [@way2020compressing; @moon2019visualizing].
Diffusion mapping has strengths over these methods because it is non-linear and robust to the "horseshoe effect" (unlike PCA) [@diaconis2008horseshoes], preserves distances between samples (unlike tSNE), and is insensitive to sampling density.
As a dimensionality reduction algorithm, diffusion maps extract latent variables that represent composite sources of variation between samples.
These latent variables are unmeasured but are inferred from relationships in the sequencing data.

Here, we use diffusion maps to analyze bulk time-series RNA-sequencing of global shifts of *S. cerevisiae* metabolism during wine fermentation.
We use diffusion maps to resynchronize gene expression across treatment groups and to extract latent variables that represent the dominant sources of structure in the data sets.
Diffusion maps *per se* provide no suggestion of the underlying genes that lead to separation of samples along latent variables.
We apply differential expression along each latent variable to determine the genes responsible for separation among samples.
We find that the primary latent variable represents the dominant global shift in gene expression experienced by the majority of fermentations, while subsequent diffusion components represent subtler differences between fermentations that may reflect differences in the grape musts and underlie variable sensory outcomes.

This method is not exclusively useful for the analysis of RNA sequencing from wine fermentations and can be applied in other systems where asynchronous progression of groups with respect to sampling time masks true differences between groups.
We additionally apply this method to time series RNA sequencing of hypoxia onset among *S. cerevisiae* transcription factor knockout mutants.
We show that, as in wine, the first latent variable captures the dominant global shift in gene expression that occurs when yeast transitions from aerobic to anaerobic metabolism, and that subsequent diffusion components reveal similarities in gene expression disruption among knockouts that indicate shared regulatory programs among transcription factors.
Our method enables analysis of diverse asynchronous time series gene expression data, revealing global shifts and subtle differences among groups.


<!-- Identifying differences between groups that are separate from or in excess to the dominant and ubiquitous global shift in gene expression is another challenge. -->
<!-- Structure in gene expression exists at different scales [@moon2019visualizing], but dominant shifts experienced by all groups may overpower weaker but biologically important signals. -->
<!-- In experimental designs where groups undergo the same primary shift in metabolism but also react in subtly different ways, it's important to isolate and understand the transcriptomic context of those differences. -->
<!-- <!-- Further, while coordinated shifts in gene expression may occur across groups, sometimes there are additional differences between groups that can be difficult to identify when they are masked by global shifts.  -->

# Results and Discussion

Diffusion maps extract latent variables that represent non-linear composites of genes that vary between samples. 
We refer to these latent variables as *diffusion components*.
Within each diffusion component, each sample is represented by a number and samples that have similar gene expression profiles for genes summarized in that component are closer along the diffusion component.
Samples at the origin of a diffusion component do not vary along that component.
Differential expression along a diffusion component reveals the genes that contribute most to similarity among samples. 
Diffusion components capture diminishing structure among samples with the first diffusion component accounting for the largest variation among all samples.


```{r metadata}
info <- read_csv("samples_ava.csv")
info$ava_id <- factor(info$ava_id, levels = c('OR1', 'OR2', 'AV1', 'AV2', 
                                               'SNC1', 'SNC2', 'RRV1', 'RRV2', 'RRV3',
                                              'CRN1', 'AS1', 'AS2', 'SMV1', 
                                              'SMV2','SRH1'))

info$ava <- factor(info$ava, levels = c('OR', 'AV', 'SNC', 'RRV', 'CRN', 
                                        'AS', 'SMV','SRH'))
```

```{r diffusion_mapping_17, include = F}
## preprocess counts
counts17 <- read_tsv("v2017/outputs/counts/raw_counts.tsv") %>%
  filter(grepl(pattern = "^Y", x = gene)) %>%
  as.data.frame() %>%
  column_to_rownames("gene")
colnames(counts17) <- gsub("_readcounts.txt", "", colnames(counts17))
m17 <- sweep(counts17, 2, colSums(counts17), `/`)
m17 <- t(m17)

## build diffusion map
## as implemented here: https://doi.org/10.6084/m9.figshare.12864011.v4

nm <- m17 %>% norm_mat()          # normalize
aff <- nm %>%
  get_euc(., n.threads = 1) %>% 
  threshold(., top_k = 10)      # make affinity matrix
Lij <- aff %>% get_laplac()     # compute laplacian
eig <- eigen(Lij)               # smallest keig vectors
evl <- eig$values %>%           # get eigenvalues
  Re() %>%
  round(., digits = 10)
evc <- eig$vectors %>%          # get eigenvectors
  Re() %>%
  round(., digits = 10)

# create objects containing diffusion components
for(d in 1:length(evl)){
  assign(paste('dim', d, sep = '_'),
         Re(evc[, rank(evl) == (d + 1)])
  )
}

k_eig <- 10                     # specify num of variables you want to keep
dat <- do.call(mapply, c(FUN = cbind, mget(paste0("dim_", 1:(k_eig))))) %>%
  t() %>%
  as.data.frame()               # merge in array

colnames(dat) <- paste(paste('DC', 1:(k_eig), sep = ''))
rownames(dat) <- rownames(Lij)  # add labels

dc17 <- dat %>%                  # join with metadata
  rownames_to_column("sample") %>%
  left_join(info, by = c("sample" = "id"))
```

```{r plot_eig_vals17, eval = F}
tmp <- data.frame(val = eig$values) %>%
  arrange(val) %>%
  mutate(num = 1:length(eig$values))

ggplot(tmp, aes(y = val, x = num)) + 
  geom_point() +
  theme_minimal() +
  ggtitle("k=10 2017")
```

```{r diffusion_mapping_19, include = F}
counts19 <- read_tsv("v2019/outputs/counts/raw_counts.tsv") %>%
  dplyr::filter(grepl(pattern = "^Y", x = gene)) %>%
  dplyr::select(-`2019_inoculum_SRH1_readcounts.txt`,
                -`2019_inoculum_AS1_readcounts.txt`, 
                -`2019_inoculum_SMV2_readcounts.txt`,
                -`2019_inoculum_AS2_readcounts.txt`,
                -`2019_inoculum_RRV1_readcounts.txt`,
                -`2019_inoculum_SMV1_readcounts.txt`) %>%
  as.data.frame() %>%
  column_to_rownames("gene")
colnames(counts19) <- gsub("_readcounts.txt", "", colnames(counts19))

m19 <- sweep(counts19, 2, colSums(counts19), `/`)
m19 <- t(m19)                                             # transpose counts

## build diffusion map
## as implemented here: https://doi.org/10.6084/m9.figshare.12864011.v4

nm <- m19 %>% norm_mat()        # normalize
aff <- nm %>%
  get_euc(., n.threads = 1) %>% 
  threshold(., top_k = 10)      # make affinity matrix
Lij <- aff %>% get_laplac()     # compute laplacian
eig <- eigen(Lij)               # smallest keig vectors
evl <- eig$values %>%           # get eigenvalues
  Re() %>%
  round(., digits = 10)
evc <- eig$vectors %>%          # get eigenvectors
  Re() %>%
  round(., digits = 10)

# create objects containing diffusion components
for(d in 1:length(evl)){
  assign(paste('dim', d, sep = '_'),
         Re(evc[, rank(evl) == (d + 1)])
  )
}

k_eig <- 10                     # specify num of variables you want to keep
dat <- do.call(mapply, c(FUN = cbind, mget(paste0("dim_", 1:(k_eig))))) %>%
  t() %>%
  as.data.frame()               # merge in array

colnames(dat) <- paste(paste('DC', 1:(k_eig), sep = ''))
rownames(dat) <- rownames(Lij)  # add labels

dc19 <- dat %>%                 # join with metadata
  rownames_to_column("sample") %>%
  left_join(info, by = c("sample" = "id"))
```

```{r plot_eig_vals19, eval = F}
tmp <- data.frame(val = eig$values) %>%
  arrange(val) %>%
  mutate(num = 1:length(eig$values))

ggplot(tmp, aes(y = val, x = num)) + 
  geom_point() +
  theme_minimal() +
  ggtitle("k=10 2019")
```

```{r diffex17}
d0_17 <- DGEList(counts17)                         # initialize the DGE object
d17 <- calcNormFactors(d0_17)
dc_mm17 <- model.matrix(~dc17$DC1 + ~dc17$DC2 + ~dc17$DC3 + dc17$DC4 + dc17$DC5 + 
                          dc17$DC6 + dc17$DC7 + dc17$DC8 + dc17$DC9 + dc17$DC10)
dc_y17 <- voom(d17, dc_mm17, plot = F)
dc_fit17 <- lmFit(dc_y17, dc_mm17)
dc1_17 <- dc_contrast(dc_fit = dc_fit17, coeff = 2) # Test "DC1" coefficient
dc2_17 <- dc_contrast(dc_fit = dc_fit17, coeff = 3) # Test "DC2" coefficient
dc3_17 <- dc_contrast(dc_fit = dc_fit17, coeff = 4) # Test "DC3" coefficient
dc4_17 <- dc_contrast(dc_fit = dc_fit17, coeff = 5) # Test "DC4" coefficient
dc5_17 <- dc_contrast(dc_fit = dc_fit17, coeff = 6) # Test "DC5" coefficient
dc6_17 <- dc_contrast(dc_fit = dc_fit17, coeff = 7) # Test "DC6" coefficient
dc7_17 <- dc_contrast(dc_fit = dc_fit17, coeff = 8) # Test "DC7" coefficient
dc8_17 <- dc_contrast(dc_fit = dc_fit17, coeff = 9) # Test "DC8" coefficient
dc9_17 <- dc_contrast(dc_fit = dc_fit17, coeff = 10) # Test "DC9" coefficient
dc10_17 <- dc_contrast(dc_fit = dc_fit17, coeff = 11) # Test "DC10" coefficient
```

```{r diffex19}
d0_19 <- DGEList(counts19)              # initialize the DGE object
d19 <-calcNormFactors(d0_19)            # calculate normalization factors
dc_mm19 <- model.matrix(~dc19$DC1 + ~dc19$DC2 + ~dc19$DC3 + dc19$DC4 + dc19$DC5 + 
                          dc19$DC6 + dc19$DC7 + ~dc19$DC8 + dc19$DC9 + dc19$DC10)
dc_y19 <- voom(d19, dc_mm19, plot = F)
dc_fit19 <- lmFit(dc_y19, dc_mm19)
dc1_19 <- dc_contrast(dc_fit = dc_fit19, coeff = 2) # Test "DC1" coefficient
dc2_19 <- dc_contrast(dc_fit = dc_fit19, coeff = 3) # Test "DC2" coefficient
dc3_19 <- dc_contrast(dc_fit = dc_fit19, coeff = 4) # Test "DC3" coefficient
dc4_19 <- dc_contrast(dc_fit = dc_fit19, coeff = 5) # Test "DC4" coefficient
dc5_19 <- dc_contrast(dc_fit = dc_fit19, coeff = 6) # Test "DC5" coefficient
dc6_19 <- dc_contrast(dc_fit = dc_fit19, coeff = 7) # Test "DC6" coefficient
dc7_19 <- dc_contrast(dc_fit = dc_fit19, coeff = 8) # Test "DC7" coefficient
dc8_19 <- dc_contrast(dc_fit = dc_fit19, coeff = 9) # Test "DC8" coefficient
dc9_19 <- dc_contrast(dc_fit = dc_fit19, coeff = 10) # Test "DC9" coefficient
dc10_19 <- dc_contrast(dc_fit = dc_fit19, coeff = 11) # Test "DC10" coefficient
```

## Diffusion mapping captures the global shift of gene expression during primary fermentation

We performed 60 inoculated primary fermentations of genetically similar Pinot noir grapes grown in 15 vineyards in California and Oregon over two vintages (**Figure \@ref(fig:fig1)A**).
The initial grape must varied in parameters like initial nitrogen, pH, malic acid, tartaric acid, non-*Saccharomyces* microbial profile, and elemental profile while the final wines differed in volatile profiles and sensory outcomes (CITE sig of site paper).
Even still, *S. cerevisiae* had a consistent gene expression profile across fermentations from different vineyards and vintages (CITE brix paper).
Given the variable input media, we aimed to understand how fermentations differed, and whether those differences were related to genes known to impact the sensory outcome of wine [@cordente2012flavour].
We profiled each fermentation using time-course RNA sequencing and used *S. cerevisiae* gene expression as an indicator of similarities and differences across fermentations.
In the 2017 vintage, we took samples corresponding approximately to the early log phase (16 hours), late log phase (40 hours), stationary phase (64, 88 hours), and end of fermentation (112 hours) (**Figure \@ref(fig:fig1)B**). 
In the 2019 vintage, we shifted sampling forward to capture cellular adaptation after inoculation (2, 6 hours) and reduced sampling later in fermentation (16, 64, and 112 hours).
Fermentations progressed at different rates leading to asynchronous biological progression among samples with respect to absolute time (**Figure \@ref(fig:fig1)C**).

```{r map, eval = FALSE, fig.cap = "A. Map of vineyard site American Viticulture Areas. B. Primary fermentation sampling time points for 2017 and 2019 vintages. Times are shown in hours and are relative to inoculation."}
map <- include_graphics("figures/map_and_sampling.pdf") 
```

```{r brixplot}
brix17 <- read_csv("../2020-pn-tmp/site_metadata/2017_fermentation_log.csv")
brix19 <- read_csv("../2020-pn-tmp/site_metadata/2019_fermentation_log.csv") 
brix <- rbind(brix17, brix19) %>%
  filter(!is.na(hours_post_innoculation))%>%
  dplyr::select(-time) %>%
  pivot_longer(cols = A:D, names_to = "tank", values_to = "brix") %>%
  mutate(ava = gsub("[123]", "", ava_id))
brix_plot <- ggplot(brix, aes(x = brix, y = hours_post_innoculation, color = ava_id, shape = tank)) +
  geom_line(alpha = .7) +
  facet_wrap(~year, ncol = 1, nrow = 2) +
  labs(y = "time (hours)", color = "vineyard", shape = "tank") +
  theme_minimal() +
  theme(legend.position = "bottom",
        strip.text.x = element_text(size = 14, face = "bold")) +
  # theme(legend.position = "bottom",
  #       axis.text.x = element_text(angle = 90),
  #       legend.title = element_text(size=12, face = "bold"),
  #       legend.text = element_text(size = 9, colour = "black"),
  #       legend.key=element_blank(),
  #       strip.text.x = element_text(size = 12, face = "bold"),
  #       strip.background = element_blank(),
  #       panel.background = element_blank(), 
  #       panel.grid =  element_blank(),
  #       panel.border = element_rect(colour = "black", fill = NA, size = 1.2)) +
  scale_color_manual(values = c(SRH1 = srh, SMV2 = smv, SMV1 = smv, AS1 = as, 
                                AS2 = as, SNC1 = snc, SNC2 = snc, CRN1 = crn, 
                                RRV1 = rrv, RRV2 = rrv, RRV3 = rrv, AV1 = av,
                                AV2 = av, OR1 = or, OR2 = or),
                     breaks = c('OR1', 'OR2', 'AV1', 'AV2',  'SNC1','SNC2', 'RRV1', 'RRV2', 'RRV3',
                                 'CRN1', 'AS1', 'AS2', 'SMV1', 'SMV2','SRH1')) +
  scale_shape(guide = 'none')
```

```{r diffex_brix_17}
brix_counts17  <- counts17                    # non-wine samples already filtered
d0_17 <- DGEList(brix_counts17)               # initialize the DGE object
d17 <- calcNormFactors(d0_17)
brix_mm17 <- model.matrix(~dc17$brix)
brix_y17 <- voom(d17, brix_mm17, plot = F)
brix_fit17 <- lmFit(brix_y17, brix_mm17)
brix_17 <- dc_contrast(dc_fit = brix_fit17, coeff = 2) # Test brix coefficient
```

```{r diffex_brix_19, eval = T}
brix_counts19  <- counts19[ , 1:150]               # remove inoculum samples
d0_19 <- DGEList(brix_counts19)                    # initialize the DGE object
d19 <- calcNormFactors(d0_19)
brix19 <- info %>%
  dplyr::filter(year == 2019) %>%
  dplyr::filter(!is.na(brix))
brix_mm19 <- model.matrix(~brix19$brix)
brix_y19 <- voom(d19, brix_mm19, plot = F)
brix_fit19 <- lmFit(brix_y19, brix_mm19)
brix_19 <- dc_contrast(dc_fit = brix_fit19, coeff = 2) # Test brix coefficient
```

```{r brixcorr}
# make sure that everything has the same number of genes by not filtering by expression
# consider doing this from the outset (see above code) instead of just for this
# figure

brix_19_all <- dc_contrast(dc_fit = brix_fit19, coeff = 2, 
                           adj.p.filt = 1, filter_logFC = F) %>%
  arrange(ORF) %>%
  mutate(logFC = logFC*-1)
brix_17_all <- dc_contrast(dc_fit = brix_fit17, coeff = 2,
                           adj.p.filt = 1, filter_logFC = F) %>%
  arrange(ORF) %>%
  mutate(logFC = logFC*-1)
dc1_17_all <- dc_contrast(dc_fit = dc_fit17, coeff = 2, 
                          adj.p.filt = 1, filter_logFC = F) %>%
  arrange(ORF)
dc1_19_all <- dc_contrast(dc_fit = dc_fit19, coeff = 2,
                          adj.p.filt = 1, filter_logFC = F) %>%
  arrange(ORF)

# all.equal(brix_19_all$ORF, dc1_19_all$ORF, brix_17_all$ORF, dc1_17_all$ORF)

all_l2fc <- data.frame(brix19 = brix_19_all$logFC,
                       brix17 = brix_17_all$logFC,
                       DC1_17 = dc1_17_all$logFC,
                       DC1_19 = dc1_19_all$logFC) 
corr_l2fc <- cor(all_l2fc)
colnames(corr_l2fc) <- c("Brix 2019", "Brix 2017", "DC1 2017", "DC1 2019")
rownames(corr_l2fc) <- c("Brix 2019", "Brix 2017", "DC1 2017", "DC1 2019")


corr_plt <- ggcorrplot(corr_l2fc, type = "upper", show.diag = T, lab = T, 
                       hc.order = T, digits = 3, lab_size = 3, tl.cex = 9, 
                       lab_col = "white", show.legend = T,  legend.title =  "",
                       colors = brewer.pal(n = 3, name = "RdYlBu")) + 
  scale_fill_gradient2(low = brewer.pal(n = 3, name = "RdYlBu")[1], 
                       high = brewer.pal(n = 3, name = "RdYlBu")[3], 
                       mid = brewer.pal(n = 3, name = "RdYlBu")[2], 
                       midpoint = 0, limit = c(0, 1), space = "Lab", 
                       name = "") +
  theme(legend.position = "bottom")
```

```{r dc1, eval = FALSE, fig.height = 6, fig.width = 3.5, fig.cap = "Diffusion component 1 captures the metabolic transition that occurs as brix decreases during fermentation. The same transition occurs in **A.** the 2017 vintage and **B.** the 2019 vintage. Each point represents a sample from one time from one fermentation. Points that are closer along the x axis are more similar. The y axis is jittered to allow all points to be visualized. Points are colored by brix, where brix = 0 indicates end of fermentation. **C.** High concordance between differentially expressed genes in diffusion component 1 and genes that are differentially expressed as brix decreases."}
p17 <- ggplot(dc17, aes(x = DC1, y = 0, color = brix, label = ava_id)) +
  geom_jitter() +
  scale_color_viridis_c() +
  theme_minimal() +
  theme(plot.title = element_text(hjust = 0.5),
        axis.text.y = element_blank(),
        axis.title.y = element_blank()) +
  ggtitle("2017") 

p19 <- ggplot(dc19, aes(x = DC1, y = 0, color = brix, label = ava_id)) +
  geom_jitter() +
  scale_color_viridis_c() +
  theme_minimal() +
  theme(plot.title = element_text(hjust = 0.5),
        axis.text.y = element_blank(),
        axis.title.y = element_blank()) +
  ggtitle("2019") 

# ggarrange(ggarrange(p17, p19, ncol = 1, nrow = 2, common.legend = T, legend = "bottom", labels = c(" ", " ")), corr_plt, labels = c("", " "), ncol = 1, nrow =2)
ggarrange(ggarrange(p17, p19, ncol = 1, nrow = 2, common.legend = T, legend = "bottom", labels = c(" ", " ")), corr_plt, labels = c("", " "), ncol = 2, nrow = 1)
```

```{r fig1, fig.cap = "**A** Map of vineyard site American Viticulture Areas. **B** Primary fermentation sampling time points for 2017 and 2019 vintages. Times are shown in hours and are relative to inoculation."}
include_graphics("figures/fig1.pdf") 
```

When we applied diffusion mapping to wine fermentation, we found that diffusion component 1 captured the dominant metabolic transition of *S. cerevisiae* during fermentation and re-ordered samples according to stage of fermentation (**Figure \@ref(fig:fig2)A-B**).
While we used different sampling schemes in the 2017 and 2019 vintages, samples were ordered by decreasing brix in both vintages.
Brix is a proxy for sugar concentration during fermentation, and sugar concentrations continuously drop as *S. cerevisiae* ferments sugar into ethanol.

In a previous investigation, we used brix to perform continuous differential expression and identified the ubiquitous transcriptome response of *S. cerevisiae* to Pinot noir fermentation (CITE BRIX PAPER).
To test whether diffusion mapping captured the genes known to change expression during fermentation, we performed differential expression over diffusion component 1 and compared the log~2~ fold change values of genes against those obtained with differential expression using brix.
Log~2~ fold change was strongly correlated between both methods of differential expression (**Figure \@ref(fig:fig2)C, Figure \@ref(fig:upsetgrid)**), indicating that the primary diffusion component captured the dominant global shift in gene expression during fermentation.

```{r fig2, fig.cap = "**A, B)** Diffusion component 1 captures the metabolic transition that occurs as brix decreases during fermentation. The same transition occurs in the 2017 vintage and the 2019 vintage. Each point represents a sample from one time from one fermentation. Points that are closer along the x axis are more similar. The y axis is jittered to allow all points to be visualized. Points are colored by brix, where brix = 0 indicates end of fermentation. **C** High concordance between differentially expressed genes in diffusion component 1 and genes that are differentially expressed as brix decreases."}
knitr::include_graphics("figures/fig2.pdf")
```

## Metabolic remodeling occurs throughout fermentation

We next explored subsequent diffusion components to determine the non-dominant drivers of structure among groups (**Figure \@ref(fig:allDCs2017)**, **Figure \@ref(fig:allDCs2019)**).
While progression through fermentation was the primary driver of separation among samples, consistent transitions in early fermentation and late fermentation drove secondary structure among samples.
Diffusion components captured cellular remodeling in early fermentation, entry into stationary phase, and starvation during late fermentation.

```{r all_dc19_brix, eval = F}
dc1_19p <- ggplot(dc19, aes(x = DC1, y = 0, color = brix, label = ava_id)) +
  geom_jitter(alpha = .8) +
  xlim(c(-0.51, .51)) +
  scale_color_viridis_c() +
  theme_minimal() +
  theme(plot.title = element_text(hjust = 0.5),
        axis.text.y = element_blank(),
        axis.title.y = element_blank()) 

dc2_19p <- ggplot(dc19, aes(x = DC2, y = 0, color = brix, label = ava_id)) +
  geom_jitter(alpha = .8) +
  scale_color_viridis_c() +
  xlim(c(-0.51, .51)) +
  theme_minimal() +
  theme(plot.title = element_text(hjust = 0.5),
        axis.text.y = element_blank(),
        axis.title.y = element_blank()) 

dc3_19p <- ggplot(dc19, aes(x = DC3, y = 0, color = brix, label = ava_id)) +
  geom_jitter(alpha = .8) +
  scale_color_viridis_c() +
  xlim(c(-0.51, .51)) +
  theme_minimal() +
  theme(plot.title = element_text(hjust = 0.5),
        axis.text.y = element_blank(),
        axis.title.y = element_blank())

dc4_19p <- ggplot(dc19, aes(x = DC4, y = 0, color = brix, label = ava_id)) +
  geom_jitter(alpha = .8) +
  scale_color_viridis_c() +
  xlim(c(-0.51, .51)) +
  theme_minimal() +
  theme(plot.title = element_text(hjust = 0.5),
        axis.text.y = element_blank(),
        axis.title.y = element_blank()) 

dc5_19p <-ggplot(dc19, aes(x = DC5, y = 0, color = brix, label = ava_id)) +
  geom_jitter(alpha = .8) +
  xlim(c(-0.51, .51)) +
  scale_color_viridis_c() +
  theme_minimal() +
  theme(plot.title = element_text(hjust = 0.5),
        axis.text.y = element_blank(),
        axis.title.y = element_blank()) 

dc6_19p <-ggplot(dc19, aes(x = DC6, y = 0, color = brix, label = ava_id)) +
  geom_jitter(alpha = .8) +
  xlim(c(-0.51, .51)) +
  scale_color_viridis_c() +
  theme_minimal() +
  theme(plot.title = element_text(hjust = 0.5),
        axis.text.y = element_blank(),
        axis.title.y = element_blank())

dc7_19p <-ggplot(dc19, aes(x = DC7, y = 0, color = brix, label = ava_id)) +
  geom_jitter(alpha = .8) +
  xlim(c(-0.51, .51)) +
  scale_color_viridis_c() +
  theme_minimal() +
  theme(plot.title = element_text(hjust = 0.5),
        axis.text.y = element_blank(),
        axis.title.y = element_blank()) 

dc8_19p <-ggplot(dc19, aes(x = DC8, y = 0, color = brix, label = ava_id)) +
  geom_jitter(alpha = .8) +
  xlim(c(-0.51, .51)) +
  scale_color_viridis_c() +
  theme_minimal() +
  theme(plot.title = element_text(hjust = 0.5),
        axis.text.y = element_blank(),
        axis.title.y = element_blank()) 

dc9_19p <-ggplot(dc19, aes(x = DC9, y = 0, color = brix, label = ava_id)) +
  geom_jitter(alpha = .8) +
  xlim(c(-0.51, .51)) +
  scale_color_viridis_c() +
  theme_minimal() +
  theme(plot.title = element_text(hjust = 0.5),
        axis.text.y = element_blank(),
        axis.title.y = element_blank()) 

dc10_19p <-ggplot(dc19, aes(x = DC10, y = 0, color = brix, label = ava_id)) +
  geom_jitter(alpha = .8) +
  xlim(c(-0.51, .51)) +
  scale_color_viridis_c() +
  theme_minimal() +
  theme(plot.title = element_text(hjust = 0.5),
        axis.text.y = element_blank(),
        axis.title.y = element_blank()) 
all_dc_19 <- ggarrange(dc1_19p, dc2_19p, dc3_19p, dc4_19p, dc5_19p, dc6_19p,
                       dc7_19p, dc8_19p, dc9_19p, dc10_19p, ncol = 1, legend = "bottom",
                       common.legend = T, labels = c("B", "", "", "", "", "", "", "", "", ""))
```

```{r all_dc19_hour}

dc1_19p <- ggplot(dc19, aes(x = DC1, y = 0, color = factor(hr, levels = c("0", "2", "6", "16", "64", "112")), label = ava_id)) +
  geom_jitter(alpha = .8) +
  xlim(c(-0.51, .51)) +
  scale_color_brewer(palette = "Paired", name = "hour")  +
  theme_minimal() +
  theme(plot.title = element_text(hjust = 0.5),
        axis.text.y = element_blank(),
        axis.title.y = element_blank()) 

dc2_19p <- ggplot(dc19, aes(x = DC2, y = 0, color =  factor(hr, levels = c("0", "2", "6", "16", "64", "112")), label = ava_id)) +
  geom_jitter(alpha = .8) +
  scale_color_brewer(palette = "Paired", name = "hour")  +
  xlim(c(-0.51, .51)) +
  theme_minimal() +
  theme(plot.title = element_text(hjust = 0.5),
        axis.text.y = element_blank(),
        axis.title.y = element_blank()) 

dc3_19p <- ggplot(dc19, aes(x = DC3, y = 0, color = factor(hr, levels = c("0", "2", "6", "16", "64", "112")), label = ava_id)) +
  geom_jitter(alpha = .8) +
  scale_color_brewer(palette = "Paired", name = "hour")  +
  xlim(c(-0.51, .51)) +
  theme_minimal() +
  theme(plot.title = element_text(hjust = 0.5),
        axis.text.y = element_blank(),
        axis.title.y = element_blank())

dc4_19p <- ggplot(dc19, aes(x = DC4, y = 0, color = factor(hr, levels = c("0", "2", "6", "16", "64", "112")), label = ava_id)) +
  geom_jitter(alpha = .8) +
  scale_color_brewer(palette = "Paired", name = "hour")  +
  xlim(c(-0.51, .51)) +
  theme_minimal() +
  theme(plot.title = element_text(hjust = 0.5),
        axis.text.y = element_blank(),
        axis.title.y = element_blank()) 

dc5_19p <-ggplot(dc19, aes(x = DC5, y = 0, color = factor(hr, levels = c("0", "2", "6", "16", "64", "112")), label = ava_id)) +
  geom_jitter(alpha = .8) +
  xlim(c(-0.51, .51)) +
  scale_color_brewer(palette = "Paired", name = "hour")  +
  theme_minimal() +
  theme(plot.title = element_text(hjust = 0.5),
        axis.text.y = element_blank(),
        axis.title.y = element_blank()) 

dc6_19p <-ggplot(dc19, aes(x = DC6, y = 0, color = factor(hr, levels = c("0", "2", "6", "16", "64", "112")), label = ava_id)) +
  geom_jitter(alpha = .8) +
  xlim(c(-0.51, .51)) +
  scale_color_brewer(palette = "Paired", name = "hour")  +
  theme_minimal() +
  theme(plot.title = element_text(hjust = 0.5),
        axis.text.y = element_blank(),
        axis.title.y = element_blank())

dc7_19p <-ggplot(dc19, aes(x = DC7, y = 0, color = factor(hr, levels = c("0", "2", "6", "16", "64", "112")), label = ava_id)) +
  geom_jitter(alpha = .8) +
  xlim(c(-0.51, .51)) +
  scale_color_brewer(palette = "Paired", name = "hour")  +
  theme_minimal() +
  theme(plot.title = element_text(hjust = 0.5),
        axis.text.y = element_blank(),
        axis.title.y = element_blank()) 

dc8_19p <-ggplot(dc19, aes(x = DC8, y = 0, color = factor(hr, levels = c("0", "2", "6", "16", "64", "112")), label = ava_id)) +
  geom_jitter(alpha = .8) +
  xlim(c(-0.51, .51)) +
  scale_color_brewer(palette = "Paired", name = "hour")  +
  theme_minimal() +
  theme(plot.title = element_text(hjust = 0.5),
        axis.text.y = element_blank(),
        axis.title.y = element_blank()) 

all_dc_19_hr <- ggarrange(dc1_19p, dc2_19p, dc3_19p, dc4_19p, dc5_19p, dc6_19p,
                       dc7_19p, dc8_19p, ncol = 1, legend = "bottom",
                       common.legend = T, labels = c("B", "", "", "", "", "", "", ""))
```

```{r all_dc19_ava}
dc1_19p <- ggplot(dc19, aes(x = DC1, y = 0, color = ava, label = ava_id)) +
  geom_jitter(alpha = .8) +
  xlim(c(-0.51, .51)) +
  theme_minimal() +
  theme(plot.title = element_text(hjust = 0.5),
        axis.text.y = element_blank(),
        axis.title.y = element_blank()) +
  scale_color_manual(values = c("OR"=or,  "AV" = av, "SNC" = snc, "RRV" = rrv, "CRN" = crn,
          "AS" = as, "SMV" = smv, "SRH" = srh))

dc2_19p <- ggplot(dc19, aes(x = DC2, y = 0, color = ava, label = ava_id)) +
  geom_jitter(alpha = .8) +
  scale_color_manual(values = c("OR"=or,  "AV" = av, "SNC" = snc, "RRV" = rrv, "CRN" = crn,
                                "AS" = as, "SMV" = smv, "SRH" = srh)) +
  xlim(c(-0.51, .51)) +
  theme_minimal() +
  theme(plot.title = element_text(hjust = 0.5),
        axis.text.y = element_blank(),
        axis.title.y = element_blank()) 

dc3_19p <- ggplot(dc19, aes(x = DC3, y = 0, color = ava, label = ava_id)) +
  geom_jitter(alpha = .8) +
  scale_color_manual(values = c("OR"=or,  "AV" = av, "SNC" = snc, "RRV" = rrv, "CRN" = crn,
                                "AS" = as, "SMV" = smv, "SRH" = srh)) +
  xlim(c(-0.51, .51)) +
  theme_minimal() +
  theme(plot.title = element_text(hjust = 0.5),
        axis.text.y = element_blank(),
        axis.title.y = element_blank())

dc4_19p <- ggplot(dc19, aes(x = DC4, y = 0, color = ava, label = ava_id)) +
  geom_jitter(alpha = .8) +
  scale_color_manual(values = c("OR"=or,  "AV" = av, "SNC" = snc, "RRV" = rrv, "CRN" = crn,
                                "AS" = as, "SMV" = smv, "SRH" = srh)) +
  xlim(c(-0.51, .51)) +
  theme_minimal() +
  theme(plot.title = element_text(hjust = 0.5),
        axis.text.y = element_blank(),
        axis.title.y = element_blank()) 

dc5_19p <-ggplot(dc19, aes(x = DC5, y = 0, color = ava, label = ava_id)) +
  geom_jitter(alpha = .8) +
  xlim(c(-0.51, .51)) +
  scale_color_manual(values = c("OR"=or,  "AV" = av, "SNC" = snc, "RRV" = rrv, "CRN" = crn,
                                "AS" = as, "SMV" = smv, "SRH" = srh)) +
  theme_minimal() +
  theme(plot.title = element_text(hjust = 0.5),
        axis.text.y = element_blank(),
        axis.title.y = element_blank()) 

dc6_19p <-ggplot(dc19, aes(x = DC6, y = 0, color = ava, label = ava_id)) +
  geom_jitter(alpha = .8) +
  xlim(c(-0.51, .51)) +
  scale_color_manual(values = c("OR"=or,  "AV" = av, "SNC" = snc, "RRV" = rrv, "CRN" = crn,
                                "AS" = as, "SMV" = smv, "SRH" = srh)) +
  theme_minimal() +
  theme(plot.title = element_text(hjust = 0.5),
        axis.text.y = element_blank(),
        axis.title.y = element_blank())

dc7_19p <-ggplot(dc19, aes(x = DC7, y = 0, color = ava, label = ava_id)) +
  geom_jitter(alpha = .8) +
  xlim(c(-0.51, .51)) +
  scale_color_manual(values = c("OR"=or,  "AV" = av, "SNC" = snc, "RRV" = rrv, "CRN" = crn,
                                "AS" = as, "SMV" = smv, "SRH" = srh)) +
  theme_minimal() +
  theme(plot.title = element_text(hjust = 0.5),
        axis.text.y = element_blank(),
        axis.title.y = element_blank()) 

dc8_19p <-ggplot(dc19, aes(x = DC8, y = 0, color = ava, label = ava_id)) +
  geom_jitter(alpha = .8) +
  xlim(c(-0.51, .51)) +
  scale_color_manual(values = c("OR"=or,  "AV" = av, "SNC" = snc, "RRV" = rrv, "CRN" = crn,
                                "AS" = as, "SMV" = smv, "SRH" = srh)) +
  theme_minimal() +
  theme(plot.title = element_text(hjust = 0.5),
        axis.text.y = element_blank(),
        axis.title.y = element_blank()) 

all_dc_19_ava <- ggarrange(dc1_19p, dc2_19p, dc3_19p, dc4_19p, dc5_19p, dc6_19p,
                           dc7_19p, dc8_19p, ncol = 1, legend = "bottom",
                           common.legend = T, labels = c("A", "", "", "", "", "", "", ""))
```

```{r allDCs2019, fig.height= 7.5, fig.cap = "One-dimensional plots of diffusion components 1-10 for the 2019 vintage colored by **A** AVA and **B** hour. Diffusion components capture different latent structure in the dataset. Each component represents a different relationship among samples."}
ggarrange(all_dc_19_ava, all_dc_19_hr, legend = "bottom")
```

In general, the number of differentially expressed genes diminished across diffusion components indicating increasingly subtle differences between samples (**Table  \@ref(tab:numdiffex)**).
This, combined with lower diffusion components capturing more structure across samples, allowed us to see what stages of fermentation undergo the most gene expression remodeling. 
In the 2019 vintage, we found metabolic remodeling during early fermentation accounted for more variation than at the end of fermentation given that diffusion components 2 and 3 differentiated early fermentation samples while diffusion component 4, 6, and 8 differentiated late fermentation samples (**Figure \@ref(fig:allDCs2019)**).


```{r numdiffex}
num_diffex <- data.frame(diffusion_component = c("DC1", "DC2", "DC3", "DC4", 
                                                 "DC5", "DC6", "DC7", "DC8"),
                         induced = c(length(filter(dc1_17, logFC > 0)$ORF),
                                     length(filter(dc2_17, logFC > 0)$ORF),
                                     length(filter(dc3_17, logFC > 0)$ORF),
                                     length(filter(dc4_17, logFC > 0)$ORF),
                                     length(filter(dc5_17, logFC > 0)$ORF),
                                     length(filter(dc6_17, logFC > 0)$ORF),
                                     length(filter(dc7_17, logFC > 0)$ORF),
                                     length(filter(dc8_17, logFC > 0)$ORF)),
                         repressed = c(length(filter(dc1_17, logFC < 0)$ORF),
                                       length(filter(dc2_17, logFC < 0)$ORF),
                                       length(filter(dc3_17, logFC < 0)$ORF),
                                       length(filter(dc4_17, logFC < 0)$ORF),
                                       length(filter(dc5_17, logFC < 0)$ORF),
                                       length(filter(dc6_17, logFC < 0)$ORF),
                                       length(filter(dc7_17, logFC < 0)$ORF),
                                       length(filter(dc8_17, logFC < 0)$ORF)),
                         induced = c(length(filter(dc1_19, logFC > 0)$ORF),
                                     length(filter(dc2_19, logFC > 0)$ORF),
                                     length(filter(dc3_19, logFC > 0)$ORF),
                                     length(filter(dc4_19, logFC > 0)$ORF),
                                     length(filter(dc5_19, logFC > 0)$ORF),
                                     length(filter(dc6_19, logFC > 0)$ORF),
                                     length(filter(dc7_19, logFC > 0)$ORF),
                                     length(filter(dc8_19, logFC > 0)$ORF)),
                         repressed = c(length(filter(dc1_19, logFC < 0)$ORF),
                                       length(filter(dc2_19, logFC < 0)$ORF),
                                       length(filter(dc3_19, logFC < 0)$ORF),
                                       length(filter(dc4_19, logFC < 0)$ORF),
                                       length(filter(dc5_19, logFC < 0)$ORF),
                                       length(filter(dc6_19, logFC < 0)$ORF),
                                       length(filter(dc7_19, logFC < 0)$ORF),
                                       length(filter(dc8_19, logFC < 0)$ORF))
)

kable(num_diffex, "latex", booktabs = T, col.names = c("diffusion component", "induced", "repressed", "induced", "repressed"),
      caption = 'Number of significantly differentially expressed genes for the top diffusion components for the 2017 and 2019 vintages') %>%
  kable_styling(font_size = 9) %>%
  add_header_above(c(" " = 1, "2017" = 2, "2019" = 2), align = "l", escape = F)
```

Diffusion component two captured the transition from 2 to 6 hours of fermentation, while the diffusion component 3 capture the transition from 6 to 16 hours of fermentation.
Along diffusion component 2, arginine biosynthetic process was enriched in genes that were induced in the 2 hour samples (*ARG1*, *ARG3*, *ARG5,6*, *ARG8*) (**Figure \@ref(fig:annotatedplts19hr)**, **Figure \@ref(fig:DC2viz2019)**).
Arginine is one of the most abundant amino acids during fermentation and is used heavily in protein biosynthesis [@crepin2017management].
Arginine consumption was found to provide all anabolic requirements for *S. cerevisiae* during wine fermentation [@rollero2019comparison], so induction of these biosynthetic genes in early fermentation may be involved in early cellular adaptation to the wine environment.
Interestingly, four of the 16 genes induced in 2 hour samples have no known function.
Given the few genes induced in 2 hour samples and enrichment of arginine biosynthesis, these genes may play a related role in arginine metabolism.

```{r v2019dc2dc3, fig.height = 4, fig.width =5, fig.cap = "Metabolic transitions in early and late fermentation separate samples. **A, B** Diffusion components 2 and 3 capture variation in early fermentation. While late-fermentation samples cluster at the origin, early fermentation samples form three clusters. **C, D** Diffusion components 4 and 6 capture variation in late fermentation. While early fermentation samples cluster at the origin, late fermentation samples separate by time in fermentation and vineyard site region."}
by_hour <- ggplot(dc19, aes(x = DC2, y = DC3, color = factor(hr, levels = c("0", "2", "6", "16", "64", "112")), label = ava)) +
  geom_point(alpha = .6, size = 2) +
  scale_color_brewer(palette = "Paired", name = "hour") + 
  theme_minimal() +
  theme(plot.title = element_text(hjust = 0.5)) 

by_ava <- ggplot(dc19, aes(x = DC2, y = DC3, color = ava, label = hr)) +
  geom_point(alpha = .6, size = 2) +
  scale_color_manual(values = c("OR"=or,  "AV" = av, "SNC" = snc, "RRV" = rrv, "CRN" = crn,
                                "AS" = as, "SMV" = smv, "SRH" = srh)) + 
  theme_minimal() +
  theme(plot.title = element_text(hjust = 0.5)) 

by_hour_46 <- ggplot(dc19, aes(x = DC4, y = DC6, color = factor(hr, levels = c("0", "2", "6", "16", "64", "112")), label = ava)) +
  geom_point(alpha = .6, size = 2) +
  scale_color_brewer(palette = "Paired", name = "hour") +
  theme_minimal() +
  theme(plot.title = element_text(hjust = 0.5))

by_ava_46 <- ggplot(dc19, aes(x = DC4, y = DC6, color = ava, label = ava_id)) +
  geom_point(alpha = .6, size = 2) +
  theme_minimal() +
  scale_color_manual(values = c("OR"=or,  "AV" = av, "SNC" = snc, "RRV" = rrv, "CRN" = crn,
                                "AS" = as, "SMV" = smv, "SRH" = srh)) +
  theme(plot.title = element_text(hjust = 0.5))


ggarrange(ggarrange(by_hour, by_hour_46, labels = c("A", "C"), legend = "bottom",
                    common.legend = T, nrow = 2),
          ggarrange(by_ava, by_ava_46, labels = c("B", "D"), legend = "bottom",
                    common.legend = T, nrow = 2), ncol = 2)

```


By 6 hours, gene expression supports a transition to anaerobic metabolism.
We detected induction of anaerobic translation elongation factor eiF-5A, *ANB1* (**Figure \@ref(fig:annotatedplts19hr)**).
*ANB1* is optimally expressed below 0.5 umol/L O~2~ [@kwast1998oxygen], likely indicating a drop in fermentation oxygen by this time.
Glycolysis/gluconeogenesis was enriched among genes induced in the six-hour samples (**Figure \@ref(fig:DC2viz2019)**), potentially indicating that carbon dioxide production began shielding fermentation from oxygen by 6 hours after inoculation.
Genes important for cell wall processes were also induced at 6 hours; indeed, *TIR1-4*, which encode cell wall mannoproteins [@abramova2001reciprocal], were four of the top five genes induced in 6 hour samples (**Figure \@ref(fig:annotatedplts19hr)**).
*TIR1*, *TIR3*, and *TIR4* are required for anaerobic growth [@abramova2001reciprocal].

Diffusion component 3 captured the transition from 6 to 16 hours of fermentation.
We observed continued induction of cell wall remodeling proteins, suggesting these processes play a pivotal role throughout early fermentation (**Figure \@ref(fig:annotatedplts19hr)**).
In addition to *TIR1-4*, *DAN1* was induced in 16 hour samples along with many *PAU* genes (*PAU2*, *PAU3*, *PAU4*, *PAU5*, *PAU7*, *PAU8*, *PAU10*, *PAU11*, *PAU12*, *PAU15*, *PAU16*, *PAU17*, *PAU19*, *PAU20*, *PAU23*, *PAU24*).
Many biological processes, cellular compartments, and molecular functions were enriched in the 293 genes that were induced in the 16 hour samples (**Figure \@ref(fig:DC3viz2019)**), which is consistent with the early log phase of growth.
Entry into log phase of growth is likely supported by ribosomal remodeling that begins at 6 hours after inoculation.
Of 32 genes induced at 6 hours compared to 16 hours, 13 were involved in ribosomal processes.

Diffusion component 4 separated the 64 hour samples from the 112 hour samples.
This separation was likely driven by a decreased availability of preferred nutrients at 112 hours.
In the 64 hour samples, transmembrane transport, including amino acid and polyamine transport, were enriched in genes that were induced in the 64 hour samples (**Figure \@ref(fig:DC4viz2019)**). 
Two of the top induced genes (*DUR3* and *DAL5*), are involved in allantoin metabolism, a non-preferred nitrogen source (**Figure \@ref(fig:annotatedplts19hr)**).
Induction of these genes at 64 hours likely indicates relief of nitrogen could tablet light repression consistent with decreasing nitrogen concentrations.
Conversely, *HXT13* and *MAN2*  for the top induced genes in the 112 our samples, along with *HXT17* (**Figure \@ref(fig:annotatedplts19hr)**). 
These *HXT* genes are mannitol transporters [@Jordan2016] while and *MAN2* encodes mannitol dehydrogenase. 
Mannitol is produced by fructophilic acetic acid bacteria like *Lactobacillus kunkeii* [@bisson2017two], and expression of these genes likely signals that mannitol was in the media and that no preferred sugars were left.

Unlike the 2019 vintage, we did not observe clear delineations among samples taken at the same time in late fermentation in the 2017 vintage (**Figure \@ref(fig:allDCs2017)**).
While we detected dominant drivers of variation among samples (e.g. global shift in gene expression during primary fermentation), we think that lack of grouping among samples may indicate a data quality issue.
All gene expression data are inherently noisey [@hansen2011sequencing].
With advances in sequencing technology, we sequenced the 2019 vintage with unique molecular identifier (UMI) barcodes and thus were better able to demultiplex sequencing lanes and remove PCR duplicates. 
While diffusion mapping faithfully recapitulates the dominant metabolic processes in fermentation, we are skeptical that noise in the data interferes with our ability to draw further conclusions from higher diffusion components in the 2017 vintage.

```{r all_dc17_brix}
dc1_17p <- ggplot(dc17, aes(x = DC1, y = 0, color = brix, label = ava_id)) +
  geom_jitter(alpha = .8) +
  xlim(c(-0.3, .3)) +
  scale_color_viridis_c() +
  theme_minimal() +
  theme(plot.title = element_text(hjust = 0.5),
        axis.text.y = element_blank(),
        axis.title.y = element_blank()) 

dc2_17p <- ggplot(dc17, aes(x = DC2, y = 0, color = brix, label = ava_id)) +
  geom_jitter(alpha = .8) +
  scale_color_viridis_c() +
  xlim(c(-0.3, .3)) +
  theme_minimal() +
  theme(plot.title = element_text(hjust = 0.5),
        axis.text.y = element_blank(),
        axis.title.y = element_blank()) 

all_dc_17 <- ggarrange(dc1_17p, dc2_17p, legend = "bottom",
                       ncol = 1, common.legend = T, labels = c("B", ""))
```

```{r all_dc17_ava}
dc1_17p <- ggplot(dc17, aes(x = DC1, y = 0, color = ava, label = ava_id)) +
  geom_jitter(alpha = .8) +
  scale_color_manual(values = c("OR"=or,  "AV" = av, "SNC" = snc, "RRV" = rrv, "CRN" = crn,
                                "AS" = as, "SMV" = smv, "SRH" = srh)) +
  xlim(c(-0.3, .3)) +
  theme_minimal() +
  theme(plot.title = element_text(hjust = 0.5),
        axis.text.y = element_blank(),
        axis.title.y = element_blank()) 

dc2_17p <- ggplot(dc17, aes(x = DC2, y = 0, color = ava, label = ava_id)) +
  geom_jitter(alpha = .8) +
  scale_color_manual(values = c("OR"=or,  "AV" = av, "SNC" = snc, "RRV" = rrv, "CRN" = crn,
                                "AS" = as, "SMV" = smv, "SRH" = srh)) +
  xlim(c(-0.3, .3)) +
  theme_minimal() +
  theme(plot.title = element_text(hjust = 0.5),
        axis.text.y = element_blank(),
        axis.title.y = element_blank()) 

all_dc_17_ava <- ggarrange(dc1_17p, dc2_17p, legend = "bottom",
                           ncol = 1, common.legend = T, labels = c("A", ""))
```

```{r allDCs2017, fig.height= 3, fig.cap = "One-dimensional plots of diffusion components 1 and 2 for the 2017 vintage colored by **A** AVA and **B** brix. Diffusion components capture different latent structure in the data set. Unlike in the 2019 vintage, samples in lower diffusion components do not cluster, potentially indicating a data quality issue that was remediated in the 2019 vintage through the use of UMI barcodes."}
ggarrange(all_dc_17_ava, all_dc_17, legend = "bottom")
```

## Vineyard site differentiates samples within fermentation stages

Along diffusion components that separated stages of fermentation, we detected outliers that may indicate fundamentally different metabolic strategies during those stages of fermentation.

In the 2017 vintage, samples taken at 40 hours from fermentations from SMV1, SMV2, and OR1 were shifted toward the 16 hour samples indicating a different metabolic strategy in fermentations from these vineyards (**Figure \@ref(fig:allDCs2017)**).
Expression of genes involved in mannitol at the end of fermentation was highest for the SMV sites, likely indicating higher concentrations of mannitol in these fermentations.
This, combined with an altered metabolic program in early fermentation, may be a signature of [GAR+] prion state induced by acetic acid produced by *L. kunkeei* [@bisson2017two] (CITE BRIX PAPER).
We were not able to test for this prion state in the 2017 vintage and did not detect it in the 2019 vintage (CITE BRIX PAPER), so altered core metabolic trajectory may be caused by other factors.

In the 2019 vintage, 6 hour, and to a lesser extent 2 hour, samples from SRH1 do not cluster with other samples from these time points (**Figure \@ref(fig:v2019dc2dc3)**).
The SRH1 6 hour samples were shifted toward other 16 hour samples, potentially indicating faster cellular adaptation to the wine environment.
In support of this, fermentations from SRH1 initially decreased in brix faster than other sites in both the 2017 and 2019 vintages.
In the future, closer inspection of the initial chemical and microbiological state of SRH1 grape musts relative to grape musts from other sites may reveal the differences that lead to faster onset of fermentation.

We also observed outliers in the fermentation.
Along diffusion component 4, 64 hour samples from OR1 and OR2 were shifted toward the 112 hour cluster (**Figure \@ref(fig:v2019dc2dc3)**), indicating attenuated expression of genes induced in other 64 hour samples.
Indeed, while expression of some genes at 64 hours matches that of other fermentations (e.g. *DUR3*, *DAL5* involved in metabolism of alternative nitrogen source allantoin), expression of some genes (e.g. *THI72*, *THI80*, involved in thiamine metabolism) was lower than for many samples at the extrema of diffusion component 4 (**Figure \@ref(fig:or1)**).
This may indicate shared nutrient limitation that produces gene expression patterns similar to the end of fermentation in fermentations from the OR region.
Similarly, 112 hour samples from SNC1 and AS2 were shifted toward the 64 hour cluster (**Figure \@ref(fig:v2019dc2dc3)**).

```{r or1, fig.cap = "Samples from OR1 and OR2 taken at 64 hours post-inoculation do not cluster along diffusion component 4 with other 64 hour samples. While expression of some genes like *DUR3* and *DAL5* is similar between OR samples and other samples, expression of other genes like *THI72* and *THI80* is relatively lower in OR samples."}
gene <- 'YOR143C' # thi80
Y <- dc_y19$E[gene, ]
dc19$Y = Y
thi80 <- ggplot(dc19, aes(y = Y, x = DC4, color = ava)) +
  geom_smooth(method='lm', color = "grey", se = F, alpha = .5) +
  geom_point(alpha = .9) +
  ggtitle("THI80") +
  theme_minimal() +
  labs(y = "norm log expression value") +
  theme(legend.position = "bottom",
        axis.title.y = element_blank(), 
        legend.title = element_text(face = "bold"),
        axis.text.x = element_text(angle = 90, vjust = .5),
        plot.title = element_text(size=12, face = "bold.italic", hjust = .5)) +
  scale_color_manual(values = c(SRH = srh, SMV = smv,  
                               AS = as, SNC = snc,  CRN = crn, 
                               RRV = rrv, AV = av, OR = or),
                    breaks = c('OR', 'AV', 'SNC', 'RRV', 'CRN', 'AS', 'SMV', 'SRH'), 
                    name = "AVA")

gene <- 'YOR192C' # thi72
Y <- dc_y19$E[gene, ]
dc19$Y = Y
thi72 <- ggplot(dc19, aes(y = Y, x = DC4, color = ava)) +
  geom_smooth(method='lm', color = "grey", se = F, alpha = .5) +
  geom_point(alpha = .9) +
  ggtitle("THI72") +
  theme_minimal() +
  labs(y = " ") +
  theme(legend.position = "bottom",
        legend.title = element_text(face = "bold"),
        axis.text.x = element_text(angle = 90, vjust = .5),
        plot.title = element_text(size=12, face = "bold.italic", hjust = .5)) +
  scale_color_manual(values = c(SRH = srh, SMV = smv,  
                               AS = as, SNC = snc,  CRN = crn, 
                               RRV = rrv, AV = av, OR = or),
                    breaks = c('OR', 'AV', 'SNC', 'RRV', 'CRN', 'AS', 'SMV', 'SRH'), 
                    name = "AVA")

gene <- 'YHL016C' # DUR3
Y <- dc_y19$E[gene, ]
dc19$Y = Y
dur3 <- ggplot(dc19, aes(y = Y, x = DC4, color = ava)) +
  geom_smooth(method='lm', color = "grey", se = F, alpha = .5) +
  geom_point(alpha = .9) +
  ggtitle("DUR3") +
  theme_minimal() +
  labs(y = "norm log expression value") +
  theme(legend.position = "bottom",
        legend.title = element_text(face = "bold"),
        axis.title.x = element_blank(),
        axis.text.x = element_blank(),
        axis.title.y = element_text(hjust = 4),
        plot.title = element_text(size=12, face = "bold.italic", hjust = .5)) +
  scale_color_manual(values = c(SRH = srh, SMV = smv,  
                               AS = as, SNC = snc,  CRN = crn, 
                               RRV = rrv, AV = av, OR = or),
                    breaks = c('OR', 'AV', 'SNC', 'RRV', 'CRN', 'AS', 'SMV', 'SRH'), 
                    name = "AVA")
gene <- 'YJR152W' # DAL5
Y <- dc_y19$E[gene, ]
dc19$Y = Y
dal5 <- ggplot(dc19, aes(y = Y, x = DC4, color = ava)) +
  geom_smooth(method='lm', color = "grey", se = F, alpha = .5) +
  geom_point(alpha = .9) +
  ggtitle("DAL5") +
  theme_minimal() +
  labs(y = "norm log expression value") +
  theme(legend.position = "bottom",
        legend.title = element_text(face = "bold"),
        axis.title.y = element_blank(),
        axis.title.x = element_blank(),
        axis.text.x = element_blank(),
        plot.title = element_text(size=12, face = "bold.italic", hjust = .5)) +
  scale_color_manual(values = c(SRH = srh, SMV = smv,  
                               AS = as, SNC = snc,  CRN = crn, 
                               RRV = rrv, AV = av, OR = or),
                    breaks = c('OR', 'AV', 'SNC', 'RRV', 'CRN', 'AS', 'SMV', 'SRH'), 
                    name = "AVA")

ggarrange(dur3, dal5, thi72, thi80, ncol = 2, nrow = 2, common.legend = T, legend = "bottom")
```

The extremes of each diffusion component represent samples with the most differentiated transcriptome for genes in that component.
Given that some higher diffusion components separated samples within a stage of fermentation, we used this information to determine vineyard-specific differences within a stage of fermentation.
In this way, we identified vineyard-specific gene expression patterns.

Samples taken at 2 hours separated along diffusion component 5 with samples from SMV1, SMV2, SRH1, AV2, and RRV3 separating to one extrema and samples from our RRV2, CRN1, SNC1, and AS2 separating to the other.
While no genes were significantly induced among the latter group, 13 were induced in the former (**Table \@ref(tab:numdiffex)**).
This set was enriched for vitamin metabolic process and cell wall (**Figure \@ref(fig:DC5viz2019)**).
Given that co-culture experiments of wine fermentations have demonstrated that *S. cerevisiae* induces genes involved in cell wall remodeling and vitamin biosynthesis in response to the presence of non-*Saccharomyces* yeasts [@barbosa2015genomic; @kosel2017influence; @curiel2017different], we tested whether gene expression by non-*Saccharomyces* yeasts in 2 hour samples correlated with diffusion component 5.
Total gene expression of *Hanseniaspora uvarum* correlated with diffusion component 5 (R^2^ = 0.49, p < 0.001) (**Figure \@ref(fig:DC2uvarum)**), suggesting that interactions with *H. uvarum* in early fermentation altered *S. cerevisiae* metabolism in some fermentations.
This is consistent with the previous study that found that *S. cerevisiae* remodels its cell wall in the presence of *H. uvarum* at three hours post-inoculation in wine fermentation [@curiel2017different].
Interestingly, *PDC5* was also induced in fermentations with higher gene expression by *H. uvarum* (**Figure \@ref(fig:annotatedplts19ava)**).
*PDC5* encodes one of three isoforms of pyruvate decarboxylase, an enzyme involved in the formation of flavor–active higher alcohols in wine via the Ehrlich pathway [@ter1998pyruvate].
This suggests that the presence of *H. uvarum* may impact wine sensory outcomes.
 
```{r DC2uvarum, fig.cap = "Total gene expression by *H. uvarum* at 2 hours post-inoculation of *S. cerevisiae* correlated with diffusion component 5 values for 2 hour samples. Diffusion components represented composites of *S. cerevisiae* gene expression that capture differences between samples."}
dc19_2hr <- read_csv("samples_ava.csv") %>%
  filter(!is.na(tank)) %>%
  filter(hr == 2) %>%
  dplyr::select(id, hr, year, tank, ava_id) %>%
  left_join(dc19)

## 2019
v19 <- read_tsv("v2019/outputs/counts_all_organisms/raw_counts.tsv") %>%
  filter(!gene %in% c("__no_feature", "__ambiguous", "__too_low_aQual", "__not_aligned", "__alignment_not_unique")) 
colnames(v19) <- gsub("outputs\\/htseq_all_organisms\\/", "", colnames(v19))
colnames(v19) <- gsub("_readcounts\\.txt", "", colnames(v19))
v19 <- dplyr::select(v19, c(colnames(v19)[colnames(v19) %in% dc19_2hr$id], gene))
v19 <- v19[rowSums(v19[-which(names(v19) %in% "gene")]) > 0, ] # remove rows that sum to zero
v19 <- column_to_rownames(v19, "gene") # move gene to rownames

# normalize counts
v0_19 <- DGEList(v19)              # initialize the DGE object
v19 <-calcNormFactors(v0_19)       # calculate normalization factors
v_mm19 <- model.matrix(~dc19_2hr$DC1 + ~dc19_2hr$DC2 + ~dc19_2hr$DC3 + dc19_2hr$DC4 + 
                         dc19_2hr$DC5 + dc19_2hr$DC6 + dc19_2hr$DC7 + ~dc19_2hr$DC8 + 
                         dc19_2hr$DC9 + dc19_2hr$DC10)
v_y19 <- voom(v19, v_mm19, plot = F)
# v_y19$E contains normalized counts


v19 <- v_y19$E %>%
  as.data.frame() %>%
  rownames_to_column("gene") %>%
  mutate(organism = gsub("_.*", "", gene)) %>%
  mutate(organism = gsub("KLTH0.*", "KLTH0", organism)) %>%
  mutate(organism = gsub("PEGC.*", "PEGC", organism)) %>%
  mutate(organism = gsub("MWSF.*", "MWSF", organism)) %>%
  mutate(organism = gsub("^Y.*", "scer", organism)) %>%
  mutate(organism = gsub("sn.*", "scer", organism)) %>%
  mutate(organism = gsub("LPNK.*", "LPNK", organism)) %>%
  mutate(organism = gsub("^t.*", "scer", organism)) %>%
  mutate(organism = gsub("RPR1", "scer", organism)) %>%
  mutate(organism = gsub("SCR1", "scer", organism)) %>%
  mutate(organism = gsub("TLC1", "scer", organism)) %>%
  mutate(organism = gsub("LSR1", "scer", organism)) %>%
  mutate(organism = gsub("PWR1", "scer", organism)) %>%
  mutate(organism = gsub("NME1", "scer", organism)) %>%
  mutate(organism = gsub("ICR1", "scer", organism)) %>%
  mutate(organism = gsub("RUF23", "scer", organism)) %>%
  mutate(organism = gsub("RNA170", "scer", organism)) %>%
  mutate(organism = gsub("ZOD1", "scer", organism)) %>%
  mutate(organism = gsub("IRT1", "scer", organism)) %>%
  mutate(organism = gsub("RUF20", "scer", organism)) %>%
  mutate(organism = gsub("RUF21", "scer", organism)) %>%
  mutate(organism = gsub("RUF22", "scer", organism)) %>%
  mutate(organism = gsub("RUF5-2", "scer", organism)) %>%
  mutate(organism = gsub("SRG1", "scer", organism)) %>%
  mutate(organism = gsub("RME2", "scer", organism)) %>%
  mutate(organism = gsub("RME3", "scer", organism)) %>%
  mutate(organism = gsub("RDN5-1", "scer", organism)) %>%
  mutate(organism = gsub("RDN37-2", "scer", organism)) %>%
  mutate(organism = gsub("HRA1", "scer", organism)) %>%
  mutate(organism = gsub("KLTH0", "Lachancea_thermotolerans", organism)) %>%
  mutate(organism = gsub("PEGC", "Cladosporium_sp_SL_16", organism)) %>%
  mutate(organism = gsub("MWSF", "Starmerella_bacillaris", organism)) %>%
  mutate(organism = gsub("LPNK", "Metschnikowia_sp_AWRI3582", organism)) %>%
  mutate(organism = gsub("scer", "Saccharomyces_cerevisiae", organism)) %>%
  mutate(organism = gsub("BCIN", "Botrytis_cinerea", organism)) %>%
  mutate(organism = gsub("BcDW1", "Botrytis_cinerea", organism)) %>%
  mutate(organism = gsub("VIT", "Vitis_vinifera", organism)) %>%
  mutate(organism = gsub("CU098", "Rhizopus_stolonifer", organism)) %>%
  mutate(organism = gsub("metfru2", "Metschnikowia_fructicola", organism)) %>%
  mutate(organism = gsub("M438DRAFT", "Aureobasidium_pullulans", organism)) %>%
  mutate(organism = gsub("AWRI3578", "Hanseniaspora_opuntiae", organism)) %>%
  mutate(organism = gsub("C5L36", "Pichia_kudriavzevii", organism)) %>%
  mutate(organism = gsub("D499", "Hanseniaspora_uvarum", organism)) %>%
  mutate(organism = gsub("D6D18", "Aureobasidium_pullulans", organism))

dc19_w_v19 <- v19 %>%
  pivot_longer(cols = starts_with("2019"), names_to = "sample", values_to = "counts") %>%
  group_by(sample, organism) %>%
  summarize(total_counts = sum(counts)) %>%
  pivot_wider(id_cols = sample, names_from = organism, values_from = total_counts) %>%
  right_join(dc19, v19, by = "sample")

# linear model:
# summary(lm(DC5 ~ Hanseniaspora_uvarum, data = dc19_w_v19))


ggplot(dc19_w_v19, aes(x = DC5, y =  Hanseniaspora_uvarum, color = ava)) + 
  geom_point(size = 2, alpha = .9) +
  labs(y = expression(italic("Hanseniaspora uvarum"))) +
  theme_minimal()+   
  theme(legend.position = "bottom",
        axis.text.x = element_text(angle = 90, vjust = .5),
        legend.title = element_text(size=12, face ="bold"),
        strip.text.x = element_text(size = 12, face = "bold")) +
  scale_color_manual(values = c(SRH = srh, SMV = smv,  
                               AS = as, SNC = snc,  CRN = crn, 
                               RRV = rrv, AV = av, OR = or),
                    breaks = c('OR', 'AV', 'SNC', 'RRV', 'CRN', 'AS', 'SMV', 'SRH'), 
                    name = "AVA")

# orgs to test:
# Aureobasidium_pullulans   Botrytis_cinerea     
# Cladosporium_sp_SL_16     Hanseniaspora_opuntiae    Hanseniaspora_uvarum     
# Lachancea_thermotolerans  Metschnikowia_fructicola  Metschnikowia_sp_AWRI3582
# Pichia_kudriavzevii       Rhizopus_stolonifer       Saccharomyces_cerevisiae
# ggplot(dc19_w_v19, aes(x = DC5, y = Cladosporium_sp_SL_16)) + 
#   geom_point()
# ggplot(dc19_w_v19, aes(x = DC5, y = Hanseniaspora_opuntiae)) + 
#   geom_point()
# ggplot(dc19_w_v19, aes(x = DC5, y = Lachancea_thermotolerans)) + 
#   geom_point()
# ggplot(dc19_w_v19, aes(x = DC5, y = Saccharomyces_cerevisiae)) + 
#   geom_point()
# ggplot(dc19_w_v19, aes(x = DC5, y =  Metschnikowia_fructicola)) + 
#   geom_point()
# ggplot(dc19_w_v19, aes(x = DC5, y =  Pichia_kudriavzevii)) + 
#   geom_point()
# ggplot(dc19_w_v19, aes(x = DC5, y = Rhizopus_stolonifer)) + 
#   geom_point()
```

Diffusion component 7 predominantly captured separation in 16 hour samples, where unlike at 2 hours, fermentations from SMV1 and SMV2 segregated the opposite extrema from SRH1 (**Figure \@ref(fig:allDCs2019)**).
SMV and SRH are neighboring American Viticulture Areas in Southern California (**Figure \@ref(fig:fig1)A**).
While few genes are induced in SMV samples, *ADH4* is the top induced gene (**Figure \@ref(fig:annotatedplts19ava)**).
Alcohol dehydrogenase plays an important role in fermentation by catabolising acetaldehyde to ethanol and regenerating the redox cofactor NAD+.
However Adh1p is the primary isoform responsible for this reaction during wine fermentation [@ciriacy1975genetics].
This may indicate that formation of fusel alcohols was more prevalent in SMV fermentations via action of Adh4p as the last step in the Ehrlich pathway [@dickinson2003catabolism].

Diffusion component six separated the 64 hour samples, with the OR samples and RRV2 samples segregating to one extrema (**Figure \@ref(fig:annotatedplts19ava)**).
While no pathways were enriched among genes that were induced in these samples, genes associated with nitrogen limitation (*DAL5*, *PUT1*, *PUT2*; [@rossignol2003genome]) where induced while genes involved in ammonia metabolism (*MEP3*, *SSY1*, *AUA1*) were induced at the other extrema (**Figure \@ref(fig:annotatedplts19ava)**).
Diffusion component 6 correlated with initial NOPA and NH~3~ in grape must (initial NOPA: R^2^ = 0.62, p < 0.001, initial NH~3~: R^2^ = 0.60, p < 0.001), driven by low initial nitrogen in OR1, OR2, and RRV2 (**Figure \@ref(fig:DC6nitrogen)**).
Interestingly, while initial nitrogen was low and OR1, OR2, and RRV2, we supplemented primary fermentations with low nitrogen using diammonia phosphate to a target yeast assimilable nitrogen (YAN) level of 250 mg/L at approximately 24 hours after inoculation.
Yet, we still detected signatures of nitrogen limitation from for these sites, suggesting that standard commercial remediations may not be sufficient for some fermentations.
On the opposite extreme from the nitrogen limited fermentations, induced genes were enriched and fatty acid metabolism (*ELO1*, *ELO3*) which may impact wine aroma outcomes [@liu2019comparing].

```{r DC6nitrogen, fig.cap = "Diffusion component 6 separates samples by initial nitrogen concentration in the grape must. **A)** Initial concentration of NH~3~ and NOPA was lower in fermentations from OR1, OR2, and RRV2 than other vineyards. **B, C)** Initial NH~3~ and NOPA in grape must correlates with diffusion component 6, driven by low nitrogen in fermentations from vineyard sites OR1, OR2, RRV3."}
nitrogen <- info %>% 
  filter(!is.na(tank)) %>%
  filter(year == 2019) %>%
  dplyr::select(ava_id, year, ava, initial_nh3, initial_nopa) %>%
  distinct() %>%
  pivot_longer(cols = starts_with("initial"), names_to = "variable", values_to = "values") %>%
  mutate(variable = gsub("initial_nh3", "initial NH3", variable)) %>%
  mutate(variable = gsub("initial_nopa", "initial NOPA", variable)) 

nitrogen$ava_id <- factor(nitrogen$ava_id, 
                          levels = c('OR1', 'OR2', 'AV1', 'AV2',  
                                     'SNC1', 'SNC2', 'RRV1', 'RRV2', 'RRV3',  
                                     'CRN1', 'AS1', 'AS2', 'SMV1', 
                                     'SMV2','SRH1'))
nitrogen <- ggplot(nitrogen, aes(x = ava_id, y = values, fill = ava)) +
  geom_col() +
  facet_wrap(~variable, ncol = 2) +
  labs(x = "vineyard site", y = "concentration") +
  theme_minimal() +
  theme(legend.position = "bottom",
        axis.text.x = element_text(angle = 90, vjust = .5),
        legend.title = element_text(size=12, face ="bold"),
        strip.text.x = element_text(size = 12, face = "bold")) +
  scale_fill_manual(values = c(SRH = srh, SMV = smv,  
                                AS = as, SNC = snc,  CRN = crn, 
                                RRV = rrv, AV = av, OR = or),
                    breaks = c('OR', 'AV', 'SNC',  'RRV', 'CRN', 'AS', 'SMV', 'SRH'), 
                    name = "AVA")


# correlation -----------------
dc19_64 <- dc19 %>%
  filter(hr == 64)

# linear models
# summary(lm(DC6 ~ initial_nopa, dc19_64))
# summary(lm(DC6 ~ initial_nh3, dc19_64))

# correlation results
nh3_corr <- ggplot(dc19_64, aes(x = DC6, y = initial_nh3, color = ava)) + 
  geom_point() + 
  labs(y = "initial NH3") +
  theme_minimal()+
  theme(legend.position = "none") +
  scale_color_manual(values = c(SRH = srh, SMV = smv,  
                               AS = as, SNC = snc,  CRN = crn, 
                               RRV = rrv, AV = av, OR = or),
                    breaks = c('OR', 'AV', 'SNC', 'RRV', 'CRN', 'AS', 'SMV', 'SRH'))

nopa_corr <- ggplot(dc19_64, aes(x = DC6, y = initial_nopa, color = ava)) + 
  geom_point() +
  labs(y = "initial NOPA") +
  theme_minimal() +
  theme(legend.position = "none") +
  scale_color_manual(values = c(SRH = srh, SMV = smv,  
                               AS = as, SNC = snc,  CRN = crn, 
                               RRV = rrv, AV = av, OR = or),
                    breaks = c('OR', 'AV', 'SNC', 'RRV', 'CRN', 'AS', 'SMV', 'SRH'))

ggarrange(nitrogen, ggarrange(nh3_corr, nopa_corr, ncol = 2), nrow = 2, 
          common.legend = T, legend = "bottom")
```

Interestingly, while diffusion component 8 reflected separation among 112 hour samples with SNC1 and AS2 segregating from other samples (**Figure \@ref(fig:v2019dc2dc3)**), 14 of the 24 genes induced in those samples are of unknown function.
Cellular response to hydrogen peroxide was enriched among identifiable genes (*DDR2*, *HSP30*), as well as extracellular region (*UTH1*, *AFB1*, *MFA1*), suggesting that the induced genes of unknown function may be involved in these pathways (**Figure \@ref(fig:DC8viz2019)**).

Across diffusion components that reflect differences among samples within a stage of fermentation, some fermentations from the same American Viticulture Area segregated similarly across diffusion components.
Fermentations from OR, AV, and SMV often group together (**Figure \@ref(fig:allDCs2019)**), providing support for the concept of American Viticulture Areas and regional terroir.
However, we do not observe grouping among all fermentations from the same American Viticulture Area along all diffusion components.
For example, samples from AS group together along diffusion component 5 (2 hours) and diffusion component 6 (64 hours), but not in diffusion component 8 (112 hours).
Interestingly, AS sites are separated by 1 km and yet have detectable variation in *S. cerevisiae* metabolism in primary fermentation.
Similarly, fermentations from RRV do not cluster together along any diffusion component, suggesting variation in subappellations within the Russian River Valley leads to differential outcomes in wine (**Figure \@ref(fig:allDCs2019)**).

```{r eval = F}
tmp <- dc19 %>%
  column_to_rownames("sample") %>%
  dplyr::select(-wine_tube_id, -year, -ts_sum_id, -micro_id, -tank, -tank_letter, 
                -rootstock, -ava_id, -ava, -nutristart, -dap, -initial_yan,
                -total_gdd_f, -initial_brix)
ggcorrplot(cor(tmp), show.diag = T, lab = T,
           sig.level = 0.05, p.mat = cor_pmat(tmp), insig = "blank", 
           hc.order = T, digits = 2, lab_size = 2, tl.cex = 9, 
           lab_col = "white", show.legend = T,  legend.title =  "",
           colors = brewer.pal(n = 3, name = "RdYlBu")) + 
  theme(legend.position = "bottom")

####################
ggplot(dc19, aes(x = DC9, y = 0, color = factor(hr, levels = c("0", "2", "6", "16", "64", "112")), label = ava)) +
  geom_jitter(alpha = .6, size = 2) +
  scale_color_brewer(palette = "Paired", name = "hour") + 
  theme_minimal() +
  theme(plot.title = element_text(hjust = 0.5)) 
```

```{r plot, eval = F}
ggplotly(ggplot(dc19, aes(x = DC8, y = 0, color = ava, label = ava_id)) +
  geom_jitter(size = 2) +
  theme_minimal() +
  scale_color_manual(values = c("OR"=or,  "AV" = av, "SNC" = snc, "RRV" = rrv, "CRN" = crn,
                                "AS" = as, "SMV" = smv, "SRH" = srh)) +
  theme(plot.title = element_text(hjust = 0.5)))
```

## Diffusion mapping identifies global shifts in gene expression during hypoxia

While diffusion maps successfully distinguished major and minor differences in fermentations from vineyard sites, we were curious whether this method could be applied to other RNA-sequencing data where global shifts in gene expression occur.
To address this, we identified a publicly available time series gene expression profile of the onset of hypoxia in *S. cerevisiae*.
Hypoxia occurs when a cell becomes oxygen-limited.
While oxygen is necessary for biosynthesis of some metabolites (e.g. ergosterol, heme), *S. cerevisiae* can grow in anoxic conditions if these metabolites are exogenously supplied [@parks1978metabolism].
This response is organized by transcription factors and other regulatory proteins in response to heme/oxygen availability, sterol availability, and unsaturated fatty acid availability (summarized in **Table \@ref(tab:hyptbl)**). 
Heme, sterols, and unsaturated fatty acids require oxygen for biosynthesis, so decreasing concentration of these metabolites triggers activation of transcription factors and their regulators [@parks1978metabolism; @labbe1990tetrapyrrole]. 
The heme-dependent transcription factors and regulatory proteins include Hap1, the Hap2/3/4/5 complex, Rox1, and Mot3 (reviewed in @kwast1998oxygen). 
Under normoxia, Hap1 and Hap2/3/4/5 induce genes required for respiration while Rox1 is aided by Mot3 in repressing the expression of hypoxic genes [@klinkenberg2005combinatorial].
Both Hap1 and the Hap2/3/4/5 complex induce expression of genes outside of aerobia -- namely, Hap1 is necessary for hypoxic growth while Hap2/3/4/5 regulates expression of some genes in a carbon-source dependent manner [@chantrel1998transcriptional; @pinkham1994heme].
Hap1, Hap2/3/4/5, Rox1, and Mot3 regulate less than half of all hypoxia-induced genes [@becerra2002yeast].
The transcription factor Mga2 activates lipid synthesis under hypoxia in a unsaturated fatty acid dependent manner [@kandasamy2004regulation].
Similarly, paralogous transcription factors Upc2 and Ecm22 regulate induction and repression of sterol biosynthesis, sterol transport, and DAN/TIR cell wall genes in a sterol-dependent manner [@davies2006role]. 
Ecm22 activates constitutive expression of sterol biosynthetic targets under normal laboratory conditions while Upc2 takes over under hypoxia (reviewed in @jorda2020regulation).

```{r hyptbl}
hyp_tbl <- "
1 | Oxygen sensing strategy   | Protein | Role   
2 | Heme dependent | Hap1 | Aerobic activation of respiratory genes. Required for hypoxic growth.\\textsuperscript{1} 
3 |Heme dependent | Rox1 | Aerobic repression of hypoxic genes.\\textsuperscript{2}
4 |Heme dependent | Mot3 | Aids Rox1 in aerobic repression of hypoxic genes.\\textsuperscript{2}
5 |Heme dependent | Hap2/Hap3/Hap4/Hap5 | Aerobic activation of respiratory genes.\\textsuperscript{3} Regulates some genes in a heme-independent, carbon-source dependent manner.\\textsuperscript{4}
6 |Sterol dependent | Upc2 | Hypoxic activation of sterol biosynthesis, sterol transport, and DAN/TIR cell wall genes.\\textsuperscript{5}
7 |Sterol dependent | Ecm22 | Aerobic repression of sterol biosynthesis, sterol transport, and DAN/TIR cell wall genes.\\textsuperscript{5}
8 |Unsaturated fatty acid dependent  | Mga2 | Hypoxic activation of lipid synthesis.\\textsuperscript{6}
9 |Unknown | Tpa1 | Represses Hap1 activity. Likely functions as an oxygen sensor through a prolyl-4-hydroxylase domain. Also   modulates translation termination by reducing stop codon read through.\\textsuperscript{7} 
10 |Unknown | Ett1 | No known role in hypoxia. Modulates translation termination by reducing stop codon read through. The Ett1 homolog in \\textit{S. pombe}, Nro1, performs oxygen-regulated degradation of the sterol-dependent transcription factor Sre1.\\textsuperscript{7} 
11 |Unknown | Dsc1/Dsc2/Dsc3/Dcs4 | Tul1/Dsc2/Dsc3/Ubx3 forming the Tul1 E3 ligase. No known role in hypoxia. Performs protein quality control under non-stress conditions. Homolog of \\textit{S. pombe} Dsc1 E3 ligase which is required for proteolytic activation of fungal sterol regulatory element-binding protein transcription factors.\\textsuperscript{8}"

df <- read.delim(textConnection(hyp_tbl), header=T, sep="|", strip.white=TRUE, stringsAsFactors=FALSE, row.names = 1)
row.names(df)<-NULL
kable(df, caption = "Transcription factors and regulatory proteins involved in the hypoxis response in \\textit{S. cerevisiae}. \\textsuperscript{1}(ref:chantrel1998transcriptional), \\textsuperscript{2}(ref:klinkenberg2005combinatorial), \\textsuperscript{3}(ref:pinkham1985cloning), \\textsuperscript{4}(ref:pinkham1994heme), \\textsuperscript{5}(ref:davies2006role), \\textsuperscript{6}(ref:kandasamy2004regulation), \\textsuperscript{7}(ref:henri2010structural), \\textsuperscript{8}(ref:tong2014identification).", format = "latex", booktabs = T, escape = F, 
      col.names = c("Oxygen sensing strategy", "Protein", "Role")) %>%
  kable_styling(font_size = 9)%>%
  column_spec(1, width = "1in") %>%
  column_spec(2, width = "1.5in") %>%
  column_spec(3, width = "3.5in") %>%
  collapse_rows(columns = 1, valign = "bottom")
```
(ref:chantrel1998transcriptional) @chantrel1998transcriptional
(ref:klinkenberg2005combinatorial) @klinkenberg2005combinatorial
(ref:pinkham1985cloning) @pinkham1985cloning
(ref:pinkham1994heme) @pinkham1994heme
(ref:davies2006role) @davies2006role
(ref:kandasamy2004regulation) @kandasamy2004regulation
(ref:henri2010structural) @henri2010structural
(ref:tong2014identification) @tong2014identification


Interestingly, *S. cerevisiae* hypoxic regulators deviate from those in other fungi such as the fission yeast *Schizosaccharomyces pombe*.
Highly conserved sterol regulatory element-binding proteins (SREBPs) regulate biosynthesis of lipids, ergosterol, and heme during the *S. pombe* hypoxic response [@todd2006sterol].
SREBPs have been replaced by Upc2 in *S. cerevisiae* [@maguire2014zinc].

In *S. pombe*, one SREBP transcription factor is Sre1.
Sre1 is liberated from the cell membrane when sterol levels drop due to oxygen limitation, whereby it enters the nucleus and activates hypoxic genes [@hughes2005srebp].
Sre1 is regulated by Ofd1 and Nro1. 
Ofd1 degrades Sre1 under normoxia, but Nro1 negatively regulates Ofd1 under hypoxic conditions.
While there is no known homolog of Sre1 in *S. cerevisiae* [@siso2012yeast], both Ofd1 and Nro1 have homologs.
The *S. cerevisiae* Ofd1 homolog Tpa1 modulates translation termination, but also represses Hap1 activity through an unknown mechanism [@henri2010structural].
The Nro1 homolog Ett1 also modulates translation termination presumably in the same pathway as Ofd1 and has no known role in regulation of hypoxia [@henri2010structural].
<!-- However, given that Ett1 works in concert with Tpa1 in other pathways, it is possible it is also involved in regulation of hypoxia. -->

In *S. pombe*, Sre1 cleavage from the cell membrane requires the Golgi Dsc E3 ligase, a complex that contains five subunits Dsc1-Dsc5 [@stewart2011yeast].
This complex also has a homolog in *S. cerevisiae*, the Golgi Tul1 E3 ligase, which plays a role in protein quality control under non-stress conditions [@tong2014identification]. 
However, the *S. cerevisiae* Golgi Tul1 E3 ligase is under explored, and some stress conditions may require this ligase activity [@tong2014identification]. 

We explored the transcriptional regulatory landscape of hypoxia onset using an unanalyzed, publicly available data set that captured the transition to hypoxia among 22 knockout mutants and wild type *S. cerevisiae*.
This data set included mutants of canonical aerobic and hypoxic transcription factors (*hap1*, *hap2/3/4/5*, *hap2*, *hap3*, *hap4*, *hap5*, *mot3*, *rox1*, *upc2*, *ecm22*, *mga3*), combinations of these transcription factors (*rox1/mot3*, *hap1/rox1/mot3*), and homologs of *S. pombe* SREBP hypoxic response (*dsc1*, *dsc2*, *dsc3*, *dsc5*, *dsc1/2/3/5*, *tpa1*, *ett1*, *tpa1/ett1*).
While many of these homologs have no doubt role in regulation of hypoxia, we demonstrate aberrant transcriptome during the hypoxic transition in these mutants.
We also show that along with the dominant global shift of gene expression experienced during hypoxia, there is a transient 30 minute shift that appears to be critical for long-term maintenance of metabolism during hypoxia.
Mutation of *upc2*, *ecm22*, or *mga3* arrests progression through the hypoxic transition, potentially indicating an unknown shared regulatory role for these three transcription factors.

```{r TEST_what_is_deleted, eval = F}
# figure out what dsc1 is 
tmp2 <- counts %>%
  rownames_to_column("gene") %>%
  filter(gene == "YOL073C") %>%
  select(contains("NH_78"))
  
counts_filt <- counts[rowSums(counts) > 0, ] # remove rows that sum to zero
dim(counts_filt)
tmp <- counts %>% 
  select(contains("NH_76"))
tmp$total_counts <- rowSums(tmp)
tmp2 <- tmp %>%
  rownames_to_column("gene") %>%
  select(gene, total_counts) %>%
  filter(total_counts == 0)
View(tmp2)

orfs <- orf_to_common(tmp2$gene)
```

```{r hypoxia_functions}
dc_top_enrich_plot_hyp <- function(dc, dc_num, dc_con, show_genes = F){
  # dc <- dc19
  # dc_num <- "DC3"
  # dc_con <- dc3_19
  # separate to induced and repressed -- note that filtering for 
  # log2fc is done in dc_contrast() when normalize_logFC = T. 
  # default is a normalizes logFC > abs(2) 
  # therefore, up down separates the positive and negative logFC values
  
  up <- dc_con[dc_con$logFC > 0, ]
  down <- dc_con[dc_con$logFC < 0, ] 
               
  # KEGG enrichment
  if(nrow(up) != 0){
    ekegg_up <- enrichKEGG(up$ORF, "sce", pAdjustMethod = "bonferroni") 
    ekegg_up_top <- ekegg_up@result %>%
      filter(p.adjust < 0.05) %>%
      dplyr::select(Description, geneID) %>%
      mutate(enrichment = "KEGG") 
  } else {
    ekegg_up_top <- data.frame("Description" = character(),
                               "geneID" = character(),
                               "enrichment" = character())
  }
  
  if(nrow(down) != 0){
    if(!identical(down$ORF, down$COMMON)) { # Filter out datasets where everything is unknown
      ekegg_down <- enrichKEGG(down$ORF, "sce", pAdjustMethod = "bonferroni")
      ekegg_down_top <- ekegg_down@result %>%
        filter(p.adjust < 0.05) %>%
        dplyr::select(Description, geneID) %>%
        mutate(enrichment = "KEGG")
    } else {
      ekegg_down_top <- data.frame("Description" = character(),
                                   "geneID" = character(),
                                   "enrichment" = character())
    }
  } else {
    ekegg_down_top <- data.frame("Description" = character(),
                                 "geneID" = character(),
                                 "enrichment" = character())
  }
  
  # GO enrichment
  if(nrow(up) != 0){
    ego_bp_up <- enrichGO(up$COMMON, OrgDb = org.Sc.sgd.db, keyType = "GENENAME", 
                          ont = "BP", pAdjustMethod = "bonferroni")
    ego_bp_up_top <- ego_bp_up@result %>%
      filter(p.adjust < 0.05) %>%
      dplyr::select(Description, geneID) %>%
      mutate(enrichment = "GO BP")
    
    ego_mf_up <- enrichGO(up$COMMON, OrgDb = org.Sc.sgd.db, keyType = "GENENAME", 
                          ont = "MF", pAdjustMethod = "bonferroni")
    ego_mf_up_top <- ego_mf_up@result%>%
      filter(p.adjust < 0.05) %>%
      dplyr::select(Description, geneID) %>%
      mutate(enrichment = "GO MF")
    
    ego_cc_up <- enrichGO(up$COMMON, OrgDb = org.Sc.sgd.db, keyType = "GENENAME", 
                          ont = "CC", pAdjustMethod = "bonferroni")
    ego_cc_up_top <- ego_cc_up@result %>%
      filter(p.adjust < 0.05) %>%
      dplyr::select(Description, geneID) %>%
      mutate(enrichment = "GO CC")
  } else {
    ego_cc_up_top <- ego_mf_up_top <- ego_bp_up_top <- data.frame("Description" = character(),
                                                                  "geneID" = character(),
                                                                  "enrichment" = character())
  }
  
  if(nrow(down) != 0){
    if(!identical(down$ORF, down$COMMON)) { # Filter out datasets where everything is unknown
      ego_bp_down <- enrichGO(down$COMMON, OrgDb = org.Sc.sgd.db, keyType = "GENENAME", 
                              ont = "BP", pAdjustMethod = "bonferroni")
      ego_bp_down_top <- ego_bp_down@result %>%
        filter(p.adjust < 0.05) %>%
        dplyr::select(Description, geneID) %>%
        mutate(enrichment = "GO BP")
      
      ego_mf_down <- enrichGO(down$COMMON, OrgDb = org.Sc.sgd.db, keyType = "GENENAME", 
                              ont = "MF", pAdjustMethod = "bonferroni")
      ego_mf_down_top <- ego_mf_down@result %>%
        filter(p.adjust < 0.05) %>%
        dplyr::select(Description, geneID) %>%
        mutate(enrichment = "GO MF")
      
      ego_cc_down <- enrichGO(down$COMMON, OrgDb = org.Sc.sgd.db, keyType = "GENENAME", 
                              ont = "CC", pAdjustMethod = "bonferroni")
      if(class(ego_cc_down) == "NULL") {
        message("No enriched GO CC Down, moving on!")
        # make empty df so obj is rbind compliant
        ego_cc_down_top <- data.frame("Description" = character(), 
                                      geneID = character(), 
                                      enrichment = character())
      } else {
        ego_cc_down_top <- ego_cc_down@result %>%
          filter(p.adjust < 0.05) %>%
          dplyr::select(Description, geneID) %>%
          mutate(enrichment = "GO CC")
      } 
    } else {
      ego_cc_down_top <- ego_mf_down_top <- ego_bp_down_top <- data.frame("Description" = character(),
                                                                          "geneID" = character(),
                                                                          "enrichment" = character())
    }
  }  else {
    ego_cc_down_top <- ego_mf_down_top <- ego_bp_down_top <- data.frame("Description" = character(),
                                                                        "geneID" = character(),
                                                                        "enrichment" = character())
  }
  
  # COMBINE ENRICHMENT TABLES
  if(show_genes == T){
    down_tbl <- rbind(ego_bp_down_top, ego_cc_down_top, ego_mf_down_top, ekegg_down_top) 
    up_tbl <- rbind(ego_bp_up_top, ego_cc_up_top, ego_mf_up_top, ekegg_up_top)
  } else if(show_genes == F) {
    down_tbl <- rbind(ego_bp_down_top, ego_cc_down_top, ego_mf_down_top, ekegg_down_top) %>%
      dplyr::select(-geneID, -enrichment, Pathway = Description)
    up_tbl <- rbind(ego_bp_up_top, ego_cc_up_top, ego_mf_up_top, ekegg_up_top) %>%
      dplyr::select(-geneID, -enrichment, Pathway = Description)
  }


  # PRUNE PATHWAY NAMES TO FIRST X WORDS
  up_tbl$Pathway <- gsub(" \\(.*\\)", "", up_tbl$Pathway)
  up_tbl$Pathway <-gsub(" SSU.* from tricistronic rRNA transcript", " rRNA", up_tbl$Pathway)
  down_tbl$Pathway <- gsub(" \\(.*\\)", "", down_tbl$Pathway)
  down_tbl$Pathway <-gsub(" SSU.* from tricistronic rRNA transcript", " rRNA", down_tbl$Pathway)
  
  # PLOT ANNOTATION TABLES
  if(nrow(up_tbl) == 0){
    # make an empty table if there are no results
    up_tbl <- ggtexttable(data.frame("No_sig_genes" = ""), rows = NULL, theme = ttheme(base_size = 5, base_style = "blank", padding = unit(c(.5, .5), "mm")))
  } else {
    up_tbl <- ggtexttable(up_tbl,rows = NULL, theme = ttheme(base_size = 5, base_style = "blank", padding = unit(c(.5, .5), "mm")))
  }
  
  if(nrow(down_tbl) == 0){
    # make an empty table if there are no results
    down_tbl <- ggtexttable(data.frame("No_sig_genes" = ""), rows = NULL, theme = ttheme(base_size = 5, base_style = "blank", padding = unit(c(.5, .5), "mm")))
  } else {
    down_tbl <- ggtexttable(down_tbl,rows = NULL, theme = ttheme(base_size = 5, base_style = "blank", padding = unit(c(.5, .5), "mm")))
  }
  
  # PLOT DC
  # prep DF for plotting. Rename DC* to DC so that the same innvocation can be used
  # for ggplot across all plots
  dc <- dc %>%
    dplyr::select(DC = all_of(dc_num), sample, time, deletion, category)
  
  plt_dc <- ggplot(dc, aes(x = DC, y = 0, color = time)) +
    geom_jitter() +
    scale_color_manual(values = brewer.pal(9, "Blues")[2:9]) +
    labs(x = dc_num) +
    theme_minimal() +
    theme(plot.title = element_text(hjust = 0.5),
          axis.text.y = element_blank(),
          axis.title.y = element_blank(),
          legend.position = "bottom") 


  # Plot chart and table into one object
  return(ggarrange(down_tbl, plt_dc, up_tbl, ncol = 3))
}

dc_top_diffex_plot_hyp <- function(dc, dc_num, dc_con){
  # dc <- dc19
  # dc_num <- "DC7"
  # dc_con <- dc7_19
  # color_by = "ava"
  # separate to induced and repressed -- note that filtering for 
  # log2fc is done in dc_contrast() when normalize_logFC = T. 
  # default is a normalizes logFC > abs(2) 
  # therefore, up down separates the positive and negative logFC values
  
  up_most <- dc_con %>%
    filter(logFC_adj > 0) %>%
    dplyr::select(COMMON, logFC_adj) %>%
    mutate(logFC = round(logFC_adj, digits = 1)) %>%
    dplyr::select(-logFC_adj) %>%
    head(n = 10)
  
  down_most <- dc_con %>%
    filter(logFC_adj < 0) %>%
    dplyr::select(COMMON, logFC_adj) %>%
    mutate(logFC = round(logFC_adj, digits = 1)) %>% 
    dplyr::select(-logFC_adj) %>%
    head(n = 10)
  
  # COMBINE ENRICHMENT TABLES
  # make an annotation table
  
  if(nrow(up_most) == 0){
    # make an empty table if there are no results
    up_tbl <- ggtexttable(data.frame("Pathway" = ""), rows = NULL, theme = ttheme(base_size = 6, base_style = "classic", padding = unit(c(1, 1), "mm")))
  } else{
    up_tbl <- ggtexttable(up_most, rows = NULL, theme = ttheme(base_size = 6, base_style = "classic", padding = unit(c(1, 1), "mm")))
  }
  if(nrow(down_most) == 0){
    # make an empty table if there are no results
    down_tbl <- ggtexttable(data.frame("Pathway" = ""), rows = NULL, theme = ttheme(base_size = 6, base_style = "classic", padding = unit(c(1, 1), "mm")))
  } else {
    down_tbl <- ggtexttable(down_most,rows = NULL, theme = ttheme(base_size = 6, base_style = "classic", padding = unit(c(1, 1), "mm")))
  }
  # PLOT DC
  # prep DF for plotting. Rename DC* to DC so that the same innvocation can be used
  # for ggplot across all plots
  dc <- dc %>%
    dplyr::select(DC = all_of(dc_num), time, deletion, category, sample)
  
    plt_dc <- ggplot(dc, aes(x = DC, y = 0, color = time)) +
      geom_jitter() +
      scale_color_manual(values = brewer.pal(9, "Blues")[2:9]) +
      labs(x = dc_num) +
      theme_minimal() +
      theme(plot.title = element_text(hjust = 0.5),
            axis.text.y = element_blank(),
            axis.title.y = element_blank(),
            legend.position = "bottom") 

  # Plot chart and table into one object
  return(ggarrange(down_tbl, plt_dc, up_tbl, ncol = 3))
}
```

```{r preprocess_hypoxia}
hypoxia_info <- read_csv("hypoxia/hypoxia_info.csv")
files <- list.files("hypoxia/GSE115171_RAW", pattern = ".txt.gz$", recursive = T, full.names = T)
files <- c(files, list.files("hypoxia/GSE85595_RAW", pattern = "tab.gz", recursive = T, full.names = T))
counts <- files %>%
  purrr::set_names() %>% 
  map_dfr(function(x) read_tsv(x, col_names = c("gene", "count")), .id = "source") %>%
  mutate(source = gsub("\\.tabular\\.txt\\.gz", "", source)) %>%
  mutate(source = gsub("\\.tab.gz", "", source)) %>%
  mutate(source = gsub(".*\\/", "", source)) %>%
  filter(grepl(pattern = "^Y", x = gene)) %>%
  pivot_wider(id_cols = gene, names_from = source, values_from = count) %>%
  as.data.frame() %>%
  column_to_rownames("gene")

m <- counts
m <- sweep(m, 2, colSums(m), `/`)
m <- m[grepl("^Y", rownames(m)), ]                    # subset to mRNA
m <- t(m)                                             # transpose counts
```

```{r diffusion_mapping_hypoxia}
## Build diffusion map
nm <- m %>% norm_mat()        # normalize
aff <- nm %>%
  get_euc(., n.threads = 1) %>% 
  threshold(., top_k = 20)      # make affinity matrix
Lij <- aff %>% get_laplac()     # compute laplacian
eig <- eigen(Lij)               # smallest keig vectors
evl <- eig$values %>%           # get eigenvalues
  Re() %>%
  round(., digits = 10)
evc <- eig$vectors %>%          # get eigenvectors
  Re() %>%
  round(., digits = 10)

# create objects containing diffusion components
for(d in 1:length(evl)){
  assign(paste('dim', d, sep = '_'),
         Re(evc[, rank(evl) == (d + 1)])
  )
}

k_eig <- 20                     # specify num of variables you want to keep
dat <- do.call(mapply, c(FUN = cbind, mget(paste0("dim_", 1:(k_eig))))) %>%
  t() %>%
  as.data.frame()               # merge in array

colnames(dat) <- paste(paste('DC', 1:(k_eig), sep = ''))
rownames(dat) <- rownames(Lij)  # add labels

dc_hyp <- dat %>%               # join with metadata
  rownames_to_column("sample") %>%
  mutate(sample = gsub("GSM......._", "", sample)) %>%
  left_join(hypoxia_info, by = "sample")
```

```{r, eval = F}
tmp <- data.frame(val = eig$values) %>%
  arrange(val) %>%
  mutate(num = 1:length(eig$values))

ggplotly(ggplot(tmp, aes(y = val, x = num)) + 
  geom_point() +
  theme_minimal() +
  ggtitle("k=20 hypoxia"))
```


```{r DC1hypoxia, fig.height = 2, fig.cap="Diffusion map of wildtype and 22 mutant strains exposed to 100% nitrogen for 0-240 minutes. Wildtype cells are triangles, while mutants appear as points."}
dc_hyp$time <- as.factor(dc_hyp$time)
scat1 <- ggplot(data=subset(dc_hyp, deletion != "HAP1"), aes(x = DC1, y = 0, color = time, label = deletion)) +
  geom_jitter(size = 2) +
  scale_color_manual(values = brewer.pal(9, "Blues")[2:9]) +
  theme_minimal() +
  theme(plot.title.position = "plot", 
        axis.text.y = element_blank(),
        axis.title.y = element_blank()) 

# add WT
scat1 + 
  geom_point(data=subset(dc_hyp, deletion == "HAP1"), aes(x = DC1, y = 0), shape = 17, size = 4, color = "black")+
  geom_point(data=subset(dc_hyp, deletion == "HAP1"), aes(x = DC1, y = 0), shape = 17, size = 3, fill = "black")
```

```{r diffex_hypoxia}
d0 <- DGEList(counts)                          # initialize the DGE object
d <- calcNormFactors(d0)                       # calculate normalization factors
dc_hyp_mm <- model.matrix(~dc_hyp$DC1 + ~dc_hyp$DC2 + ~dc_hyp$DC3 + dc_hyp$DC4 + dc_hyp$DC5 + ~dc_hyp$DC6 + ~dc_hyp$DC7 + dc_hyp$DC8 + dc_hyp$DC9 + dc_hyp$DC10)
dc_hyp_y <- voom(d, dc_hyp_mm, plot = F)
dc_hyp_fit <- lmFit(dc_hyp_y, dc_hyp_mm)

dc1_hyp <- dc_contrast(dc_fit = dc_hyp_fit, coeff = 2)
dc2_hyp <- dc_contrast(dc_fit = dc_hyp_fit, coeff = 3)
dc3_hyp <- dc_contrast(dc_fit = dc_hyp_fit, coeff = 4)
dc4_hyp <- dc_contrast(dc_fit = dc_hyp_fit, coeff = 5)
dc5_hyp <- dc_contrast(dc_fit = dc_hyp_fit, coeff = 6)
dc6_hyp <- dc_contrast(dc_fit = dc_hyp_fit, coeff = 7)
dc7_hyp <- dc_contrast(dc_fit = dc_hyp_fit, coeff = 8)
dc8_hyp <- dc_contrast(dc_fit = dc_hyp_fit, coeff = 9)
dc9_hyp <- dc_contrast(dc_fit = dc_hyp_fit, coeff = 10)
dc10_hyp <- dc_contrast(dc_fit = dc_hyp_fit, coeff = 11)

dc1_hyp_all <- dc_contrast(dc_fit = dc_hyp_fit, coeff = 2, adj.p.filt = 1, filter_logFC = F)
dc2_hyp_all <- dc_contrast(dc_fit = dc_hyp_fit, coeff = 2, adj.p.filt = 1, filter_logFC = F)
```

```{r dc_hyp1_filt}
# with log2fc > 2
dc1_hyp_up <- dc1_hyp %>%
  filter(logFC_adj > 0)
dc1_hyp_down <- dc1_hyp %>%
  filter(logFC_adj < 0)

# all sig
dc1_hyp_all_up <- dc1_hyp_all %>%
  filter(logFC_adj > 0) %>%
  filter(adj.P.Val < 0.05)
dc1_hyp_all_down <- dc1_hyp_all %>%
  filter(logFC_adj < 0) %>%
  filter(adj.P.Val < 0.05)
```

```{r dc_hyp2_filt}
dc2_hyp_up <- dc2_hyp %>%
  filter(logFC_adj > 0)
dc2_hyp_down <- dc2_hyp %>%
  filter(logFC_adj < 0)

dc2_hyp_all_up <- dc2_hyp_all %>%
  filter(logFC_adj > 0) %>%
  filter(adj.P.Val < 0.05)
dc2_hyp_all_down <- dc2_hyp_all %>%
  filter(logFC_adj < 0) %>%
  filter(adj.P.Val < 0.05)
```

Diffusion component 1 captured the time-dependent transition to a hypoxic phenotype (**Figure \@ref(fig:DC1hypoxia)**). 
Wild type cells experienced a sharp transition from 0-5 minutes of nitrogen exposure, indicating a fast metabolic transition to hypoxia that matured over the remainder of the time course.
We investigated whether genes differentially expressed along diffusion component 1 matched oxygen-regulated genes identified by previous studies of hypoxia.
Across seven microarray studies, 11 genes (3 aerobic, 8 hypoxic) were consistently identified as being involved in aerobiosis or anaerobiosis in at least six studies (studies compiled by @bendjilali2017time).
Of genes differentially expressed along diffusion component 1, we identified 8 of 11 hypoxia genes with a log~2~ fold change greater than two, and 11 of 11 genes with p < 0.05.
We then compared our results to a time-series RNA-sequencing profile of wild type *S. cerevisiae* during hypoxic response [@bendjilali2017time].
Two hundred thirty-nine of 291 (82.1%) aerobic genes were significantly induced prior to exposure to nitrogen, while 422 of 519 (81.3%) hypoxic genes were significantly induced after prolonged exposure to nitrogen (**Table \@ref(tab:bendjilali)**). 
We found that genes induced at time zero were enriched in pathways like ribosome biogenesis, oxidative phosphorylation, and sterol metabolic process, while genes induced after prolonged exposure to nitrogen were enriched in pathways like oxidation reduction process, cell wall, glycogen metabolic process, and glycolysis/gluconeogenesis (**Figure \@ref(fig:DC1vizhyp)**).
This matches our knowledge of the hypoxic transition in yeast [@kwast2002genomic; @bendjilali2017time].
These results indicate that the first diffusion component captured the global shift in gene expression during the hypoxic shift.

```{r oxygen_regulated_genes}
url <- "https://www.g3journal.org/highwire/filestream/473070/field_highwire_adjunct_files/15/TableS3.xlsx"
download.file(url = url, destfile = "hypoxia/inputs/tabs3_oxygen_regulated_genes.xlsx")
ox_reg <- readxl::read_excel(path = "hypoxia/inputs/tabs3_oxygen_regulated_genes.xlsx", 
                             col_names = c('ORF', 'COMMON', 'description', 'Becerra_2002',
                                           'Hickman_2007', 'Hickman_2011', 'Kwast_2002', 
                                           'Lai_2005', 'Lai_2006', 'terLinde_1999', 
                                           'terLinde_2002', 'hypoxic', 'aerobic', 'total'),
                             sheet = 1, skip = 1)

hypoxic1 <- ox_reg %>%
  group_by(hypoxic) %>%
  tally()

aerobic1 <- ox_reg %>%
  group_by(aerobic) %>%
  tally()
```

```{r test_hypoxia_intersect_microarray}
hypoxic_genes <- ox_reg %>%
  filter(hypoxic >= 6) %>%
  dplyr::select(ORF, COMMON)

hypoxic_genes_dc1_hyp <- hypoxic_genes[hypoxic_genes$ORF %in% dc1_hyp$ORF, ]

# try with log fc 1.5 instead of 2
dc1_hyp_p <- dc_contrast(dc_fit = dc_hyp_fit, coeff = 2, adj.p.filt = 1, filter_logFC = F) %>%
  filter(adj.P.Val < 0.05)
hypoxic_genes_dc1_hyp <- hypoxic_genes[hypoxic_genes$ORF %in% dc1_hyp_p$ORF, ]

```

```{r test_aerobic_intersect_microarray}
aerobic_genes <- ox_reg %>%
  filter(aerobic >= 6) %>%
  dplyr::select(ORF, COMMON)

aerobic_genes_dc1_hyp <- aerobic_genes[aerobic_genes$ORF %in% dc1_hyp$ORF, ]
aerobic_genes_dc2_hyp <- aerobic_genes[aerobic_genes$ORF %in% dc2_hyp$ORF, ]

# table(aerobic_genes_dc1_hyp$ORF %in% aerobic_genes_dc2_hyp$ORF)
# table(aerobic_genes$ORF %in% dc1_hyp$ORF)
# table(aerobic_genes$ORF %in% dc2_hyp$ORF)
```

```{r bendjilali}
url <- 'https://www.g3journal.org/highwire/filestream/473070/field_highwire_adjunct_files/13/TableS1.xlsx'
download.file(url = url, destfile = "hypoxia/inputs/tabs1_rnaseq_res.xlsx")
rnaseq_oxreg <- readxl::read_excel(path = "hypoxia/inputs/tabs1_rnaseq_res.xlsx", sheet = 1)%>%
  filter(grepl("^Y", gene))

hypoxic <- rnaseq_oxreg %>%
  filter(oxygen.regulated == "hypoxic") 

aerobic <- rnaseq_oxreg %>%
  filter(oxygen.regulated == "aerobic")

oxreg_int <- data.frame(intersection = c("DC1 induced", "DC1 repressed"),
                        hypoxic= c(table(hypoxic$gene %in% dc1_hyp_all_up$ORF)[2],
                                   table(hypoxic$gene %in% dc1_hyp_all_down$ORF)[2]),
                        aerobic = c(table(aerobic$gene %in% dc1_hyp_all_up$ORF)[2],
                                    table(aerobic$gene %in% dc1_hyp_all_down$ORF)[2]))
kable(oxreg_int, "latex", booktabs = T, col.names = c("Intersection", "Hypoxic (519 predicted)", "Aerobic (291 predicted)"),
      caption = 'Consistent hypoxic metabolic remodelling detected by Bendjilali et al. 2017 and this study. Genes induced along diffusion component 1 are involved in the hypoxic reponse while those that are repressed are involved in aerobic metabolism.') %>%
  kable_styling(font_size = 9)
```

Knockouts deviated from this pattern and had slower transitions to a hypoxic phenotype, did not acheive the hypoxic phenotype, or lacked maintenance of that phenotype over time.
Most severely, *hap1*  at 240 minutes and *hap2/3/4/5*  at 180 minutes segregated with wild type type 0 samples, indicating inability to maintain transfer to remodeling during hypoxia.
Alternatively, this pattern could be a result of a sample labeling issue an thus should be experimentally validated.
More convincingly, many mutants migrate partially to a mature hypoxic transcriptome or fail to achieve the full hypoxic response over the time course.
*upc2*, *ecm22*, *upc2/ecm22*, and *mga2* achieve the equivalent of wild type five minute remodeling by 240 minutes.
Interestingly, while *upc2* progresses marginally further along diffusion component 1 than *ecm22* and *upc2/ecm22*, these samples largely cluster together, indicating that *upc2* or *ecm22* alone was not sufficient to rescue hypoxic remodeling.
This is in contrast to previous studies which have found that for diverse genes, Ecm22 or Upc2 alone was sufficient for induction of hypoxia–regulated genes [@davies2006role; @woods2016zinc; @hofken2017ecm22; @serratore2018novel].
This potentially suggests a critical unknown interaction between Ecm22 and Upc2 during response to hypoxia.
Further, consistent arrests in the hypoxic progression between Upc2, Ecm22, and Mga2 also suggests coordinated regulation between sterol and fatty acid biosynthesis relevant hypoxic adaptation consistent with previous reports of coordination between these systems [@rice2010role].

Other knockouts that did not complete hypoxia transcriptome remodeling along diffusion component 1 include *hap2/3/4/5* and *hap3*, as well as *tpa1/ett1* and *ett1* (but not *tpa1*).
The latter findings suggests that Ett1 maybe involved in hypoxic response in a Tpa1-independent manner.

<!-- TR TODO TF binding site analysis -->
<!-- Mutants *hap1*, *hap2*, and *hap3* after at least 30 minutes of hypoxia separated from aerobic samples along diffusion component 2. -->
<!-- No pathways were enriched among genes induced in *hap1*, *hap2*, and *hap3*, however 46 of the 88 induced genes have no known function. -->
<!-- However, separation away from wild type samples reinforces the role of e.g. *hap1* transcriptional regulation hypoxia (CITE). -->

While a dominant global shift in gene expression occured in reponse to hypoxia, diffusion component 6 differentiated a transient shift at 30 minutes.
Such transient shifts in gene expression have been previously identified at this time in the hypoxic response and shown to partially overlap with the environmental stress response [@bendjilali2017time].
We found that knockouts *upc2*, *upc2/ecm22*, *mga2*, and to a lesser extent *ecm22*, failed to make this transition and segragated with the wild type time zero samples (**Figure \@ref(fig:dc6labelled)**).
This suggests Upc2, Ecm22, and Mga2 mediate the hypoxic response at this time.
However, pathways related to redox balance were enriched among genes induced in the 30 minute samples (NADPH regeneration, pentose-phosphate shunt, oxidoreductase activity) along with pathways like glycolysis/gluconeogenesis and fatty acid degradation (**Figure \@ref(fig:DC6vizhyp)**), suggesting mutants were unable to accomplish these functions. 
In contrast, iron homeostasis and transport pathways were enriched in genes induced in *upc2*, *upc2/ecm22*, *mga2* samples (**Figure \@ref(fig:DC6vizhyp)**).
This suggests that the maintenance of redox balance is a critical regulatory function of Upc2 and Mga2 at 30 minutes of hypoxia.

```{r dc6labelled, fig.height = 2, fig.cap = "Diffusion component 6 separates a 30 minute transition in hypoxia. Many mutants segregate with wild type cells, but many do not, indicating that the transcriptome in these mutants was not remodeled at this timepoint. Only samples taken at 30 minutes are plotted."}
dc6_hyp <- dc_hyp %>%
  filter(time == 30) %>%
  mutate(deletion = gsub("_", "\\/", deletion))

scat1 <- ggplot(data=subset(dc6_hyp, deletion != "HAP1"), aes(x = DC6, y = 0, color = time, label = deletion)) +
  geom_jitter(size = 2, position = position_jitter(seed = 2)) +
  scale_color_manual(values = brewer.pal(9, "Blues")[4]) +
  theme_minimal() +
  theme(plot.title.position = "plot", 
        axis.text.y = element_blank(),
        axis.title.y = element_blank()) +
  geom_label_repel(position = position_jitter(seed = 2), fontface = "italic", size = 2)

# add WT
scat1 + 
  geom_point(data=subset(dc6_hyp, deletion == "HAP1"), aes(x = DC6, y = 0), shape = 17, size = 4, color = "black")+
  geom_point(data=subset(dc6_hyp, deletion == "HAP1"), aes(x = DC6, y = 0), shape = 17, size = 3, fill = "black")
```

The 30 minute transient response was additionally blunted in *tpa1/ett1* but not in *tpa1* or *ett1* alone, suggesting that Tpa1 and Ett1 may interact synergistically to aid in hypoxic remodeling after 30 minutes of low oxygen.
Tpa1 and Ett1 physically interact [@krogan2006global].
Tpa1 represses Hap1 by an unknown mechanism, but *hap1* mutants have a wildtype phenotype along this component, suggesting that Tpa1/Ett1 may be involved in the hypoxic response via a separate mechanism.
Interestingly, *dsc1* also segregated with *mga2* and *upc2* knockouts, suggesting Dsc1 may be important for the 30 minute hypoxic response. 
However, *dsc1/2/3/5* has a wild type phenotype along this component which makes the potential role of Dsc1 unclear. 

Repeated clustering of different knockouts along multiple diffusion components revealed knockouts with similar gene expression profiles, indicating shared regulatory function of transcriptome homeostasis in hypoxia. 
Knockouts *mga2*, *upc2*, and to a lesser extent *ecm22*, clustered together along many diffusion components (diffusion component 1, diffusion component 3, diffusion component 4, diffusion component 5).
Clustering along many diffusion components, not just diffusion component 1, further suggests essential coordinated regulation by these three transcription factors during hypoxia.
Unsurprisingly, *rox1*, *mot3*, and *rox1/mot3* mutants also clustered together along many diffusion components and often with wild type samples (diffusion component 2, diffusion component 5, diffusion component 8).
This agrees with the predominant role of Rox1 in repressing expression of hypoxic genes in aerobia, and Mot3 enhancing this repression [@deckert1995multiple; @klinkenberg2005combinatorial].

<!-- TR TO DO Lit review on Rox1, Mot3 during hypoxia -->
<!-- Unsurprisingly, knockouts of aerobic repressors of hypoxic genes (*rox1*, *mot3*, *ecm22*) clustered together with wildtype cells after 120 minutes of nitrogen exposure along diffusion components XXXX.  -->
<!-- NOTE MOT3 ACTIVATES SOME GENES. -->
<!-- Mot3 activates some genes in hypoxia (CITE), and clustering with wildtype samples suggests that Mot3 activation is important for initial hypoxic remodeling but not sustained response.  -->
<!-- ROX1 LEVELS FALL AS OXYGEN DECREASES, DEREPRESSING GENES -->


Knockouts *dsc1/2/3/5*, *dsc5* (but not *dsc1*, *dsc2*, or *dsc3*), and *tpa1* (but not *tpa1/ett1* or *ett1*) from time points 180 and 240 minutes also clustered along multiple diffusion components (diffusion component 3, diffusion component 7, diffusion component 10).
These proteins are involved in protein homeostasis under normoxia [@henri2010structural; @tong2014identification].
Other than its role in repressing Hap1, Tpa1 modulates translation termination by reducing stop codon read through [@henri2010structural], while the Dsc1/2/3/5 complex performs protein quality control under non-stress conditions [@tong2014identification].
Co-clustering and segregation away from wildtype phenotype suggests that these proteins play a role in maintaining cellular homeostatis in hypoxia.
In the case of Dsc1/2/3/5, this is likely in a Dsc5 dependent manner as only knock out of *dsc5* or *dsc1/2/3/5* produced this phenotype.
However, it is unclear why *tpa1* but not *tpa1/ett1* or *ett1* produces a similar phenotype.
Of genes induced in *dsc1/2/3/5*, *dsc5*, and *tpa1* at 180 and 240 minutes of nitrogen exposure, many are targets of other hypoxia transcription factors. 
Genes induced along diffusion component 3 are enriched for targets of Upc2 (*PAU24, ARE1, HEM13, HXT3, TSA2, PAU2, PAU5, ATF2, RTA1, TIR3, DAN4, PAU23, YMR317W, AQR1, YNR071C*).
Genes induced along diffusion component 7 are enriched for targets of Hap1 (*YPC1, YCR061W, INH1, HXT7, HSP12, SOL4, QCR10, YHR022C, HXT5, HAP4, CWP2, CWP1, MOT3, YTP1, YNL194C, MDG1, ATO2, DDR2, RSB1, FAA1, FIT2, USV1, AQY1, COX26*), Mot3 (*TKL2, OM14, MIX14, FMP16, HXT7, HSP12, SOL4, QCR10, YHR022C, YHR140W, HAP4, CWP2, CWP1, YKL070W, MSC1, MOT3, YMR173W-A, YTP1, MDG1, DDR2, RSB1, PNS1, AQY1, YRO2*), Rox1 (*TKL2, OM14, HXT7, HSP12, RCK1, SOL4, HXT5, POT1, HAP4, CWP2, CWP1, ATX1, YNL195C, DDR2, RSB1, PNS1, FAA1, FIT2, USV1, OPT2, YRO2*) and Hap4 (*YPC1, FMP45, INH1, MIX14, FMP16, YFL015C, HSP12, SOL4, QCR10, HAP4, CWP1, YLR053C, YMR173W-A, ATX1, YTP1, ATO2, PNS1, FAA1, FIT2, USV1, BDH2, YRO2*).
Genes induced along diffusion component 10 are enriched for targets of Upc2 (*PAU2, CYC7, RTA1, PAU13, CYC1, PAU23, PBI1*), Hap1 (*CYC7, CYC1, HAP4, TMA10, HUG1, YER053C-A, COX26, YPR145C-A*), and Hap4 (*YDR210W, CYC7, RGI1, RTA1, CYC1, HAP4, TMA10, HUG1, PBI1*).
Genes that are induced are classicaly aerobic (*CYC1*, *CWP1*, *CWP2*) and anaerobic (*CYC7*, *HUG1*, *DAN1*), and include transcription factors themselves (*MOT3*, *UPC2*, *HAP4*).
These results suggest a role for *dsc1/2/3/5*, *dsc5*, and *tpa1* in regulating the transcriptome during sustained exposure to low oxygen.

```{r plot_2d, eval = F}
dc_hyp$time <- as.factor(dc_hyp$time)
scat1 <- ggplot(data=subset(dc_hyp, deletion != "HAP1"), aes(x = DC7, y = DC10, color = time, label = deletion)) +
  geom_jitter(size = 2) +
  scale_color_manual(values = brewer.pal(9, "Blues")[2:9]) +
  theme_minimal() +
  theme(plot.title.position = "plot") 

# add WT
scat <- scat1 + 
  geom_point(data=subset(dc_hyp, deletion == "HAP1"), aes(x = DC7, y = DC10), shape = 17, size = 4, color = "black")+
  geom_point(data=subset(dc_hyp, deletion == "HAP1"), aes(x = DC7, y = DC10), shape = 17, size = 3, fill = "black")
ggplotly(scat)
```

Interestingly, while some mutants segregated to the extrema of some diffusion components, the deleted transcription factor was not reciprocally induced by samples at the opposite extrema, indicating that expression of transcription factor targets drives more similarity between samples than expression of the transcription factors themselves.
This is expected given the role that transcription factors play in activating and repressing many genes.
One exception to this is diffusion component 7, where *mot3* and *rox1/mot3* knockouts segregate to one extrema, and *MOT3* is induced at the other extrema (**Figure \@ref(fig:tophypoxia)**).

<!-- TR TO DO differentiate T=0 samples -->
<!-- Knockouts of aerobic repressors of hypoxic genes (*rox1*, *mot3*, *ecm22*) and aerobic activators of aerobic genes (*hap4*, *hap5*, *hap2/hap3/hap4/hap5*) displayed altered transcriptomes at time zero during aerobic metabolism. -->
<!-- Time 0, DC2/3 -- mot3, rox1, hap1/mot3/rox1, hap4 -->
<!-- Time 0, DC2/3 -- tpa1, ett1, hap1 -->
<!-- + DC5, 0min: rox1, mot3, hap4, hap5, hap2/hap3/hap4/hap5, dcs1 inappropriately induce anaerobic genes under full oxygen. -->
<!-- + genes may be repressed in part by these transcription factors, and induction under aerobic conditions may indicate insufficient repression in the absence of these transcription factors. -->

While many of the genes that were differentially expressed along diffusion components encode proteins of known function, genes induced at both extrema of diffusion component 4 are unknown.
Knockouts *upc2*, *mga2*, and to a lesser extent *ecm22* from all time points segregated to one extrema, while knockouts *dsc2*, *dsc3*, *dsc5*, *tpa1*, *hap1*, *hap2* and *hap3* from times zero and five minutes segregated to the other.
Of 581 genes induced in the former set, 387 are unknown including nine of the top 10 induced genes, while of 139 genes induced in the latter set, 53 are of unknown function including eight of the top 10 induced genes (**Figure \@ref(fig:tophypoxia)**).
Induced genes that were annotated were enriched in pathways like ion transport and iron ion transport (**Figure \@ref(fig:DC4vizhyp)**).
Among genes that were induced in the *upc2* *mga2* knockouts, pathways involved in meiosis such as spore wall assembly and reproductive process were enriched  (**Figure \@ref(fig:DC4vizhyp)**).
Given that meitotically produced spores are a highly adaptive state for stress resistance (reviewed in @bernstein2013evolutionary), this suggests that Upc2 and Mga2 are needed for hypoxic survival and without these transcription factors *S. cerevisiae* resorts to sporulation.
Unknown genes induced in these samples may be involved in meitotic sporulation.

# Conclusion

In this study, we paired diffusion mapping with differential expression and captured global shifts in gene expression. 
This method revealed differences in primary fermentation of Pinot noir wine from 15 vineyard sites, as well as regulatory functions of yeast transcription factors during hypoxia.
Diffusion mapping was especially well suited for these studies as in both cases groups (fermentations, cell cultures) were asynchronous in progression through transcriptome shifts with respect to absolute time.

Through our analysis of wine fermentations, we found vineyard-site specific gene expression patterns that correlated with *H. uvarum* metabolic activity and initial nitrogen composition of grape must. 
While these factors are known to impact wine outcomes in controlled laboratory fermentations, our analysis demonstrates that these factors are detectable amongst large chemical and biological diversity in grape musts and fermentations.

Through our analysis of hypoxia onset, we found that deletion of some transcription factors arrests progression through hypoxic transcriptome remodeling, while others make little noticeable impact on the transcriptome compared to wild type.
Surprisingly, as transcriptomes of *ett1*, *tpa1*, and *dsc1/2/3/5* deviated from wild type phenotype along multiple diffusion components, this analysis support a role for these proteins in hypoxia response.
These proteins are homologs of *S. pombe* SREBP hypoxic response genes, and while Tpa1 represses Hap1 through an unknown mechanism, these proteins have no other known role in the hypoxic response.
In this experiment, no exogenous heme or ergosterol were supplied. 
These conditions may lead to involvement of Ett1, Tpa1, and Dsc1/2/3/5 in hypoxia, however more research is needed to fully elucide the roles of these proteins in hypoxia.
<!-- The Dsc1/2/3/5 complex is involved in cellular protein homeostasis under normal laboratory conditions [@tong2014indentification], and our data further support a role for  -->

# Methods

## Sampling, sequencing and preprocessing of wine fermentation samples

We detail fermentation conditions, sampling, nucleotide extraction, sequencing in Chapter [1](#chp-brix).
We detail read preprocessing and quantification procedures for *S. cerevisiae* in Chapter [1](#chp-brix) and for non-*Saccharomyces* microorganisms in Chapter [3](#chp-site).

## Hypoxia data set

We downloaded gene expression matrices using from GEO using accession numbers GSE85595 and GSE115171.

## Construction of Diffusion Maps

We built diffusion maps as implemented previously [@fahimipour2020mapping].
To build diffusion maps from wine fermentations samples, we used `k = 10` nearest samples, while for hypoxia we used `k = 20`.
We increased the `k` size for hypoxia given the larger number of samples (n = 146 in 2017 vintage, n = 150 in 2019 vintage, and n = 336 in hypoxia).

Prior to diffusion map construction, we filtered gene counts to non-mitochondrial mRNA and normalized read counts based on total number of reads per sample (library size). 

## Differential expression

To determine which genes drove separation of samples along each component, we used differential expression to correlate each gene with diffusion component values.
We used limma to fit a linear regression model to each gene [@limma2015].
As input to differential expression, we used raw sequencing counts and performed filtering and normalization with the limma package using the `calcNormFactors()` function [@limma2015].
Using this model, the log~2~ fold change is the slope of the line for each unit increase in the diffusion component. 
We normalized log~2~ fold change values by calculating the length of the diffusion component and multiplying all log~2~ fold change values by this amount, `(max - minimum) * log~2~ fold change`. 
We analyzed log~2~ fold change values greater than two, i.e. genes with a log~2~ fold change of at least 2 between the most separated samples along a diffusion component.
We performed Gene Ontology and KEGG enrichment analysis using the R clusterProfiler package [@yu2012clusterprofiler].

# References

<div id="refs"></div>


\newpage
# Supplementary material {-}

\beginsupplement

```{r brix_upset17}
brix_list_17 <- list(dc1_17_induced = filter(dc1_17, logFC > 0)$ORF,
                     dc1_17_repressed = filter(dc1_17, logFC < 0)$ORF,
                     brix_17_induced = filter(brix_17, logFC > 0)$ORF,
                     brix_17_repressed = filter(brix_17, logFC < -0)$ORF
)
upset_brix_17 <- UpSetR::upset(fromList(brix_list_17), nsets = 8, nintersects = 200, 
                       order.by = "freq", keep.order = T)
```

```{r brix_upset19}
brix_list_19 <- list(dc1_19_induced = filter(dc1_19, logFC > 0)$ORF,
                     dc1_19_repressed = filter(dc1_19, logFC < 0)$ORF,
                     brix_19_induced = filter(brix_19, logFC > 0)$ORF,
                     brix_19_repressed = filter(brix_19, logFC < 0)$ORF
)
upset_brix_19 <- UpSetR::upset(fromList(brix_list_19), nsets = 8, nintersects = 200, 
                               order.by = "freq", keep.order = T)
```

```{r dc1_upset}
dc1_list <- list(dc1_17_induced = filter(dc1_17, logFC > 0)$ORF,
                 dc1_17_repressed = filter(dc1_17, logFC < 0)$ORF,
                 dc1_19_induced = filter(dc1_19, logFC > 0)$ORF,
                 dc1_19_repressed = filter(dc1_19, logFC < 0)$ORF)
upset_dc1 <- UpSetR::upset(fromList(dc1_list), nsets = 8, nintersects = 200, 
                   order.by = "freq", keep.order = T)
```

```{r brix_upset}
brix_list <- list(brix_17_induced = filter(brix_17, logFC > 0)$ORF,
                  brix_17_repressed = filter(brix_17, logFC < -0)$ORF,
                  brix_19_induced = filter(brix_19, logFC > 0)$ORF,
                  brix_19_repressed = filter(brix_19, logFC < 0)$ORF)
upset_brix <- UpSetR::upset(fromList(brix_list), nsets = 8, nintersects = 200, 
                   order.by = "freq", keep.order = T)
```

```{r upsetgrid, fig.height= 5, fig.cap="Upset plot depicting intersections between differentially expressed genes using diffusion component 1 or brix as continuous variable. **A.** Intersection of differentially expressed genes from the 2017 vintage. **B.** Intersection of genes from the 2019 vintage. **C.** Intersection of genes differentially expressed along diffusion component 1 from the 2017 and 2019 vintages. **D** Intersection of genes differentially expressed relative to change in brix from the 2017 and 2019 vintages."}

plot_grid(as.grob(upset_brix_17), as.grob(upset_brix_19), 
          as.grob(upset_dc1), as.grob(upset_brix),
          ncol = 2, labels = c("A", "B", "C", "D"))
```

```{r try_all_by_all_dc_corr, eval = F}

# brix, hr, initial_brix, initial_ph,
# initial_nopa, initial_nh3, initial_MA, lon, lat, total_gdd_c, total_ppt
dc1_17_all <- dc_contrast(dc_fit = dc_fit17, coeff = 2,
                          adj.p.filt = 1, filter_logFC = F) %>%
  arrange(ORF) %>%
  select(ORF, COMMON, dc1_17_lfc = logFC_adj)
dc2_17_all <- dc_contrast(dc_fit = dc_fit17, coeff = 3,
                          adj.p.filt = 1, filter_logFC = F)  %>%
  arrange(ORF) %>%
  select(ORF, COMMON, dc2_17_lfc = logFC_adj)
dc3_17_all <- dc_contrast(dc_fit = dc_fit17, coeff = 4,
                         adj.p.filt = 1, filter_logFC = F)  %>%
  arrange(ORF) %>%
  select(ORF, COMMON, dc3_17_lfc = logFC_adj)
dc4_17_all <- dc_contrast(dc_fit = dc_fit17, coeff = 5,
                          adj.p.filt = 1, filter_logFC = F)  %>%
  arrange(ORF) %>%
  select(ORF, COMMON, dc4_17_lfc = logFC_adj)
dc5_17_all <- dc_contrast(dc_fit = dc_fit17, coeff = 6,
                          adj.p.filt = 1, filter_logFC = F)  %>%
  arrange(ORF) %>%
  select(ORF, COMMON, dc5_17_lfc = logFC_adj)
dc6_17_all <- dc_contrast(dc_fit = dc_fit17, coeff = 7,
                          adj.p.filt = 1, filter_logFC = F) %>%
  arrange(ORF) %>%
  select(ORF, COMMON, dc6_17_lfc = logFC_adj)
dc7_17_all <- dc_contrast(dc_fit = dc_fit17, coeff = 8,
                          adj.p.filt = 1, filter_logFC = F)  %>%
  arrange(ORF) %>%
  select(ORF, COMMON, dc7_17_lfc = logFC_adj)
dc8_17_all <- dc_contrast(dc_fit = dc_fit17, coeff = 9,
                          adj.p.filt = 1, filter_logFC = F)  %>%
  arrange(ORF) %>%
  select(ORF, COMMON, dc8_17_lfc = logFC_adj)
dc9_17_all <- dc_contrast(dc_fit = dc_fit17, coeff = 10,
                          adj.p.filt = 1, filter_logFC = F)  %>%
  arrange(ORF) %>%
  select(ORF, COMMON, dc9_17_lfc = logFC_adj)
dc10_17_all <- dc_contrast(dc_fit = dc_fit17, coeff = 11,
                          adj.p.filt = 1, filter_logFC = F)  %>%
  arrange(ORF) %>%
  select(ORF, COMMON, dc10_17_lfc = logFC_adj)

## 2019
dc1_19_all <- dc_contrast(dc_fit = dc_fit19, coeff = 2,
                          adj.p.filt = 1, filter_logFC = F) %>%
  arrange(ORF) %>%
  select(ORF, COMMON, dc1_19_lfc = logFC_adj)
dc2_19_all <- dc_contrast(dc_fit = dc_fit19, coeff = 3,
                          adj.p.filt = 1, filter_logFC = F)  %>%
  arrange(ORF) %>%
  select(ORF, COMMON, dc2_19_lfc = logFC_adj)
dc3_19_all <- dc_contrast(dc_fit = dc_fit19, coeff = 4,
                         adj.p.filt = 1, filter_logFC = F)  %>%
  arrange(ORF) %>%
  select(ORF, COMMON, dc3_19_lfc = logFC_adj)
dc4_19_all <- dc_contrast(dc_fit = dc_fit19, coeff = 5,
                          adj.p.filt = 1, filter_logFC = F)  %>%
  arrange(ORF) %>%
  select(ORF, COMMON, dc4_19_lfc = logFC_adj)
dc5_19_all <- dc_contrast(dc_fit = dc_fit19, coeff = 6,
                          adj.p.filt = 1, filter_logFC = F)  %>%
  arrange(ORF) %>%
  select(ORF, COMMON, dc5_19_lfc = logFC_adj)
dc6_19_all <- dc_contrast(dc_fit = dc_fit19, coeff = 7,
                          adj.p.filt = 1, filter_logFC = F) %>%
  arrange(ORF) %>%
  select(ORF, COMMON, dc6_19_lfc = logFC_adj)
dc7_19_all <- dc_contrast(dc_fit = dc_fit19, coeff = 8,
                          adj.p.filt = 1, filter_logFC = F)  %>%
  arrange(ORF) %>%
  select(ORF, COMMON, dc7_19_lfc = logFC_adj)
dc8_19_all <- dc_contrast(dc_fit = dc_fit19, coeff = 9,
                          adj.p.filt = 1, filter_logFC = F)  %>%
  arrange(ORF) %>%
  select(ORF, COMMON, dc8_19_lfc = logFC_adj)
dc9_19_all <- dc_contrast(dc_fit = dc_fit19, coeff = 10,
                          adj.p.filt = 1, filter_logFC = F)  %>%
  arrange(ORF) %>%
  select(ORF, COMMON, dc9_19_lfc = logFC_adj)
dc10_19_all <- dc_contrast(dc_fit = dc_fit19, coeff = 11,
                          adj.p.filt = 1, filter_logFC = F)  %>%
  arrange(ORF) %>%
  select(ORF, COMMON, dc10_19_lfc = logFC_adj)

## COMBINE
dc_lfc_all <- list(dc1_17_all, dc2_17_all, dc3_17_all, dc4_17_all,
                   dc5_17_all, dc6_17_all, dc7_17_all, dc8_17_all,
                   dc9_17_all, dc10_17_all, dc1_19_all, dc2_19_all, 
                   dc3_19_all, dc4_19_all, dc5_19_all, dc6_19_all, 
                   dc7_19_all, dc8_19_all, dc9_19_all, dc10_19_all) %>% 
  purrr::reduce(left_join, by = c("ORF", "COMMON")) %>%
  select(-ORF) %>%
  column_to_rownames("COMMON")
corr_all <- cor(dc_lfc_all) 
is.na(corr_all) <- abs(corr_all) < 0.5
ggcorrplot(corr_all, type = "upper", show.diag = T, lab = T,
           digits = 2, lab_size = 3, tl.cex = 10, 
           lab_col = "white", show.legend = T,  legend.title =  "",
           colors = brewer.pal(n = 3, name = "RdYlBu")) + 
  theme(legend.position = "bottom") 



ggcorrplot(cor(dc_lfc_all), type = "upper", show.diag = F, lab = T,
           sig.level = 0.0000001, p.mat = cor_pmat(dc_lfc_all), insig = "blank", 
           hc.order = T, digits = 2, lab_size = 2, tl.cex = 9, 
           lab_col = "white", show.legend = T,  legend.title =  "",
           colors = brewer.pal(n = 3, name = "RdYlBu")) + 
  theme(legend.position = "bottom")
```

```{r annotatedplts17, eval = F, fig.height = 4, fig.width = 6, fig.cap="**A**, **B**"}
ggarrange(dc_top_diffex_plot(dc = dc17, dc_num = "DC1", dc_con = dc1_17, color_by = "brix"),
          dc_top_diffex_plot(dc = dc17, dc_num = "DC2", dc_con = dc2_17, color_by = "ava"),
          nrow = 2, labels = c("A", "B"))
```

<!-- TOP GENES -->

```{r annotatedplts19hr, fig.height = 6, fig.width = 6, eval = T, fig.cap="Top genes induced along diffusion components that separate stages of fermentation in the 2019 vintage. Colored by hour at which samples were taken."}
ggarrange(dc_top_diffex_plot(dc = dc19, dc_num = "DC2", dc_con = dc2_19, color_by = "hr"),
          dc_top_diffex_plot(dc = dc19, dc_num = "DC3", dc_con = dc3_19, color_by = "hr"),
          dc_top_diffex_plot(dc = dc19, dc_num = "DC4", dc_con = dc4_19, color_by = "hr"),
          nrow = 3, labels = c("A", "B", "C"))
```

```{r annotatedplts19ava, fig.height = 8, fig.width=6, eval = T, fig.cap="Top genes induced along diffusion components that separate fermentations within a stage of fermentation in the 2019 vintage. Colored by American Viticulture Area (AVA) of each sample."}
ggarrange(dc_top_diffex_plot(dc = dc19, dc_num = "DC5", dc_con = dc5_19, color_by = "ava"),
          dc_top_diffex_plot(dc = dc19, dc_num = "DC6", dc_con = dc6_19, color_by = "ava"),
          dc_top_diffex_plot(dc = dc19, dc_num = "DC7", dc_con = dc7_19, color_by = "ava"),
          dc_top_diffex_plot(dc = dc19, dc_num = "DC8", dc_con = dc8_19, color_by = "ava"),
          nrow = 4, labels = c("A", "B", "C", "D"))
```

<!-- ENRICHMENT -->


```{r DC2viz2019, fig.height = 6.5, fig.cap= "Gene set enrichment for differentially expressed genes along diffusion component 2 in the 2019 vintage. All enriched categories with p < 0.05 after Bonferroni correction are shown. Pathways on the right side of the figure are induced in samples with a high diffusion component value, while pathways on the left of the figure are induced in samples with a low diffusion component value."}
dc_top_enrich_plot(dc = dc19, dc_num = "DC2", dc_con = dc2_19, color_by = "hr", show_genes = F)
```

```{r DC3viz2019, fig.height = 7.5, fig.cap= "Gene set enrichment for differentially expressed genes along diffusion component 3 in the 2019 vintage. All enriched categories with p < 0.05 after Bonferroni correction are shown. Pathways on the right side of the figure are induced in samples with a high diffusion component value, while pathways on the left of the figure are induced in samples with a low diffusion component value."}
dc_top_enrich_plot(dc = dc19, dc_num = "DC3", dc_con = dc3_19, color_by = "hr", show_genes = F)
```

```{r DC4viz2019, fig.height = 3.2, fig.cap="Gene set enrichment for differentially expressed genes along diffusion component 4 in the 2019 vintage. All enriched categories with p < 0.05 after Bonferroni correction are shown. Pathways on the right side of the figure are induced in samples with a high diffusion component value, while pathways on the left of the figure are induced in samples with a low diffusion component value."}
dc_top_enrich_plot(dc = dc19, dc_num = "DC4", dc_con = dc4_19, color_by = "hr", show_genes = F)
```

```{r DC5viz2019, fig.height = 2, fig.cap="Gene set enrichment for differentially expressed genes along diffusion component 5 in the 2019 vintage. All enriched categories with p < 0.05 after Bonferroni correction are shown. Pathways on the right side of the figure are induced in samples with a high diffusion component value, while pathways on the left of the figure are induced in samples with a low diffusion component value."}
dc_top_enrich_plot(dc = dc19, dc_num = "DC5", dc_con = dc5_19, color_by = "ava", show_genes = F)
```

```{r DC8viz2019, fig.height = 2, fig.cap="Gene set enrichment for differentially expressed genes along diffusion component 5 in the 2019 vintage. All enriched categories with p < 0.05 after Bonferroni correction are shown. Pathways on the right side of the figure are induced in samples with a high diffusion component value, while pathways on the left of the figure are induced in samples with a low diffusion component value."}
dc_top_enrich_plot(dc = dc19, dc_num = "DC8", dc_con = dc8_19, color_by = "ava", show_genes = F)
```

## Hypoxia

```{r tophypoxia, fig.height = 4, fig.cap= "Top genes induced along diffusion components during the hypoxic response. Colored by time at which samples were taken."}
plt_dc4_top_hyp <- dc_top_diffex_plot_hyp(dc = dc_hyp, dc_num = "DC4", dc_con = dc4_hyp)
plt_dc7_top_hyp <- dc_top_diffex_plot_hyp(dc = dc_hyp, dc_num = "DC7", dc_con = dc7_hyp)
ggarrange(plt_dc4_top_hyp, plt_dc7_top_hyp, nrow = 2)
```

```{r DC1vizhyp, fig.height = 6, fig.cap="Gene set enrichment for differentially expressed genes along diffusion component 1 during onset of hypoxia. All enriched categories with p < 0.05 after Bonferroni correction are shown. Pathways on the right side of the figure are induced in samples with a high diffusion component value, while pathways on the left of the figure are induced in samples with a low diffusion component value."}
dc_top_enrich_plot_hyp(dc = dc_hyp, dc_num = "DC1", dc_con = dc1_hyp)
```

```{r DC4vizhyp, fig.height = 3, fig.cap="Gene set enrichment for differentially expressed genes along diffusion component 4 during onset of hypoxia. All enriched categories with p < 0.05 after Bonferroni correction are shown. Pathways on the right side of the figure are induced in samples with a high diffusion component value, while pathways on the left of the figure are induced in samples with a low diffusion component value."}
dc_top_enrich_plot_hyp(dc = dc_hyp, dc_num = "DC6", dc_con = dc6_hyp)
```

```{r DC6vizhyp, fig.height = 3, fig.cap="Gene set enrichment for differentially expressed genes along diffusion component 6 during onset of hypoxia. All enriched categories with p < 0.05 after Bonferroni correction are shown. Pathways on the right side of the figure are induced in samples with a high diffusion component value, while pathways on the left of the figure are induced in samples with a low diffusion component value."}
dc_top_enrich_plot_hyp(dc = dc_hyp, dc_num = "DC6", dc_con = dc6_hyp)
```

```{r TEST_plot_hypoxia, eval=F}
plot_map_hypoxia <- function(dc_hyp, dc_num){
  dc_hyp$time <- as.factor(dc_hyp$time)
  dc_hyp <- dc_hyp %>%
    dplyr::select(DC = all_of(dc_num), time, deletion, category, sample)
  scat1 <- ggplot(data=subset(dc_hyp, deletion != "HAP1"), aes(x = DC, y = 0, 
                                                               color = time, 
                                                               label = deletion)) +
    geom_jitter(size = 2) +
    scale_color_manual(values = brewer.pal(9, "Blues")[2:9]) +
    # scale_shape_manual(values = c(rox1 = 2, mot3 = 6, rox1_mot3 = 11,
    #                               upc2 = 3, ecm22 = 4, upc2_ecm22 = 3, mga2 = 8,
    #                               tpa1 = 0, ett1 = 5, tpa1_ett1 = 0, 
    #                               dsc1 = 7, dsc2 = 9, dsc3= 12, dsc5 = 14, dsc1_dsc2_dsc3_dsc5 = 15,
    #                               hap1 = 17, hap1_rox1_mot3 = 18,
    #                               hap2 = 20, hap3 = 10, hap4= 13, hap5 = 1, hap2_hap3_hap4_hap5 = 19)) +
    theme_minimal() +
    theme(plot.title.position = "plot") +
    labs(x = dc_num)
  
  # add WT
  scat1 <- scat1 + 
    geom_point(data=subset(dc_hyp, deletion == "HAP1"), aes(x = DC, y = 0), shape = 17, size = 4, color = "black") +
    geom_point(data=subset(dc_hyp, deletion == "HAP1"), aes(x = DC, y = 0), shape = 17, size = 3, fill = "black") 
  return(scat1)
}

ggplotly(plot_map_hypoxia(dc_hyp, "DC1"))
ggplotly(plot_map_hypoxia(dc_hyp, "DC2"))
ggplotly(plot_map_hypoxia(dc_hyp, "DC3"))
ggplotly(plot_map_hypoxia(dc_hyp, "DC4"))
ggplotly(plot_map_hypoxia(dc_hyp, "DC5"))
ggplotly(plot_map_hypoxia(dc_hyp, "DC6"))
ggplotly(plot_map_hypoxia(dc_hyp, "DC7"))
ggplotly(plot_map_hypoxia(dc_hyp, 'DC8'))
ggplotly(plot_map_hypoxia(dc_hyp, "DC9"))
ggplotly(plot_map_hypoxia(dc_hyp, "DC10"))
ggplotly(plot_map_hypoxia(dc_hyp, "DC11"))
ggplotly(plot_map_hypoxia(dc_hyp, "DC12"))
ggplotly(plot_map_hypoxia(dc_hyp, "DC13"))
ggplotly(plot_map_hypoxia(dc_hyp, "DC14"))
ggplotly(plot_map_hypoxia(dc_hyp, "DC15"))
ggplotly(plot_map_hypoxia(dc_hyp, "DC16"))
ggplotly(plot_map_hypoxia(dc_hyp, "DC17"))
ggplotly(plot_map_hypoxia(dc_hyp, "DC18"))
ggplotly(plot_map_hypoxia(dc_hyp, "DC19"))
ggplotly(plot_map_hypoxia(dc_hyp, "DC20"))

```

```{r TEST_plot_2d, eval = F}
dc_hyp$time <- as.factor(dc_hyp$time)
scat1 <- ggplot(data=subset(dc_hyp, deletion != "HAP1"), aes(x = DC2, y = DC5, color = time, label = deletion)) +
  geom_jitter(size = 2) +
  scale_color_manual(values = brewer.pal(9, "Blues")[2:9]) +
  # scale_shape_manual(values = c(rox1 = 2, mot3 = 6, rox1_mot3 = 11,
  #                               upc2 = 3, ecm22 = 4, upc2_ecm22 = 3, mga2 = 8,
  #                               tpa1 = 0, ett1 = 5, tpa1_ett1 = 0, 
  #                               dsc1 = 7, dsc2 = 9, dsc3= 12, dsc5 = 14, dsc1_dsc2_dsc3_dsc5 = 15,
  #                               hap1 = 17, hap1_rox1_mot3 = 18,
  #                               hap2 = 20, hap3 = 10, hap4= 13, hap5 = 1, hap2_hap3_hap4_hap5 = 19)) +
  #scale_color_brewer(palette = "Set1") +
  theme_minimal() +
  theme(plot.title.position = "plot") 

# add WT
scat <- scat1 + 
  geom_point(data=subset(dc_hyp, deletion == "HAP1"), aes(x = DC2, y = DC5), shape = 17, size = 4, color = "black")+
  geom_point(data=subset(dc_hyp, deletion == "HAP1"), aes(x = DC2, y = DC5), shape = 17, size = 3, fill = "black")
ggplotly(scat)
```
